<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>12 Clonal analysis pseudo-diploid cells | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</title>
  <meta name="description" content="12 Clonal analysis pseudo-diploid cells | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="12 Clonal analysis pseudo-diploid cells | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="12 Clonal analysis pseudo-diploid cells | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="whole-genome-sequencing-of-blood.html"/>
<link rel="next" href="phylogenetic-analysis-of-breast-cancer.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>4</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="4.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>4.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="4.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>4.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="4.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>4.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="4.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>4.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="4.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>4.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="4.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>4.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="4.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>4.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="4.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>4.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>5</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="5.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>5.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="5.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>5.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="5.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>5.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="5.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>5.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>6</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="6.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>6.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="6.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>6.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="6.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>6.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>7</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="7.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>7.1</b> Loading in data</a></li>
<li class="chapter" data-level="7.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>7.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="7.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>7.3</b> Training the model</a></li>
<li class="chapter" data-level="7.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>7.4</b> Validating the model</a></li>
<li class="chapter" data-level="7.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>7.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>8</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="8.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>8.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>9</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="9.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>9.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="9.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>9.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>10</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>10.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>11</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="11.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>11.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>12</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="12.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>12.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="12.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>12.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="12.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>12.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="12.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>12.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="12.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>12.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>13</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="13.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>13.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="13.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>13.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>14</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="14.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>14.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>16</b> Monster cells</a>
<ul>
<li class="chapter" data-level="16.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>16.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="16.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>16.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="16.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>16.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="validation-of-pseudodiploid-deletions.html"><a href="validation-of-pseudodiploid-deletions.html"><i class="fa fa-check"></i><b>17</b> Validation of pseudodiploid deletions</a></li>
<li class="chapter" data-level="18" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>18</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="18.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>18.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="18.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>18.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="18.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>18.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="18.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>18.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html"><i class="fa fa-check"></i><b>19</b> COSMIC deletions and targeted panel mutations</a>
<ul>
<li class="chapter" data-level="19.1" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>19.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="19.2" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>19.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>20</b> Session info</a>
<ul>
<li class="chapter" data-level="20.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>20.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clonal-analysis-pseudo-diploid-cells" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">12</span> Clonal analysis pseudo-diploid cells<a href="clonal-analysis-pseudo-diploid-cells.html#clonal-analysis-pseudo-diploid-cells" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This section produces all the figures used for Figure 2.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb94-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span>
<span id="cb94-3"><a href="clonal-analysis-pseudo-diploid-cells.html#cb94-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-4"><a href="clonal-analysis-pseudo-diploid-cells.html#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load functions</span></span>
<span id="cb94-5"><a href="clonal-analysis-pseudo-diploid-cells.html#cb94-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotHeatmap.R&quot;</span>)</span></code></pre></div>
<div id="running-medicc2" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> Running MEDICC2<a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To run MEDICC2 on the pseudo-diploid cells we have to prepare the data. We won’t save this table here, but you can save this and use this as input for MEDICC2. We only show how to run this for P3, but this is the exact same for P6, simply change the input while you load in the data. Because these steps take some time, we don’t actually run this (<code>eval = FALSE</code>) in this Rmarkdown file.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in data</span></span>
<span id="cb95-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-2" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_pseudodiploid_500kb.rds&quot;</span>)</span>
<span id="cb95-3"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-4"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Melt</span></span>
<span id="cb95-5"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-5" aria-hidden="true" tabindex="-1"></a>dt_m <span class="ot">=</span> <span class="fu">melt</span>(dt, <span class="at">id.vars =</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>))</span>
<span id="cb95-6"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-7"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Reduce data table</span></span>
<span id="cb95-8"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-8" aria-hidden="true" tabindex="-1"></a>dt_short <span class="ot">=</span> dt_m[, <span class="fu">as.data.table</span>(<span class="fu">reduce</span>(<span class="fu">IRanges</span>(start, end))), by <span class="ot">=</span> .(chr, variable, value)]</span>
<span id="cb95-9"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-9" aria-hidden="true" tabindex="-1"></a>dt_short <span class="ot">=</span> <span class="fu">dcast</span>(dt_short, chr <span class="sc">+</span> start <span class="sc">+</span> end <span class="sc">+</span> width <span class="sc">~</span> variable)</span>
<span id="cb95-10"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-11"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill NAs</span></span>
<span id="cb95-12"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-12" aria-hidden="true" tabindex="-1"></a>dt_short <span class="ot">=</span> dt_short <span class="sc">%&gt;%</span></span>
<span id="cb95-13"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb95-14"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fill</span>(<span class="fu">names</span>(.), <span class="at">.direction =</span> <span class="st">&quot;updown&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb95-15"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb95-16"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setDT</span>()</span>
<span id="cb95-17"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-18"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Add diploid</span></span>
<span id="cb95-19"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-19" aria-hidden="true" tabindex="-1"></a>dt_short[, diploid <span class="sc">:</span><span class="er">=</span> <span class="dv">2</span>]</span>
<span id="cb95-20"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-20" aria-hidden="true" tabindex="-1"></a>dt_short[chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>, diploid <span class="sc">:</span><span class="er">=</span> <span class="dv">1</span>]</span>
<span id="cb95-21"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-22"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique start/end sites (unique based on start site) and go back to long format</span></span>
<span id="cb95-23"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-23" aria-hidden="true" tabindex="-1"></a>dt_final <span class="ot">=</span> <span class="fu">unique</span>(dt_short, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>))</span>
<span id="cb95-24"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-24" aria-hidden="true" tabindex="-1"></a>dt_final <span class="ot">=</span> <span class="fu">melt</span>(dt_final, <span class="at">id.vars =</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;width&quot;</span>))</span>
<span id="cb95-25"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-25" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dt_final, <span class="fu">c</span>(<span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;width&quot;</span>, <span class="st">&quot;sample_id&quot;</span>, <span class="st">&quot;total_cn&quot;</span>))</span>
<span id="cb95-26"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-27"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename chromosomes</span></span>
<span id="cb95-28"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-28" aria-hidden="true" tabindex="-1"></a>dt_final[chrom <span class="sc">==</span> <span class="st">&quot;X&quot;</span>, chrom <span class="sc">:</span><span class="er">=</span> <span class="dv">23</span>]</span>
<span id="cb95-29"><a href="clonal-analysis-pseudo-diploid-cells.html#cb95-29" aria-hidden="true" tabindex="-1"></a>dt_final[, chrom <span class="sc">:</span><span class="er">=</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, chrom)]</span></code></pre></div>
<p>After we prepared the input for MEDICC2 we can run it. This is down using the following command:</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb96-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="ex">medicc2</span> <span class="at">--plot</span> none <span class="at">-j</span> {nthreads} <span class="at">--total-copy-numbers</span> <span class="at">-a</span> <span class="st">&#39;total_cn&#39;</span> {input} {output.dir}</span></code></pre></div>
</div>
<div id="plot-medicc2-results" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Plot MEDICC2 results<a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After running MEDICC2 we can plot the tree that has been generated alongside the scCUTseq genomewide heatmaps. We load in the previously generated input data and the phylogenetic Newick tree from MEDICC2. First we do this for P3.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in data</span></span>
<span id="cb97-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-2" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/phylotrees/P3_input.tsv&quot;</span>)</span>
<span id="cb97-3"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-3" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">=</span> <span class="fu">read.tree</span>(<span class="st">&quot;./data/phylotrees/P3_500kb.new&quot;</span>)</span>
<span id="cb97-4"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-5"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder profiles</span></span>
<span id="cb97-6"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-6" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">dcast</span>(profiles, chrom <span class="sc">+</span> start <span class="sc">+</span> end <span class="sc">~</span> sample_id)</span>
<span id="cb97-7"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-7" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> dt[gtools<span class="sc">::</span><span class="fu">mixedorder</span>(chrom),]</span>
<span id="cb97-8"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-9"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename chromosomes</span></span>
<span id="cb97-10"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-10" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dt, <span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;chr&quot;</span>)</span>
<span id="cb97-11"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-11" aria-hidden="true" tabindex="-1"></a>dt[chr <span class="sc">==</span> <span class="st">&quot;chr23&quot;</span>, chr <span class="sc">:</span><span class="er">=</span> <span class="st">&quot;chrX&quot;</span>]</span>
<span id="cb97-12"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-13"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot tree</span></span>
<span id="cb97-14"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-14" aria-hidden="true" tabindex="-1"></a>tree_plot <span class="ot">=</span> <span class="fu">ggtree</span>(tree)</span>
<span id="cb97-15"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-16"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb97-17"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-17" aria-hidden="true" tabindex="-1"></a>col_order <span class="ot">=</span> <span class="fu">rev</span>(<span class="fu">get_taxa_name</span>(tree_plot))</span>
<span id="cb97-18"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-18" aria-hidden="true" tabindex="-1"></a>heatmap <span class="ot">=</span> <span class="fu">plotHeatmap</span>(dt[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(dt)], dt[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">dendrogram =</span> F, <span class="at">order =</span> col_order, <span class="at">linesize =</span> .<span class="dv">75</span>)</span>
<span id="cb97-19"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-20"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots</span></span>
<span id="cb97-21"><a href="clonal-analysis-pseudo-diploid-cells.html#cb97-21" aria-hidden="true" tabindex="-1"></a>tree_plot <span class="sc">+</span> heatmap <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20MEDICC2%20P3-1.png" width="2496" /></p>
<p>Repeat for P6.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in data</span></span>
<span id="cb98-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-2" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/phylotrees/P6_input.tsv&quot;</span>)</span>
<span id="cb98-3"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-3" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">=</span> <span class="fu">read.tree</span>(<span class="st">&quot;./data/phylotrees/P6_500kb.new&quot;</span>)</span>
<span id="cb98-4"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-5"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder profiles</span></span>
<span id="cb98-6"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-6" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">dcast</span>(profiles, chrom <span class="sc">+</span> start <span class="sc">+</span> end <span class="sc">~</span> sample_id)</span>
<span id="cb98-7"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-7" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> dt[gtools<span class="sc">::</span><span class="fu">mixedorder</span>(chrom),]</span>
<span id="cb98-8"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-9"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename chromosomes</span></span>
<span id="cb98-10"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-10" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dt, <span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;chr&quot;</span>)</span>
<span id="cb98-11"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-11" aria-hidden="true" tabindex="-1"></a>dt[chr <span class="sc">==</span> <span class="st">&quot;chr23&quot;</span>, chr <span class="sc">:</span><span class="er">=</span> <span class="st">&quot;chrX&quot;</span>]</span>
<span id="cb98-12"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-13"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot tree</span></span>
<span id="cb98-14"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-14" aria-hidden="true" tabindex="-1"></a>tree_plot <span class="ot">=</span> <span class="fu">ggtree</span>(tree)</span>
<span id="cb98-15"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-16"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb98-17"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-17" aria-hidden="true" tabindex="-1"></a>col_order <span class="ot">=</span> <span class="fu">rev</span>(<span class="fu">get_taxa_name</span>(tree_plot))</span>
<span id="cb98-18"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-18" aria-hidden="true" tabindex="-1"></a>heatmap <span class="ot">=</span> <span class="fu">plotHeatmap</span>(dt[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(dt)], dt[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">dendrogram =</span> F, <span class="at">order =</span> col_order, <span class="at">linesize =</span> .<span class="dv">75</span>)</span>
<span id="cb98-19"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-20"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots</span></span>
<span id="cb98-21"><a href="clonal-analysis-pseudo-diploid-cells.html#cb98-21" aria-hidden="true" tabindex="-1"></a>tree_plot <span class="sc">+</span> heatmap <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20MEDICC2%20P6-1.png" width="2496" /></p>
</div>
<div id="identify-tumour-subclones" class="section level2 hasAnchor" number="12.3">
<h2><span class="header-section-number">12.3</span> Identify tumour subclones<a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After we generated the phylogenetic tree using MEDICC2, we perform clustering upon this tree. To do this we run <a href="https://github.com/niemasd/TreeCluster">TreeCluster</a>. We run the following commands for P3 and P6, respectively.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb99-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="ex">TreeCluster.py</span> <span class="at">-i</span> {input_newick_tree_P3} <span class="at">-t</span> 3 <span class="at">-m</span> max <span class="at">-o</span> {output_clusters_P3} <span class="co"># These are the parameters for P3</span></span>
<span id="cb99-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="ex">TreeCluster.py</span> <span class="at">-i</span> {input_newick_tree_P6} <span class="at">-t</span> 4 <span class="at">-m</span> max <span class="at">-o</span> {output_clusters_P6} <span class="co"># These are the parameters for P6</span></span></code></pre></div>
<p>Next, we want to get the median copy number profile for each subclone. Sometimes it happens that there will be two subclones identified that will have the exact same median copy number profiles, if this is the case, we merge them together. Furthermore, we exclude any subclone that has fewer than 5 cells. The final list of the cell_ids and associated subclone and median copynumber profiles per subclone can be find in the <code>./data/subclones/</code> folder.</p>
</div>
<div id="calculate-scnas-per-subclones" class="section level2 hasAnchor" number="12.4">
<h2><span class="header-section-number">12.4</span> Calculate SCNAs per subclones<a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Calculate the percentage of the genome that is altered for P3.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-1" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P3_median_cn.tsv&quot;</span>)</span>
<span id="cb100-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-3"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get percentage of AMP/DEL/NEUTRAL</span></span>
<span id="cb100-4"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-4" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">dcast</span>(profiles, chr <span class="sc">+</span> start <span class="sc">+</span> end <span class="sc">~</span> cluster, <span class="at">value.var =</span> <span class="st">&quot;total_cn&quot;</span>)</span>
<span id="cb100-5"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-5" aria-hidden="true" tabindex="-1"></a>clones <span class="ot">=</span> <span class="fu">colnames</span>(profiles[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(profiles)])</span>
<span id="cb100-6"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-7"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">lapply</span>(clones, <span class="cf">function</span>(clone) {</span>
<span id="cb100-8"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Without X</span></span>
<span id="cb100-9"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-9" aria-hidden="true" tabindex="-1"></a>  amp_sum <span class="ot">=</span> <span class="fu">sum</span>(profiles[chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">&gt;</span> <span class="dv">2</span>)</span>
<span id="cb100-10"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-10" aria-hidden="true" tabindex="-1"></a>  del_sum <span class="ot">=</span> <span class="fu">sum</span>(profiles[chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">&lt;</span> <span class="dv">2</span>)</span>
<span id="cb100-11"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-11" aria-hidden="true" tabindex="-1"></a>  neut_sum <span class="ot">=</span> <span class="fu">sum</span>(profiles[chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb100-12"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb100-13"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># With X</span></span>
<span id="cb100-14"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-14" aria-hidden="true" tabindex="-1"></a>  amp_sum <span class="ot">=</span> amp_sum <span class="sc">+</span> <span class="fu">sum</span>(profiles[chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb100-15"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-15" aria-hidden="true" tabindex="-1"></a>  del_sum <span class="ot">=</span> del_sum <span class="sc">+</span> <span class="fu">sum</span>(profiles[chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">&lt;</span> <span class="dv">1</span>)</span>
<span id="cb100-16"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-16" aria-hidden="true" tabindex="-1"></a>  neut_sum <span class="ot">=</span> neut_sum <span class="sc">+</span> <span class="fu">sum</span>(profiles[chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb100-17"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb100-18"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.table</span>(<span class="at">clone =</span> clone,</span>
<span id="cb100-19"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">gain =</span> amp_sum <span class="sc">/</span> <span class="fu">nrow</span>(profiles),</span>
<span id="cb100-20"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">loss =</span> del_sum <span class="sc">/</span> <span class="fu">nrow</span>(profiles),</span>
<span id="cb100-21"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">neutral =</span> neut_sum <span class="sc">/</span> <span class="fu">nrow</span>(profiles)))</span>
<span id="cb100-22"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-22" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb100-23"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-24"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Rbind results</span></span>
<span id="cb100-25"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-25" aria-hidden="true" tabindex="-1"></a>results <span class="ot">=</span> <span class="fu">rbindlist</span>(res)</span>
<span id="cb100-26"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-26" aria-hidden="true" tabindex="-1"></a>results_m <span class="ot">=</span> <span class="fu">melt</span>(results, <span class="at">id.vars =</span> <span class="st">&quot;clone&quot;</span>)</span>
<span id="cb100-27"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-28"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove n = </span></span>
<span id="cb100-29"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-29" aria-hidden="true" tabindex="-1"></a>results_m[, clone <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>, <span class="st">&quot;&quot;</span>, clone)]</span>
<span id="cb100-30"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-31"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Set factor order</span></span>
<span id="cb100-32"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-32" aria-hidden="true" tabindex="-1"></a>results_m[, variable <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(variable, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;loss&quot;</span>))]</span>
<span id="cb100-33"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-33" aria-hidden="true" tabindex="-1"></a>results_m[, clone <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(clone, <span class="at">levels =</span> gtools<span class="sc">::</span><span class="fu">mixedsort</span>(<span class="fu">unique</span>(clone)))]</span>
<span id="cb100-34"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-35"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb100-36"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-36" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(results_m, <span class="fu">aes</span>(<span class="at">x =</span> clone, <span class="at">y =</span> value, <span class="at">fill =</span> variable)) <span class="sc">+</span></span>
<span id="cb100-37"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb100-38"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;neutral&quot;</span> <span class="ot">=</span> <span class="st">&quot;white&quot;</span>,</span>
<span id="cb100-39"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-39" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;gain&quot;</span> <span class="ot">=</span> <span class="st">&quot;firebrick3&quot;</span>,</span>
<span id="cb100-40"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-40" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;loss&quot;</span> <span class="ot">=</span> <span class="st">&quot;steelblue&quot;</span>),</span>
<span id="cb100-41"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-41" aria-hidden="true" tabindex="-1"></a>                    <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;loss&quot;</span>,</span>
<span id="cb100-42"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-42" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;neutral&quot;</span>,</span>
<span id="cb100-43"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-43" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;gain&quot;</span>),</span>
<span id="cb100-44"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-44" aria-hidden="true" tabindex="-1"></a>                    <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">&quot;loss&quot;</span>,</span>
<span id="cb100-45"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-45" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;neutral&quot;</span>,</span>
<span id="cb100-46"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-46" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;gain&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb100-47"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb100-48"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb100-49"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Percentage of genome&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb100-50"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb100-51"><a href="clonal-analysis-pseudo-diploid-cells.html#cb100-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/calculate%20SCNAs%20per%20subclone%20P3-1.png" width="672" /></p>
<p>Calculate the percentage of the genome that is altered for P6.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-1" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P6_median_cn.tsv&quot;</span>)</span>
<span id="cb101-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-3"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get percentage of AMP/DEL/NEUTRAL</span></span>
<span id="cb101-4"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-4" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">dcast</span>(profiles, chr <span class="sc">+</span> start <span class="sc">+</span> end <span class="sc">~</span> cluster, <span class="at">value.var =</span> <span class="st">&quot;total_cn&quot;</span>)</span>
<span id="cb101-5"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-5" aria-hidden="true" tabindex="-1"></a>clones <span class="ot">=</span> <span class="fu">colnames</span>(profiles[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(profiles)])</span>
<span id="cb101-6"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-7"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">lapply</span>(clones, <span class="cf">function</span>(clone) {</span>
<span id="cb101-8"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Without X</span></span>
<span id="cb101-9"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-9" aria-hidden="true" tabindex="-1"></a>  amp_sum <span class="ot">=</span> <span class="fu">sum</span>(profiles[chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">&gt;</span> <span class="dv">2</span>)</span>
<span id="cb101-10"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-10" aria-hidden="true" tabindex="-1"></a>  del_sum <span class="ot">=</span> <span class="fu">sum</span>(profiles[chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">&lt;</span> <span class="dv">2</span>)</span>
<span id="cb101-11"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-11" aria-hidden="true" tabindex="-1"></a>  neut_sum <span class="ot">=</span> <span class="fu">sum</span>(profiles[chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb101-12"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb101-13"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># With X</span></span>
<span id="cb101-14"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-14" aria-hidden="true" tabindex="-1"></a>  amp_sum <span class="ot">=</span> amp_sum <span class="sc">+</span> <span class="fu">sum</span>(profiles[chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb101-15"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-15" aria-hidden="true" tabindex="-1"></a>  del_sum <span class="ot">=</span> del_sum <span class="sc">+</span> <span class="fu">sum</span>(profiles[chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">&lt;</span> <span class="dv">1</span>)</span>
<span id="cb101-16"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-16" aria-hidden="true" tabindex="-1"></a>  neut_sum <span class="ot">=</span> neut_sum <span class="sc">+</span> <span class="fu">sum</span>(profiles[chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>, ..clone] <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb101-17"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb101-18"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.table</span>(<span class="at">clone =</span> clone,</span>
<span id="cb101-19"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">gain =</span> amp_sum <span class="sc">/</span> <span class="fu">nrow</span>(profiles),</span>
<span id="cb101-20"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">loss =</span> del_sum <span class="sc">/</span> <span class="fu">nrow</span>(profiles),</span>
<span id="cb101-21"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">neutral =</span> neut_sum <span class="sc">/</span> <span class="fu">nrow</span>(profiles)))</span>
<span id="cb101-22"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-22" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb101-23"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-24"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Rbind results</span></span>
<span id="cb101-25"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-25" aria-hidden="true" tabindex="-1"></a>results <span class="ot">=</span> <span class="fu">rbindlist</span>(res)</span>
<span id="cb101-26"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-26" aria-hidden="true" tabindex="-1"></a>results_m <span class="ot">=</span> <span class="fu">melt</span>(results, <span class="at">id.vars =</span> <span class="st">&quot;clone&quot;</span>)</span>
<span id="cb101-27"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-28"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove n = </span></span>
<span id="cb101-29"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-29" aria-hidden="true" tabindex="-1"></a>results_m[, clone <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>, <span class="st">&quot;&quot;</span>, clone)]</span>
<span id="cb101-30"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-31"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Set factor order</span></span>
<span id="cb101-32"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-32" aria-hidden="true" tabindex="-1"></a>results_m[, variable <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(variable, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;loss&quot;</span>))]</span>
<span id="cb101-33"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-33" aria-hidden="true" tabindex="-1"></a>results_m[, clone <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(clone, <span class="at">levels =</span> gtools<span class="sc">::</span><span class="fu">mixedsort</span>(<span class="fu">unique</span>(clone)))]</span>
<span id="cb101-34"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-35"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb101-36"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-36" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(results_m, <span class="fu">aes</span>(<span class="at">x =</span> clone, <span class="at">y =</span> value, <span class="at">fill =</span> variable)) <span class="sc">+</span></span>
<span id="cb101-37"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb101-38"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;neutral&quot;</span> <span class="ot">=</span> <span class="st">&quot;white&quot;</span>,</span>
<span id="cb101-39"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-39" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;gain&quot;</span> <span class="ot">=</span> <span class="st">&quot;firebrick3&quot;</span>,</span>
<span id="cb101-40"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-40" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;loss&quot;</span> <span class="ot">=</span> <span class="st">&quot;steelblue&quot;</span>),</span>
<span id="cb101-41"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-41" aria-hidden="true" tabindex="-1"></a>                    <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;loss&quot;</span>,</span>
<span id="cb101-42"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-42" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;neutral&quot;</span>,</span>
<span id="cb101-43"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-43" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;gain&quot;</span>),</span>
<span id="cb101-44"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-44" aria-hidden="true" tabindex="-1"></a>                    <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">&quot;loss&quot;</span>,</span>
<span id="cb101-45"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-45" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;neutral&quot;</span>,</span>
<span id="cb101-46"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-46" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;gain&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb101-47"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb101-48"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb101-49"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Percentage of genome&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb101-50"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb101-51"><a href="clonal-analysis-pseudo-diploid-cells.html#cb101-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/calculate%20SCNAs%20per%20subclone%20P6-1.png" width="672" /></p>
</div>
<div id="calculate-entropy-per-subclone" class="section level2 hasAnchor" number="12.5">
<h2><span class="header-section-number">12.5</span> Calculate entropy per subclone<a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To assess whether subclones are highly localized in the prostate or widespread we calculate the entropy per subclone. We first normalize the number of cells from each subclone in each section based on the total number of cells that pass QC in that section. Then we calculate the 2D Entropy.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in annotation and clusters and cn profiles</span></span>
<span id="cb102-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-2" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>,</span>
<span id="cb102-3"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">header =</span> F, <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;tissue_type&quot;</span>))</span>
<span id="cb102-4"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-4" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P3_clones.tsv&quot;</span>)</span>
<span id="cb102-5"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-6"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get library info and merge</span></span>
<span id="cb102-7"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-7" aria-hidden="true" tabindex="-1"></a>clusters[, library <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample_id)]</span>
<span id="cb102-8"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-8" aria-hidden="true" tabindex="-1"></a>total <span class="ot">=</span> <span class="fu">merge</span>(clusters, annot, <span class="at">all.x =</span> T)</span>
<span id="cb102-9"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-10"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get counts per region and total per region</span></span>
<span id="cb102-11"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-11" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> total[, .N, by <span class="ot">=</span> .(section, cluster)]</span>
<span id="cb102-12"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-12" aria-hidden="true" tabindex="-1"></a>counts_total <span class="ot">=</span> total[, .(<span class="at">total =</span> .N), by <span class="ot">=</span> cluster]</span>
<span id="cb102-13"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-14"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge and get fraction of total</span></span>
<span id="cb102-15"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-15" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">merge</span>(counts, counts_total)</span>
<span id="cb102-16"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-16" aria-hidden="true" tabindex="-1"></a>counts[, fraction <span class="sc">:</span><span class="er">=</span> N <span class="sc">/</span> total]</span>
<span id="cb102-17"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-17" aria-hidden="true" tabindex="-1"></a>counts[, cluster <span class="sc">:</span><span class="er">=</span> <span class="fu">as.character</span>(cluster)]</span>
<span id="cb102-18"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-19"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Add coordinates</span></span>
<span id="cb102-20"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-20" aria-hidden="true" tabindex="-1"></a>counts[, x <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb102-21"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-21" aria-hidden="true" tabindex="-1"></a>counts[, y <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb102-22"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-22" aria-hidden="true" tabindex="-1"></a>counts[, name <span class="sc">:</span><span class="er">=</span> <span class="fu">paste0</span>(cluster, <span class="st">&quot; (n = &quot;</span>, total, <span class="st">&quot;)&quot;</span>)]</span>
<span id="cb102-23"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-24"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize for total per section</span></span>
<span id="cb102-25"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-25" aria-hidden="true" tabindex="-1"></a>counts_overall <span class="ot">=</span> counts[, .(<span class="at">total_cells =</span> <span class="fu">sum</span>(N)), by <span class="ot">=</span> .(section, x, y)]</span>
<span id="cb102-26"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-26" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">merge</span>(counts, counts_overall, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;section&quot;</span>, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;y&quot;</span>))</span>
<span id="cb102-27"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-27" aria-hidden="true" tabindex="-1"></a>counts[, normalized_count <span class="sc">:</span><span class="er">=</span> N <span class="sc">/</span> total_cells]</span>
<span id="cb102-28"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-29"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-29" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">unique</span>(counts<span class="sc">$</span>name), <span class="cf">function</span>(clone) {</span>
<span id="cb102-30"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Complete</span></span>
<span id="cb102-31"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-31" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">=</span> counts[name <span class="sc">==</span> clone]</span>
<span id="cb102-32"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-32" aria-hidden="true" tabindex="-1"></a>  fill_dt <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">cluster =</span> <span class="fu">gsub</span>(<span class="st">&quot;c| .*&quot;</span>, <span class="st">&quot;&quot;</span>, clone), </span>
<span id="cb102-33"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-33" aria-hidden="true" tabindex="-1"></a>                       <span class="at">section =</span> annot[<span class="sc">!</span>section <span class="sc">%in%</span> dt<span class="sc">$</span>section, section],</span>
<span id="cb102-34"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-34" aria-hidden="true" tabindex="-1"></a>                       <span class="at">N =</span> <span class="cn">NA</span>,</span>
<span id="cb102-35"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-35" aria-hidden="true" tabindex="-1"></a>                       <span class="at">fraction =</span> <span class="cn">NA</span>,</span>
<span id="cb102-36"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-36" aria-hidden="true" tabindex="-1"></a>                       <span class="at">total =</span> <span class="fu">unique</span>(dt<span class="sc">$</span>total),</span>
<span id="cb102-37"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-37" aria-hidden="true" tabindex="-1"></a>                       <span class="at">normalized_count =</span> <span class="cn">NA</span>,</span>
<span id="cb102-38"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-38" aria-hidden="true" tabindex="-1"></a>                       <span class="at">total_cells =</span> <span class="cn">NA</span>,</span>
<span id="cb102-39"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-39" aria-hidden="true" tabindex="-1"></a>                       <span class="at">name =</span> clone)</span>
<span id="cb102-40"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-40" aria-hidden="true" tabindex="-1"></a>  fill_dt[, x <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb102-41"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-41" aria-hidden="true" tabindex="-1"></a>  fill_dt[, y <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb102-42"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-42" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">=</span> <span class="fu">rbind</span>(dt, fill_dt)</span>
<span id="cb102-43"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb102-44"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-44" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">=</span> <span class="fu">as.matrix</span>(<span class="fu">dcast</span>(dt, y<span class="sc">~</span>x, <span class="at">value.var =</span> <span class="st">&quot;normalized_count&quot;</span>))</span>
<span id="cb102-45"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-45" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">=</span> mat[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(mat)]</span>
<span id="cb102-46"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-46" aria-hidden="true" tabindex="-1"></a>  mat[<span class="fu">is.na</span>(mat)] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb102-47"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.table</span>(<span class="at">name =</span> clone, <span class="at">entropy =</span> <span class="fu">Entropy</span>(mat)))</span>
<span id="cb102-48"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-48" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb102-49"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-50"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Rbind results</span></span>
<span id="cb102-51"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-51" aria-hidden="true" tabindex="-1"></a>result <span class="ot">=</span> <span class="fu">rbindlist</span>(res)</span>
<span id="cb102-52"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-52" aria-hidden="true" tabindex="-1"></a>result[, clone <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>, <span class="st">&quot;&quot;</span>, name)]</span>
<span id="cb102-53"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-53" aria-hidden="true" tabindex="-1"></a><span class="fu">setorder</span>(result, entropy)</span>
<span id="cb102-54"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-55"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot P3</span></span>
<span id="cb102-56"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-56" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(result, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> entropy, <span class="at">label =</span> clone)) <span class="sc">+</span></span>
<span id="cb102-57"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb102-58"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Entropy&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb102-59"><a href="clonal-analysis-pseudo-diploid-cells.html#cb102-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> result[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">size =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/calculate%20entropy%20P3-1.png" width="672" /></p>
<p>And, once again, repeat this for P6.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in annotation and clusters and cn profiles</span></span>
<span id="cb103-2"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-2" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P6.tsv&quot;</span>,</span>
<span id="cb103-3"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">header =</span> F, <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;tissue_type&quot;</span>))</span>
<span id="cb103-4"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-4" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P6_clones.tsv&quot;</span>)</span>
<span id="cb103-5"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-6"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get library info and merge</span></span>
<span id="cb103-7"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-7" aria-hidden="true" tabindex="-1"></a>clusters[, library <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample_id)]</span>
<span id="cb103-8"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-8" aria-hidden="true" tabindex="-1"></a>total <span class="ot">=</span> <span class="fu">merge</span>(clusters, annot, <span class="at">all.x =</span> T)</span>
<span id="cb103-9"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-10"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get counts per region and total per region</span></span>
<span id="cb103-11"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-11" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> total[, .N, by <span class="ot">=</span> .(section, cluster)]</span>
<span id="cb103-12"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-12" aria-hidden="true" tabindex="-1"></a>counts_total <span class="ot">=</span> total[, .(<span class="at">total =</span> .N), by <span class="ot">=</span> cluster]</span>
<span id="cb103-13"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-14"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge and get fraction of total</span></span>
<span id="cb103-15"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-15" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">merge</span>(counts, counts_total)</span>
<span id="cb103-16"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-16" aria-hidden="true" tabindex="-1"></a>counts[, fraction <span class="sc">:</span><span class="er">=</span> N <span class="sc">/</span> total]</span>
<span id="cb103-17"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-17" aria-hidden="true" tabindex="-1"></a>counts[, cluster <span class="sc">:</span><span class="er">=</span> <span class="fu">as.character</span>(cluster)]</span>
<span id="cb103-18"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-19"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Add coordinates</span></span>
<span id="cb103-20"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-20" aria-hidden="true" tabindex="-1"></a>counts[, x <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb103-21"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-21" aria-hidden="true" tabindex="-1"></a>counts[, y <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb103-22"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-22" aria-hidden="true" tabindex="-1"></a>counts[, name <span class="sc">:</span><span class="er">=</span> <span class="fu">paste0</span>(cluster, <span class="st">&quot; (n = &quot;</span>, total, <span class="st">&quot;)&quot;</span>)]</span>
<span id="cb103-23"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-24"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize for total per section</span></span>
<span id="cb103-25"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-25" aria-hidden="true" tabindex="-1"></a>counts_overall <span class="ot">=</span> counts[, .(<span class="at">total_cells =</span> <span class="fu">sum</span>(N)), by <span class="ot">=</span> .(section, x, y)]</span>
<span id="cb103-26"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-26" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">merge</span>(counts, counts_overall, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;section&quot;</span>, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;y&quot;</span>))</span>
<span id="cb103-27"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-27" aria-hidden="true" tabindex="-1"></a>counts[, normalized_count <span class="sc">:</span><span class="er">=</span> N <span class="sc">/</span> total_cells]</span>
<span id="cb103-28"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-29"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-29" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">unique</span>(counts<span class="sc">$</span>name), <span class="cf">function</span>(clone) {</span>
<span id="cb103-30"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Complete</span></span>
<span id="cb103-31"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-31" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">=</span> counts[name <span class="sc">==</span> clone]</span>
<span id="cb103-32"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-32" aria-hidden="true" tabindex="-1"></a>  fill_dt <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">cluster =</span> <span class="fu">gsub</span>(<span class="st">&quot;c| .*&quot;</span>, <span class="st">&quot;&quot;</span>, clone), </span>
<span id="cb103-33"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-33" aria-hidden="true" tabindex="-1"></a>                       <span class="at">section =</span> annot[<span class="sc">!</span>section <span class="sc">%in%</span> dt<span class="sc">$</span>section, section],</span>
<span id="cb103-34"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-34" aria-hidden="true" tabindex="-1"></a>                       <span class="at">N =</span> <span class="cn">NA</span>,</span>
<span id="cb103-35"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-35" aria-hidden="true" tabindex="-1"></a>                       <span class="at">fraction =</span> <span class="cn">NA</span>,</span>
<span id="cb103-36"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-36" aria-hidden="true" tabindex="-1"></a>                       <span class="at">total =</span> <span class="fu">unique</span>(dt<span class="sc">$</span>total),</span>
<span id="cb103-37"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-37" aria-hidden="true" tabindex="-1"></a>                       <span class="at">normalized_count =</span> <span class="cn">NA</span>,</span>
<span id="cb103-38"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-38" aria-hidden="true" tabindex="-1"></a>                       <span class="at">total_cells =</span> <span class="cn">NA</span>,</span>
<span id="cb103-39"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-39" aria-hidden="true" tabindex="-1"></a>                       <span class="at">name =</span> clone)</span>
<span id="cb103-40"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-40" aria-hidden="true" tabindex="-1"></a>  fill_dt[, x <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb103-41"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-41" aria-hidden="true" tabindex="-1"></a>  fill_dt[, y <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb103-42"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-42" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">=</span> <span class="fu">rbind</span>(dt, fill_dt)</span>
<span id="cb103-43"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb103-44"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-44" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">=</span> <span class="fu">as.matrix</span>(<span class="fu">dcast</span>(dt, y<span class="sc">~</span>x, <span class="at">value.var =</span> <span class="st">&quot;normalized_count&quot;</span>))</span>
<span id="cb103-45"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-45" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">=</span> mat[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(mat)]</span>
<span id="cb103-46"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-46" aria-hidden="true" tabindex="-1"></a>  mat[<span class="fu">is.na</span>(mat)] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb103-47"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.table</span>(<span class="at">name =</span> clone, <span class="at">entropy =</span> <span class="fu">Entropy</span>(mat)))</span>
<span id="cb103-48"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-48" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb103-49"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-50"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Rbind results</span></span>
<span id="cb103-51"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-51" aria-hidden="true" tabindex="-1"></a>result <span class="ot">=</span> <span class="fu">rbindlist</span>(res)</span>
<span id="cb103-52"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-52" aria-hidden="true" tabindex="-1"></a>result[, clone <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>, <span class="st">&quot;&quot;</span>, name)]</span>
<span id="cb103-53"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-53" aria-hidden="true" tabindex="-1"></a><span class="fu">setorder</span>(result, entropy)</span>
<span id="cb103-54"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-55"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot P6</span></span>
<span id="cb103-56"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-56" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(result, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> entropy, <span class="at">label =</span> clone)) <span class="sc">+</span></span>
<span id="cb103-57"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb103-58"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Entropy&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb103-59"><a href="clonal-analysis-pseudo-diploid-cells.html#cb103-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> result[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">size =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/calculate%20entropy%20P6-1.png" width="672" /></p>
<p>Finally, the heatmaps of the distribution of each subclone (Figure 2k, l), can be found in the Supplementary Figure 12-15 section.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="whole-genome-sequencing-of-blood.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phylogenetic-analysis-of-breast-cancer.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
