<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Comparison of scCUTseq and ACT | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</title>
  <meta name="description" content="5 Comparison of scCUTseq and ACT | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Comparison of scCUTseq and ACT | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Comparison of scCUTseq and ACT | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="technical-validation-of-sccutseq.html"/>
<link rel="next" href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>4</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="4.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>4.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="4.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>4.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="4.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>4.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="4.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>4.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="4.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>4.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="4.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>4.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="4.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>4.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="4.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>4.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>5</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="5.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>5.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="5.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>5.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="5.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>5.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="5.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>5.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>6</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="6.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>6.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="6.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>6.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="6.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>6.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>7</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="7.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>7.1</b> Loading in data</a></li>
<li class="chapter" data-level="7.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>7.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="7.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>7.3</b> Training the model</a></li>
<li class="chapter" data-level="7.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>7.4</b> Validating the model</a></li>
<li class="chapter" data-level="7.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>7.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>8</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="8.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>8.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>9</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="9.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>9.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="9.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>9.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>10</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>10.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>11</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="11.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>11.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>12</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="12.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>12.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="12.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>12.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="12.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>12.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="12.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>12.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="12.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>12.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>13</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="13.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>13.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="13.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>13.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>14</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="14.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>14.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>16</b> Monster cells</a>
<ul>
<li class="chapter" data-level="16.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>16.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="16.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>16.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="16.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>16.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="validation-of-pseudodiploid-deletions.html"><a href="validation-of-pseudodiploid-deletions.html"><i class="fa fa-check"></i><b>17</b> Validation of pseudodiploid deletions</a></li>
<li class="chapter" data-level="18" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>18</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="18.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>18.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="18.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>18.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="18.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>18.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="18.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>18.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html"><i class="fa fa-check"></i><b>19</b> COSMIC deletions and targeted panel mutations</a>
<ul>
<li class="chapter" data-level="19.1" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>19.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="19.2" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>19.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>20</b> Session info</a>
<ul>
<li class="chapter" data-level="20.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>20.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="comparison-of-sccutseq-and-act" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Comparison of scCUTseq and ACT<a href="comparison-of-sccutseq-and-act.html#comparison-of-sccutseq-and-act" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This sections produces all the figures used in Supplementary Figure 3.<br />
We compare the Breadth of Coverage, Overdispersion and overall copynumber profiles between different sections in Patient 3 that were sequenced by both ACT and scCUTseq.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="comparison-of-sccutseq-and-act.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb28-2"><a href="comparison-of-sccutseq-and-act.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span>
<span id="cb28-3"><a href="comparison-of-sccutseq-and-act.html#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="comparison-of-sccutseq-and-act.html#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Source plotting functions</span></span>
<span id="cb28-5"><a href="comparison-of-sccutseq-and-act.html#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotProfile.R&quot;</span>)</span>
<span id="cb28-6"><a href="comparison-of-sccutseq-and-act.html#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotHeatmap.R&quot;</span>)</span></code></pre></div>
<div id="calculate-breadth-of-coverage" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Calculate breadth of coverage<a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We downsampled bam files to 800K reads using the following commands in bash and ran our copynumber calling pipeline (described in <code>Preprocessing.Rmd</code>) and use the resulting <code>cnv.rds</code> to get information about cell quality.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="comparison-of-sccutseq-and-act.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="va">${INSAMPLEBASE}</span>.bam <span class="at">-H</span> <span class="op">&gt;</span> <span class="va">${INSAMPLEBASE}</span>.sam<span class="kw">&amp;&amp;</span> <span class="ex">samtools</span> view <span class="va">$INSAMPLEBASE</span>.bam <span class="kw">|</span> <span class="fu">shuf</span> <span class="at">-n</span> 800000 <span class="at">--random-source</span><span class="op">=</span><span class="va">$INSAMPLEBASE</span>.bam <span class="op">&gt;&gt;</span> <span class="va">$INSAMPLEBASE</span>.sam<span class="kw">&amp;&amp;</span> <span class="ex">samtools</span> sort <span class="at">-o</span> ../downsampling/bamfiles/<span class="va">$OUTBAM</span> <span class="va">$INSAMPLEBASE</span>.sam<span class="kw">&amp;&amp;</span> <span class="fu">rm</span> <span class="va">$INSAMPLEBASE</span>.sam<span class="kw">&amp;&amp;</span> <span class="ex">genomeCoverageBed</span> <span class="at">-ibam</span> ../downsampling/bamfiles/<span class="va">$OUTBAM</span> <span class="at">-fs</span> 50 <span class="at">-max</span> 1 <span class="op">&gt;</span> ../downsampling/coverage/<span class="va">$COVHISTFILE</span></span></code></pre></div>
<p>We then used the <code>$COVHISTFILE</code> output to calculate the breadth of coverage.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="comparison-of-sccutseq-and-act.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define BoC function</span></span>
<span id="cb30-2"><a href="comparison-of-sccutseq-and-act.html#cb30-2" aria-hidden="true" tabindex="-1"></a>calc_coverage <span class="ot">=</span> <span class="cf">function</span>(path) {</span>
<span id="cb30-3"><a href="comparison-of-sccutseq-and-act.html#cb30-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-4"><a href="comparison-of-sccutseq-and-act.html#cb30-4" aria-hidden="true" tabindex="-1"></a>  inpaths <span class="ot">=</span> <span class="fu">Sys.glob</span>(<span class="fu">paste0</span>(path, <span class="st">&quot;*.covhist.txt&quot;</span>))</span>
<span id="cb30-5"><a href="comparison-of-sccutseq-and-act.html#cb30-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-6"><a href="comparison-of-sccutseq-and-act.html#cb30-6" aria-hidden="true" tabindex="-1"></a>  coverage.stats <span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">bed_path=</span>inpaths) <span class="sc">%&gt;%</span></span>
<span id="cb30-7"><a href="comparison-of-sccutseq-and-act.html#cb30-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">cellname =</span> <span class="fu">str_extract</span>(<span class="fu">basename</span>(bed_path), <span class="st">&quot;^[^.]*&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-8"><a href="comparison-of-sccutseq-and-act.html#cb30-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cellname) <span class="sc">%&gt;%</span></span>
<span id="cb30-9"><a href="comparison-of-sccutseq-and-act.html#cb30-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">.groups=</span><span class="st">&quot;keep&quot;</span>,</span>
<span id="cb30-10"><a href="comparison-of-sccutseq-and-act.html#cb30-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">read_tsv</span>(bed_path,</span>
<span id="cb30-11"><a href="comparison-of-sccutseq-and-act.html#cb30-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col_names=</span><span class="fu">c</span>(<span class="st">&quot;refname&quot;</span>, <span class="st">&quot;depth&quot;</span>, <span class="st">&quot;count&quot;</span>,</span>
<span id="cb30-12"><a href="comparison-of-sccutseq-and-act.html#cb30-12" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">&quot;refsize&quot;</span>, <span class="st">&quot;frac&quot;</span>),</span>
<span id="cb30-13"><a href="comparison-of-sccutseq-and-act.html#cb30-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col_types=</span><span class="fu">cols</span>(<span class="fu">col_character</span>(), <span class="fu">col_double</span>(),</span>
<span id="cb30-14"><a href="comparison-of-sccutseq-and-act.html#cb30-14" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">col_double</span>(), <span class="fu">col_double</span>(),</span>
<span id="cb30-15"><a href="comparison-of-sccutseq-and-act.html#cb30-15" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">col_double</span>())),</span>
<span id="cb30-16"><a href="comparison-of-sccutseq-and-act.html#cb30-16" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb30-17"><a href="comparison-of-sccutseq-and-act.html#cb30-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(refname<span class="sc">==</span><span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-18"><a href="comparison-of-sccutseq-and-act.html#cb30-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">breadth =</span> <span class="dv">1</span> <span class="sc">-</span> frac[depth<span class="sc">==</span><span class="dv">0</span>],</span>
<span id="cb30-19"><a href="comparison-of-sccutseq-and-act.html#cb30-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">.groups=</span><span class="st">&quot;keep&quot;</span>)</span>
<span id="cb30-20"><a href="comparison-of-sccutseq-and-act.html#cb30-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-21"><a href="comparison-of-sccutseq-and-act.html#cb30-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-22"><a href="comparison-of-sccutseq-and-act.html#cb30-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-23"><a href="comparison-of-sccutseq-and-act.html#cb30-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate BoC on the COVHIST files</span></span>
<span id="cb30-24"><a href="comparison-of-sccutseq-and-act.html#cb30-24" aria-hidden="true" tabindex="-1"></a>sccutseq <span class="ot">=</span> <span class="fu">calc_coverage</span>(<span class="st">&quot;./data/downsampling/coverage/scCUTseq/&quot;</span>)</span>
<span id="cb30-25"><a href="comparison-of-sccutseq-and-act.html#cb30-25" aria-hidden="true" tabindex="-1"></a>act <span class="ot">=</span> <span class="fu">calc_coverage</span>(<span class="st">&quot;./data/downsampling/coverage/ACT/&quot;</span>)</span>
<span id="cb30-26"><a href="comparison-of-sccutseq-and-act.html#cb30-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-27"><a href="comparison-of-sccutseq-and-act.html#cb30-27" aria-hidden="true" tabindex="-1"></a><span class="co"># SetDT</span></span>
<span id="cb30-28"><a href="comparison-of-sccutseq-and-act.html#cb30-28" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(sccutseq)</span>
<span id="cb30-29"><a href="comparison-of-sccutseq-and-act.html#cb30-29" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(act)</span>
<span id="cb30-30"><a href="comparison-of-sccutseq-and-act.html#cb30-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-31"><a href="comparison-of-sccutseq-and-act.html#cb30-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Load copynumber profiles to get HQ cells</span></span>
<span id="cb30-32"><a href="comparison-of-sccutseq-and-act.html#cb30-32" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/downsampling/P3.rds&quot;</span>)</span>
<span id="cb30-33"><a href="comparison-of-sccutseq-and-act.html#cb30-33" aria-hidden="true" tabindex="-1"></a>p3_stats <span class="ot">=</span> p3<span class="sc">$</span>stats</span>
<span id="cb30-34"><a href="comparison-of-sccutseq-and-act.html#cb30-34" aria-hidden="true" tabindex="-1"></a>p3_hq <span class="ot">=</span> p3_stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample]</span>
<span id="cb30-35"><a href="comparison-of-sccutseq-and-act.html#cb30-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-36"><a href="comparison-of-sccutseq-and-act.html#cb30-36" aria-hidden="true" tabindex="-1"></a><span class="co"># ACT</span></span>
<span id="cb30-37"><a href="comparison-of-sccutseq-and-act.html#cb30-37" aria-hidden="true" tabindex="-1"></a>CD5p <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/downsampling/CD5p.rds&quot;</span>)</span>
<span id="cb30-38"><a href="comparison-of-sccutseq-and-act.html#cb30-38" aria-hidden="true" tabindex="-1"></a>CD7p <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/downsampling/CD7p.rds&quot;</span>)</span>
<span id="cb30-39"><a href="comparison-of-sccutseq-and-act.html#cb30-39" aria-hidden="true" tabindex="-1"></a>CD4p <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/downsampling/CD4p.rds&quot;</span>)</span>
<span id="cb30-40"><a href="comparison-of-sccutseq-and-act.html#cb30-40" aria-hidden="true" tabindex="-1"></a>CD2p <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/downsampling/CD2p.rds&quot;</span>)</span>
<span id="cb30-41"><a href="comparison-of-sccutseq-and-act.html#cb30-41" aria-hidden="true" tabindex="-1"></a>CD6p <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/downsampling/CD6p.rds&quot;</span>)</span>
<span id="cb30-42"><a href="comparison-of-sccutseq-and-act.html#cb30-42" aria-hidden="true" tabindex="-1"></a>CD3p <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/downsampling/CD3p.rds&quot;</span>)</span>
<span id="cb30-43"><a href="comparison-of-sccutseq-and-act.html#cb30-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-44"><a href="comparison-of-sccutseq-and-act.html#cb30-44" aria-hidden="true" tabindex="-1"></a>act_hq <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;CD5p_&quot;</span>, CD5p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample]),</span>
<span id="cb30-45"><a href="comparison-of-sccutseq-and-act.html#cb30-45" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste0</span>(<span class="st">&quot;CD7p_&quot;</span>, CD7p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample]),</span>
<span id="cb30-46"><a href="comparison-of-sccutseq-and-act.html#cb30-46" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste0</span>(<span class="st">&quot;CD4p_&quot;</span>, CD4p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample]),</span>
<span id="cb30-47"><a href="comparison-of-sccutseq-and-act.html#cb30-47" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste0</span>(<span class="st">&quot;CD2p_&quot;</span>, CD2p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample]),</span>
<span id="cb30-48"><a href="comparison-of-sccutseq-and-act.html#cb30-48" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste0</span>(<span class="st">&quot;CD6p_&quot;</span>, CD6p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample]),</span>
<span id="cb30-49"><a href="comparison-of-sccutseq-and-act.html#cb30-49" aria-hidden="true" tabindex="-1"></a>           <span class="fu">paste0</span>(<span class="st">&quot;CD3p_&quot;</span>, CD3p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample]))</span>
<span id="cb30-50"><a href="comparison-of-sccutseq-and-act.html#cb30-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-51"><a href="comparison-of-sccutseq-and-act.html#cb30-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Select HQ samples</span></span>
<span id="cb30-52"><a href="comparison-of-sccutseq-and-act.html#cb30-52" aria-hidden="true" tabindex="-1"></a>sccutseq <span class="ot">=</span> sccutseq[cellname <span class="sc">%in%</span> p3_hq, ]</span>
<span id="cb30-53"><a href="comparison-of-sccutseq-and-act.html#cb30-53" aria-hidden="true" tabindex="-1"></a>act <span class="ot">=</span> act[cellname <span class="sc">%in%</span> act_hq, ]</span>
<span id="cb30-54"><a href="comparison-of-sccutseq-and-act.html#cb30-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-55"><a href="comparison-of-sccutseq-and-act.html#cb30-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine</span></span>
<span id="cb30-56"><a href="comparison-of-sccutseq-and-act.html#cb30-56" aria-hidden="true" tabindex="-1"></a>sccutseq[, tech <span class="sc">:</span><span class="er">=</span> <span class="st">&quot;scCUTseq&quot;</span>]</span>
<span id="cb30-57"><a href="comparison-of-sccutseq-and-act.html#cb30-57" aria-hidden="true" tabindex="-1"></a>act[, tech <span class="sc">:</span><span class="er">=</span> <span class="st">&quot;ACT&quot;</span>]</span>
<span id="cb30-58"><a href="comparison-of-sccutseq-and-act.html#cb30-58" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">rbind</span>(sccutseq, act)</span>
<span id="cb30-59"><a href="comparison-of-sccutseq-and-act.html#cb30-59" aria-hidden="true" tabindex="-1"></a>dt[, tech <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(tech, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;scCUTseq&quot;</span>, <span class="st">&quot;ACT&quot;</span>))]</span>
<span id="cb30-60"><a href="comparison-of-sccutseq-and-act.html#cb30-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-61"><a href="comparison-of-sccutseq-and-act.html#cb30-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot comparison</span></span>
<span id="cb30-62"><a href="comparison-of-sccutseq-and-act.html#cb30-62" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(dt, <span class="fu">aes</span>(<span class="at">x =</span> tech, <span class="at">y =</span> breadth)) <span class="sc">+</span></span>
<span id="cb30-63"><a href="comparison-of-sccutseq-and-act.html#cb30-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb30-64"><a href="comparison-of-sccutseq-and-act.html#cb30-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> .<span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb30-65"><a href="comparison-of-sccutseq-and-act.html#cb30-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.0125</span>)) <span class="sc">+</span></span>
<span id="cb30-66"><a href="comparison-of-sccutseq-and-act.html#cb30-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Breadth of coverage&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/calculate%20breadth%20of%20coverage-1.png" width="672" /></p>
</div>
<div id="calculate-overdispersion" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Calculate overdispersion<a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Then we used the read counts from each bin to calculate the overdispersion. We also do this on the HQ cells that are selected in the code chunk above.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="comparison-of-sccutseq-and-act.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define functions</span></span>
<span id="cb31-2"><a href="comparison-of-sccutseq-and-act.html#cb31-2" aria-hidden="true" tabindex="-1"></a>l2e.normal.sd <span class="ot">=</span> <span class="cf">function</span>(xs)</span>
<span id="cb31-3"><a href="comparison-of-sccutseq-and-act.html#cb31-3" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb31-4"><a href="comparison-of-sccutseq-and-act.html#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Need at least two values to get a standard deviation</span></span>
<span id="cb31-5"><a href="comparison-of-sccutseq-and-act.html#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(xs) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb31-6"><a href="comparison-of-sccutseq-and-act.html#cb31-6" aria-hidden="true" tabindex="-1"></a>  optim.result <span class="ot">=</span> stats<span class="sc">::</span><span class="fu">optimize</span>(</span>
<span id="cb31-7"><a href="comparison-of-sccutseq-and-act.html#cb31-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># L2E loss function</span></span>
<span id="cb31-8"><a href="comparison-of-sccutseq-and-act.html#cb31-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">f=</span><span class="cf">function</span>(sd)</span>
<span id="cb31-9"><a href="comparison-of-sccutseq-and-act.html#cb31-9" aria-hidden="true" tabindex="-1"></a>      <span class="co"># &quot;Data part&quot;, the sample average of the likelihood</span></span>
<span id="cb31-10"><a href="comparison-of-sccutseq-and-act.html#cb31-10" aria-hidden="true" tabindex="-1"></a>      <span class="sc">-</span><span class="dv">2</span> <span class="sc">*</span> <span class="fu">mean</span>(stats<span class="sc">::</span><span class="fu">dnorm</span>(xs, <span class="at">sd=</span>sd)) <span class="sc">+</span></span>
<span id="cb31-11"><a href="comparison-of-sccutseq-and-act.html#cb31-11" aria-hidden="true" tabindex="-1"></a>      <span class="co"># &quot;Theta part&quot;, the integral of the squared density</span></span>
<span id="cb31-12"><a href="comparison-of-sccutseq-and-act.html#cb31-12" aria-hidden="true" tabindex="-1"></a>      <span class="dv">1</span><span class="sc">/</span>(<span class="dv">2</span><span class="sc">*</span><span class="fu">sqrt</span>(pi)<span class="sc">*</span>sd),</span>
<span id="cb31-13"><a href="comparison-of-sccutseq-and-act.html#cb31-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Parameter: standard deviation of the normal distribution fit</span></span>
<span id="cb31-14"><a href="comparison-of-sccutseq-and-act.html#cb31-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">interval =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">diff</span>(<span class="fu">range</span>(xs))))</span>
<span id="cb31-15"><a href="comparison-of-sccutseq-and-act.html#cb31-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(optim.result<span class="sc">$</span>minimum)</span>
<span id="cb31-16"><a href="comparison-of-sccutseq-and-act.html#cb31-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-17"><a href="comparison-of-sccutseq-and-act.html#cb31-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-18"><a href="comparison-of-sccutseq-and-act.html#cb31-18" aria-hidden="true" tabindex="-1"></a><span class="co"># A function for estimating the index of dispersion, which is used when</span></span>
<span id="cb31-19"><a href="comparison-of-sccutseq-and-act.html#cb31-19" aria-hidden="true" tabindex="-1"></a><span class="co"># estimating standard errors for each segment mean</span></span>
<span id="cb31-20"><a href="comparison-of-sccutseq-and-act.html#cb31-20" aria-hidden="true" tabindex="-1"></a>overdispersion <span class="ot">=</span> <span class="cf">function</span>(v)</span>
<span id="cb31-21"><a href="comparison-of-sccutseq-and-act.html#cb31-21" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb31-22"><a href="comparison-of-sccutseq-and-act.html#cb31-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 3 elements, 2 differences, can find a standard deviation</span></span>
<span id="cb31-23"><a href="comparison-of-sccutseq-and-act.html#cb31-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(v) <span class="sc">&gt;=</span> <span class="dv">3</span>)</span>
<span id="cb31-24"><a href="comparison-of-sccutseq-and-act.html#cb31-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Differences between pairs of values</span></span>
<span id="cb31-25"><a href="comparison-of-sccutseq-and-act.html#cb31-25" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">=</span> v[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb31-26"><a href="comparison-of-sccutseq-and-act.html#cb31-26" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">=</span> v[<span class="sc">-</span><span class="fu">length</span>(v)]</span>
<span id="cb31-27"><a href="comparison-of-sccutseq-and-act.html#cb31-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Normalize the differences using the sum. The result should be around zero,</span></span>
<span id="cb31-28"><a href="comparison-of-sccutseq-and-act.html#cb31-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># plus or minus square root of the index of dispersion</span></span>
<span id="cb31-29"><a href="comparison-of-sccutseq-and-act.html#cb31-29" aria-hidden="true" tabindex="-1"></a>  vals.unfiltered <span class="ot">=</span> (y<span class="sc">-</span>x)<span class="sc">/</span><span class="fu">sqrt</span>(y<span class="sc">+</span>x)</span>
<span id="cb31-30"><a href="comparison-of-sccutseq-and-act.html#cb31-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove divide by zero cases, and--considering this is supposed to be count</span></span>
<span id="cb31-31"><a href="comparison-of-sccutseq-and-act.html#cb31-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># data--divide by almost-zero cases</span></span>
<span id="cb31-32"><a href="comparison-of-sccutseq-and-act.html#cb31-32" aria-hidden="true" tabindex="-1"></a>  vals <span class="ot">=</span> vals.unfiltered[y <span class="sc">+</span> x  <span class="sc">&gt;=</span> <span class="dv">1</span>]</span>
<span id="cb31-33"><a href="comparison-of-sccutseq-and-act.html#cb31-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check that there&#39;s anything left</span></span>
<span id="cb31-34"><a href="comparison-of-sccutseq-and-act.html#cb31-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(vals) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb31-35"><a href="comparison-of-sccutseq-and-act.html#cb31-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Assuming most of the normalized differences follow a normal distribution,</span></span>
<span id="cb31-36"><a href="comparison-of-sccutseq-and-act.html#cb31-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># estimate the standard deviation</span></span>
<span id="cb31-37"><a href="comparison-of-sccutseq-and-act.html#cb31-37" aria-hidden="true" tabindex="-1"></a>  val.sd <span class="ot">=</span> <span class="fu">l2e.normal.sd</span>(vals)</span>
<span id="cb31-38"><a href="comparison-of-sccutseq-and-act.html#cb31-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Square this standard deviation to obtain an estimate of the index of</span></span>
<span id="cb31-39"><a href="comparison-of-sccutseq-and-act.html#cb31-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># dispersion</span></span>
<span id="cb31-40"><a href="comparison-of-sccutseq-and-act.html#cb31-40" aria-hidden="true" tabindex="-1"></a>  iod <span class="ot">=</span> val.sd<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb31-41"><a href="comparison-of-sccutseq-and-act.html#cb31-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># subtract one to get the overdispersion criteria</span></span>
<span id="cb31-42"><a href="comparison-of-sccutseq-and-act.html#cb31-42" aria-hidden="true" tabindex="-1"></a>  iod.over <span class="ot">=</span> iod <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb31-43"><a href="comparison-of-sccutseq-and-act.html#cb31-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalizing by mean bincounts</span></span>
<span id="cb31-44"><a href="comparison-of-sccutseq-and-act.html#cb31-44" aria-hidden="true" tabindex="-1"></a>  iod.norm <span class="ot">=</span> iod.over<span class="sc">/</span><span class="fu">mean</span>(v)</span>
<span id="cb31-45"><a href="comparison-of-sccutseq-and-act.html#cb31-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(iod.norm)</span>
<span id="cb31-46"><a href="comparison-of-sccutseq-and-act.html#cb31-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-47"><a href="comparison-of-sccutseq-and-act.html#cb31-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-48"><a href="comparison-of-sccutseq-and-act.html#cb31-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Get count data for scCUTseq and ACT</span></span>
<span id="cb31-49"><a href="comparison-of-sccutseq-and-act.html#cb31-49" aria-hidden="true" tabindex="-1"></a>p3_counts <span class="ot">=</span> p3<span class="sc">$</span>counts[, p3_stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb31-50"><a href="comparison-of-sccutseq-and-act.html#cb31-50" aria-hidden="true" tabindex="-1"></a>CD5p_counts <span class="ot">=</span> CD5p<span class="sc">$</span>counts[, CD5p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb31-51"><a href="comparison-of-sccutseq-and-act.html#cb31-51" aria-hidden="true" tabindex="-1"></a>CD7p_counts <span class="ot">=</span> CD7p<span class="sc">$</span>counts[, CD7p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb31-52"><a href="comparison-of-sccutseq-and-act.html#cb31-52" aria-hidden="true" tabindex="-1"></a>CD4p_counts <span class="ot">=</span> CD4p<span class="sc">$</span>counts[, CD4p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb31-53"><a href="comparison-of-sccutseq-and-act.html#cb31-53" aria-hidden="true" tabindex="-1"></a>CD2p_counts <span class="ot">=</span> CD2p<span class="sc">$</span>counts[, CD2p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb31-54"><a href="comparison-of-sccutseq-and-act.html#cb31-54" aria-hidden="true" tabindex="-1"></a>CD6p_counts <span class="ot">=</span> CD6p<span class="sc">$</span>counts[, CD6p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb31-55"><a href="comparison-of-sccutseq-and-act.html#cb31-55" aria-hidden="true" tabindex="-1"></a>CD3p_counts <span class="ot">=</span> CD3p<span class="sc">$</span>counts[, CD3p<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb31-56"><a href="comparison-of-sccutseq-and-act.html#cb31-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-57"><a href="comparison-of-sccutseq-and-act.html#cb31-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine act</span></span>
<span id="cb31-58"><a href="comparison-of-sccutseq-and-act.html#cb31-58" aria-hidden="true" tabindex="-1"></a>act_counts <span class="ot">=</span> <span class="fu">do.call</span>(cbind, <span class="fu">list</span>(CD5p_counts, CD4p_counts, CD2p_counts,</span>
<span id="cb31-59"><a href="comparison-of-sccutseq-and-act.html#cb31-59" aria-hidden="true" tabindex="-1"></a>                                 CD7p_counts, CD6p_counts, CD3p_counts))</span>
<span id="cb31-60"><a href="comparison-of-sccutseq-and-act.html#cb31-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-61"><a href="comparison-of-sccutseq-and-act.html#cb31-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate overdispersion</span></span>
<span id="cb31-62"><a href="comparison-of-sccutseq-and-act.html#cb31-62" aria-hidden="true" tabindex="-1"></a>p3_overdispersion <span class="ot">=</span> <span class="fu">lapply</span>(p3_counts, <span class="cf">function</span>(x) {</span>
<span id="cb31-63"><a href="comparison-of-sccutseq-and-act.html#cb31-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">overdispersion</span>(x)</span>
<span id="cb31-64"><a href="comparison-of-sccutseq-and-act.html#cb31-64" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb31-65"><a href="comparison-of-sccutseq-and-act.html#cb31-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-66"><a href="comparison-of-sccutseq-and-act.html#cb31-66" aria-hidden="true" tabindex="-1"></a>act_overdispersion <span class="ot">=</span> <span class="fu">lapply</span>(act_counts, <span class="cf">function</span>(x) {</span>
<span id="cb31-67"><a href="comparison-of-sccutseq-and-act.html#cb31-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">overdispersion</span>(x)</span>
<span id="cb31-68"><a href="comparison-of-sccutseq-and-act.html#cb31-68" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb31-69"><a href="comparison-of-sccutseq-and-act.html#cb31-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-70"><a href="comparison-of-sccutseq-and-act.html#cb31-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Make into DT</span></span>
<span id="cb31-71"><a href="comparison-of-sccutseq-and-act.html#cb31-71" aria-hidden="true" tabindex="-1"></a>act_dt <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">method =</span> <span class="st">&quot;ACT&quot;</span>, <span class="at">overdispersion =</span> <span class="fu">unlist</span>(act_overdispersion))</span>
<span id="cb31-72"><a href="comparison-of-sccutseq-and-act.html#cb31-72" aria-hidden="true" tabindex="-1"></a>p3_dt <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">method =</span> <span class="st">&quot;scCUTseq&quot;</span>, <span class="at">overdispersion =</span> <span class="fu">unlist</span>(p3_overdispersion))</span>
<span id="cb31-73"><a href="comparison-of-sccutseq-and-act.html#cb31-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-74"><a href="comparison-of-sccutseq-and-act.html#cb31-74" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine DTs</span></span>
<span id="cb31-75"><a href="comparison-of-sccutseq-and-act.html#cb31-75" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">rbind</span>(act_dt, p3_dt)</span>
<span id="cb31-76"><a href="comparison-of-sccutseq-and-act.html#cb31-76" aria-hidden="true" tabindex="-1"></a>dt[, method <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(method, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;scCUTseq&quot;</span>, <span class="st">&quot;ACT&quot;</span>))]</span>
<span id="cb31-77"><a href="comparison-of-sccutseq-and-act.html#cb31-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-78"><a href="comparison-of-sccutseq-and-act.html#cb31-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot comparison</span></span>
<span id="cb31-79"><a href="comparison-of-sccutseq-and-act.html#cb31-79" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(dt, <span class="fu">aes</span>(<span class="at">x =</span> method, <span class="at">y =</span> overdispersion)) <span class="sc">+</span></span>
<span id="cb31-80"><a href="comparison-of-sccutseq-and-act.html#cb31-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb31-81"><a href="comparison-of-sccutseq-and-act.html#cb31-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> .<span class="dv">075</span>) <span class="sc">+</span> </span>
<span id="cb31-82"><a href="comparison-of-sccutseq-and-act.html#cb31-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Overdispersion&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/calculate%20overdispersion-1.png" width="672" /></p>
</div>
<div id="plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles<a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Plot both the ACT and scCUTseq genomewide heatmap to visually compare the two methods. Note: the plotting of these can take a while. Furthermore, we set <code>dendrogram = FALSE</code> since there is a known bug with the <code>ggdendro</code> package when you have many samples that are identical (in our case, diploid cells)</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="comparison-of-sccutseq-and-act.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in data</span></span>
<span id="cb32-2"><a href="comparison-of-sccutseq-and-act.html#cb32-2" aria-hidden="true" tabindex="-1"></a>p3_sccut <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_cnv.rds&quot;</span>)</span>
<span id="cb32-3"><a href="comparison-of-sccutseq-and-act.html#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="comparison-of-sccutseq-and-act.html#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select HQ cells</span></span>
<span id="cb32-5"><a href="comparison-of-sccutseq-and-act.html#cb32-5" aria-hidden="true" tabindex="-1"></a>p3_sccut_profiles <span class="ot">=</span> p3_sccut<span class="sc">$</span>copynumber[, p3_sccut<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb32-6"><a href="comparison-of-sccutseq-and-act.html#cb32-6" aria-hidden="true" tabindex="-1"></a>p3_sccut_profiles <span class="ot">=</span> p3_sccut_profiles[, <span class="fu">grepl</span>(<span class="st">&quot;NZ186|MS80|NZ187|NZ188|NZ189|NZ211&quot;</span>, <span class="fu">colnames</span>(p3_sccut_profiles)), with <span class="ot">=</span> F] <span class="co"># Select sections that also has ACT sequencing data</span></span>
<span id="cb32-7"><a href="comparison-of-sccutseq-and-act.html#cb32-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-8"><a href="comparison-of-sccutseq-and-act.html#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Make annotation table for scCUTseq</span></span>
<span id="cb32-9"><a href="comparison-of-sccutseq-and-act.html#cb32-9" aria-hidden="true" tabindex="-1"></a>p3_annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb32-10"><a href="comparison-of-sccutseq-and-act.html#cb32-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-11"><a href="comparison-of-sccutseq-and-act.html#cb32-11" aria-hidden="true" tabindex="-1"></a>scCUT_annot <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(p3_sccut_profiles),</span>
<span id="cb32-12"><a href="comparison-of-sccutseq-and-act.html#cb32-12" aria-hidden="true" tabindex="-1"></a>                         <span class="at">variable =</span> <span class="st">&quot;section&quot;</span>,</span>
<span id="cb32-13"><a href="comparison-of-sccutseq-and-act.html#cb32-13" aria-hidden="true" tabindex="-1"></a>                         <span class="at">value =</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(p3_sccut_profiles)))</span>
<span id="cb32-14"><a href="comparison-of-sccutseq-and-act.html#cb32-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-15"><a href="comparison-of-sccutseq-and-act.html#cb32-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge to get section information</span></span>
<span id="cb32-16"><a href="comparison-of-sccutseq-and-act.html#cb32-16" aria-hidden="true" tabindex="-1"></a>scCUT_annot <span class="ot">=</span> <span class="fu">merge</span>(scCUT_annot, p3_annot, <span class="at">by.x =</span> <span class="st">&quot;value&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;V1&quot;</span>)</span>
<span id="cb32-17"><a href="comparison-of-sccutseq-and-act.html#cb32-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-18"><a href="comparison-of-sccutseq-and-act.html#cb32-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot scCUTseq heatmap</span></span>
<span id="cb32-19"><a href="comparison-of-sccutseq-and-act.html#cb32-19" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(p3_sccut_profiles, p3_sccut<span class="sc">$</span>bins, <span class="at">dendrogram =</span> <span class="cn">FALSE</span>, <span class="at">annotation =</span> scCUT_annot[, .(sample, variable, V2)])</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20genomewideheatmaps-1.png" width="672" /></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="comparison-of-sccutseq-and-act.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get ACT profiles and make annotation</span></span>
<span id="cb33-2"><a href="comparison-of-sccutseq-and-act.html#cb33-2" aria-hidden="true" tabindex="-1"></a>p3_act <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_ACT.rds&quot;</span>)</span>
<span id="cb33-3"><a href="comparison-of-sccutseq-and-act.html#cb33-3" aria-hidden="true" tabindex="-1"></a>p3_act_profiles <span class="ot">=</span> p3_act<span class="sc">$</span>copynumber[, <span class="fu">sort</span>(p3_act<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample]), with <span class="ot">=</span> F]</span>
<span id="cb33-4"><a href="comparison-of-sccutseq-and-act.html#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="comparison-of-sccutseq-and-act.html#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Make annotation for ACT</span></span>
<span id="cb33-6"><a href="comparison-of-sccutseq-and-act.html#cb33-6" aria-hidden="true" tabindex="-1"></a>ACT_annot <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(p3_act_profiles),</span>
<span id="cb33-7"><a href="comparison-of-sccutseq-and-act.html#cb33-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">variable =</span> <span class="st">&quot;section&quot;</span>,</span>
<span id="cb33-8"><a href="comparison-of-sccutseq-and-act.html#cb33-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">value =</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(p3_act_profiles)))</span>
<span id="cb33-9"><a href="comparison-of-sccutseq-and-act.html#cb33-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-10"><a href="comparison-of-sccutseq-and-act.html#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot ACT heatmap</span></span>
<span id="cb33-11"><a href="comparison-of-sccutseq-and-act.html#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(p3_act_profiles, p3_act<span class="sc">$</span>bins, <span class="at">dendrogram =</span> <span class="cn">FALSE</span>, <span class="at">annotation =</span> ACT_annot)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20genomewideheatmaps-2.png" width="672" /></p>
</div>
<div id="plot-profile-plots-of-example-profiles" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Plot profile plots of example profiles<a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We selected 2 profiles of scCUTseq and 2 profiles of ACT to show the differences. These are selected manually but are representative for other plots (you can change the sample ID to manually check others)</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="comparison-of-sccutseq-and-act.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># profile selection</span></span>
<span id="cb34-2"><a href="comparison-of-sccutseq-and-act.html#cb34-2" aria-hidden="true" tabindex="-1"></a>sccut <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD27.rds&quot;</span>)</span>
<span id="cb34-3"><a href="comparison-of-sccutseq-and-act.html#cb34-3" aria-hidden="true" tabindex="-1"></a>act <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD1p.rds&quot;</span>)</span>
<span id="cb34-4"><a href="comparison-of-sccutseq-and-act.html#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="comparison-of-sccutseq-and-act.html#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="co"># scCUTseq cell 1</span></span>
<span id="cb34-6"><a href="comparison-of-sccutseq-and-act.html#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plotProfile</span>(sccut<span class="sc">$</span>copynumber<span class="sc">$</span>AGCCAACGGCA, sccut<span class="sc">$</span>counts_gc<span class="sc">$</span>AGCCAACGGCA <span class="sc">*</span> sccut<span class="sc">$</span>ploidies[sample <span class="sc">==</span> <span class="st">&quot;AGCCAACGGCA&quot;</span>, ploidy], sccut<span class="sc">$</span>bins)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20profile%20plots-1.png" width="672" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="comparison-of-sccutseq-and-act.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># scCUTseq cell 2</span></span>
<span id="cb35-2"><a href="comparison-of-sccutseq-and-act.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotProfile</span>(sccut<span class="sc">$</span>copynumber<span class="sc">$</span>AGCTTGCTCAT, sccut<span class="sc">$</span>counts_gc<span class="sc">$</span>AGCTTGCTCAT <span class="sc">*</span> sccut<span class="sc">$</span>ploidies[sample <span class="sc">==</span> <span class="st">&quot;AGCTTGCTCAT&quot;</span>, ploidy], sccut<span class="sc">$</span>bins)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20profile%20plots-2.png" width="672" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="comparison-of-sccutseq-and-act.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ACT cell 1</span></span>
<span id="cb36-2"><a href="comparison-of-sccutseq-and-act.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotProfile</span>(act<span class="sc">$</span>copynumber<span class="sc">$</span><span class="st">`</span><span class="at">296_S296</span><span class="st">`</span>, act<span class="sc">$</span>counts_gc<span class="sc">$</span><span class="st">`</span><span class="at">296_S296</span><span class="st">`</span> <span class="sc">*</span> act<span class="sc">$</span>ploidies[sample <span class="sc">==</span> <span class="st">&quot;296_S296&quot;</span>, ploidy], act<span class="sc">$</span>bins)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20profile%20plots-3.png" width="672" /></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="comparison-of-sccutseq-and-act.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ACT cell 2</span></span>
<span id="cb37-2"><a href="comparison-of-sccutseq-and-act.html#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotProfile</span>(act<span class="sc">$</span>copynumber<span class="sc">$</span><span class="st">`</span><span class="at">227_S227</span><span class="st">`</span>, act<span class="sc">$</span>counts_gc<span class="sc">$</span><span class="st">`</span><span class="at">227_S227</span><span class="st">`</span> <span class="sc">*</span> act<span class="sc">$</span>ploidies[sample <span class="sc">==</span> <span class="st">&quot;227_S227&quot;</span>, ploidy], act<span class="sc">$</span>bins)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20profile%20plots-4.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="technical-validation-of-sccutseq.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
