<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Copy number calling | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Copy number calling | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Copy number calling | _main.knit" />
  
  
  

<meta name="author" content="Luuk Harbers" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="preprocessing.html"/>
<link rel="next" href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>4</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>4.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="4.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>4.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="4.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>4.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>5</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>5.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="5.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>5.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="5.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>5.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="5.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>5.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="5.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>5.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>6</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>6.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="6.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>6.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="6.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>6.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="6.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>6.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>7</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="7.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>7.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="7.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>7.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="7.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>7.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="7.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>7.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="7.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>7.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="7.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>7.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="7.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>7.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="7.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>7.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>8</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="8.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>8.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="8.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>8.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="8.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>8.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="8.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>8.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>9</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>9.1</b> Loading in data</a></li>
<li class="chapter" data-level="9.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>9.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="9.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>9.3</b> Training the model</a></li>
<li class="chapter" data-level="9.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>9.4</b> Validating the model</a></li>
<li class="chapter" data-level="9.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>9.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>10</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="10.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>10.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>11</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="11.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>11.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="11.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>11.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>12</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="12.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>12.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>13</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>13.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>14</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="14.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>14.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="14.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>14.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>16</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="16.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>16.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>17</b> Monster cells</a>
<ul>
<li class="chapter" data-level="17.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>17.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="17.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>17.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="17.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>17.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html"><i class="fa fa-check"></i><b>18</b> Deletions and targeted panel</a>
<ul>
<li class="chapter" data-level="18.1" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>18.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="18.2" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>18.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a>
<ul>
<li class="chapter" data-level="19.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>19.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="copy-number-calling" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Copy number calling<a href="copy-number-calling.html#copy-number-calling" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="transform-bam-files" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Transform bam files<a href="copy-number-calling.html#transform-bam-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To be able to efficiently count reads in (variable width) genomic windows, we first transform bam files into bed files. We do this using the <code>bamtobed</code> function from <code>bedtools</code>. We make sure we only select reads that have a high mapping quality and are properly mapped (and paired in the case of paired-end sequencing). We only select chromosomes 1-22 and X and make sure the the bed file is properly sorted afterwards. We call this as follows:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="copy-number-calling.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-q</span> 30 <span class="at">-F</span> 1024 <span class="at">-b</span> {input} <span class="kw">|</span> <span class="ex">bedtools</span> bamtobed <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-E</span> <span class="st">&#39;^[0-9]|^X|^chr[0-9]|^chrX&#39;</span> <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb8-2"><a href="copy-number-calling.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span> <span class="at">-k1,1V</span> <span class="at">-k2,2n</span> <span class="at">-k3,3n</span> <span class="kw">|</span> <span class="fu">gzip</span> <span class="at">-c</span> <span class="op">&gt;</span> {output}</span></code></pre></div>
</div>
<div id="count-reads" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Count reads<a href="copy-number-calling.html#count-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We then use a file containing variable width bins, which can be found in the github repository <a href="https://github.com/BiCroLab/scCUTseq/tree/main/snakemake_pipelines/cnv-calling/files/hg19">here</a>, and count the reads in each genomic region. We use <code>bedtools intersect</code> to do this. We run this per sample and then combine all the samples (normally all 384 cells from one library) into one large file.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="copy-number-calling.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> {sample_name} <span class="op">&gt;</span> {output}</span>
<span id="cb9-2"><a href="copy-number-calling.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bedtools</span> intersect <span class="at">-nonamecheck</span> <span class="at">-F</span> 0.5 <span class="at">-sorted</span> <span class="at">-c</span> <span class="at">-a</span> {variable_width_bin_file} <span class="at">-b</span> {input.bed} <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f4</span> <span class="op">&gt;&gt;</span> {output}</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="copy-number-calling.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span> {input_files} <span class="kw">|</span> <span class="fu">gzip</span> <span class="op">&gt;</span> {output_file}</span></code></pre></div>
</div>
<div id="calling-copy-numbers" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Calling copy numbers<a href="copy-number-calling.html#calling-copy-numbers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The above generated files are then used to call copy numbers. We use a custom script to call copy number profiles and there are, depending on the use-case, different ways to run this script. Below is an example call and we will explain the parameters used after that.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="copy-number-calling.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> cnv_calling.R <span class="at">--counts</span> {counts} <span class="at">--bins</span> {bins} <span class="at">--binsize</span> {binsize} <span class="at">--blacklist</span> {blacklist} <span class="at">--normseg</span> {normseg} <span class="at">--gc</span> {gc_file} <span class="at">--segmentation</span> {segmentation_type} <span class="at">--penalty</span> {segmentation_penalty} <span class="at">--type</span> single <span class="at">--randomforest</span> {path_to_randomforest_model} <span class="at">--rfthreshold</span> {rf_threshold} <span class="at">--removethreshold</span> {removethreshold} <span class="at">--minploidy</span> {minploidy} <span class="at">--maxploidy</span> {maxploidy} <span class="at">--minpurity</span> {minpurity} <span class="at">--maxpurity</span> {maxpurity} <span class="at">--sex</span> {sex} <span class="at">--threads</span> {threads} <span class="at">--output</span> {output}</span></code></pre></div>
<p>Below you can find a brief explanation of the different parameters:</p>
<ul>
<li>counts: file containing binned counts</li>
<li>bins: file containing bins</li>
<li>binsize: integer denoting binsize used</li>
<li>blacklist: file containing regions that should be filtered out (telomeric/centromeric regions for instance)</li>
<li>normseg: additional normalization of regions if required</li>
<li>segmentation: either ‘joint’ or ‘single’ for <code>multipcf</code> or <code>CBS</code> segmentation, respectively.</li>
<li>penalty: penalty threshold used for joint segmentation. If using single segmentation this should be replaced by <code>--prune {prune_penalty}</code> and <code>--alpha {alpha_penalty}</code></li>
<li>type: ‘single’ or ‘bulk’ depending on if it’s single-cell sequencing or bulk sequencing</li>
<li>randomforest: path to the randomforest model</li>
<li>rfthreshold: threshold used for randomforest classification</li>
<li>minploidy, maxploidy, minpurity and maxpurity: minimum and maximum purity/ploidies for grid search</li>
<li>sex: either male or female depending on sex of the sample sequenced</li>
<li>threads: number of threads to use for parallel processing</li>
<li>output: output directory</li>
</ul>
</div>
<div id="generate-plots" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Generate plots<a href="copy-number-calling.html#generate-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Finally, we generate some profile plots and genomewide heatmaps based on the accomplished copy number calling. We do this with a custom R script which can be called as follows:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="copy-number-calling.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span>  plotting.R <span class="at">--rds</span> {input} <span class="at">--runtype</span> {run_type} <span class="at">--threads</span> {threads} <span class="at">--outdir</span> {outdir}</span></code></pre></div>
<p>Just like the preprocessing section, the copy number calling is also automatized and wrapped in a snakemake pipeline and is available in the github repository <a href="https://github.com/BiCroLab/scCUTseq/tree/main/snakemake_pipelines/cnv-calling">here</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="preprocessing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
