<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 Copy number profiles of brain neurons and skeletal muscle cells | scCUTseq paper</title>
  <meta name="description" content="10 Copy number profiles of brain neurons and skeletal muscle cells | scCUTseq paper" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="10 Copy number profiles of brain neurons and skeletal muscle cells | scCUTseq paper" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 Copy number profiles of brain neurons and skeletal muscle cells | scCUTseq paper" />
  
  
  

<meta name="author" content="Luuk Harbers" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="random-forest-classification.html"/>
<link rel="next" href="copy-number-profiles-of-other-prostate-cancer-patients.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>4</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>4.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="4.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>4.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="4.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>4.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>5</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>5.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="5.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>5.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="5.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>5.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="5.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>5.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="5.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>5.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>6</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>6.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="6.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>6.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="6.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>6.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="6.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>6.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>7</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="7.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>7.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="7.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>7.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="7.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>7.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="7.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>7.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="7.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>7.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="7.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>7.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="7.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>7.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="7.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>7.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>8</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="8.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>8.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="8.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>8.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="8.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>8.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="8.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>8.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>9</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>9.1</b> Loading in data</a></li>
<li class="chapter" data-level="9.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>9.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="9.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>9.3</b> Training the model</a></li>
<li class="chapter" data-level="9.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>9.4</b> Validating the model</a></li>
<li class="chapter" data-level="9.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>9.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>10</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="10.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>10.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>11</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="11.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>11.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="11.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>11.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>12</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="12.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>12.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>13</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>13.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>14</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="14.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>14.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="14.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>14.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>16</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="16.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>16.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>17</b> Monster cells</a>
<ul>
<li class="chapter" data-level="17.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>17.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="17.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>17.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="17.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>17.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html"><i class="fa fa-check"></i><b>18</b> Deletions and targeted panel</a>
<ul>
<li class="chapter" data-level="18.1" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>18.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="18.2" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>18.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a>
<ul>
<li class="chapter" data-level="19.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>19.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scCUTseq paper</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">10</span> Copy number profiles of brain neurons and skeletal muscle cells<a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This sections produces all the figures used in Supplementary Figure 6.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb104-2"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span>
<span id="cb104-3"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb104-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-4"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb104-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load functions</span></span>
<span id="cb104-5"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb104-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotHeatmap.R&quot;</span>)</span></code></pre></div>
<div id="plot-genomewide-heatmaps" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Plot genomewide heatmaps<a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First load data and then plot genomewide data using the <code>plotHeatmap()</code> function</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load all profiles</span></span>
<span id="cb105-2"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-2" aria-hidden="true" tabindex="-1"></a>dg20 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG20_cnv.rds&quot;</span>)</span>
<span id="cb105-3"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-3" aria-hidden="true" tabindex="-1"></a>dg21 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG21_cnv.rds&quot;</span>)</span>
<span id="cb105-4"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-4" aria-hidden="true" tabindex="-1"></a>dg22 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG22_cnv.rds&quot;</span>)</span>
<span id="cb105-5"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-5" aria-hidden="true" tabindex="-1"></a>dg33 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG33_cnv.rds&quot;</span>)</span>
<span id="cb105-6"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-6" aria-hidden="true" tabindex="-1"></a>dg39 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG39_cnv.rds&quot;</span>)</span>
<span id="cb105-7"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-7" aria-hidden="true" tabindex="-1"></a>dg40 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG40_cnv.rds&quot;</span>)</span>
<span id="cb105-8"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-9"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Go through each library and select HQ cells</span></span>
<span id="cb105-10"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-10" aria-hidden="true" tabindex="-1"></a>dg20_cn <span class="ot">=</span> dg20<span class="sc">$</span>copynumber[, dg20<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb105-11"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-11" aria-hidden="true" tabindex="-1"></a>dg21_cn <span class="ot">=</span> dg21<span class="sc">$</span>copynumber[, dg21<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb105-12"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-12" aria-hidden="true" tabindex="-1"></a>dg22_cn <span class="ot">=</span> dg22<span class="sc">$</span>copynumber[, dg22<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb105-13"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-13" aria-hidden="true" tabindex="-1"></a>dg33_cn <span class="ot">=</span> dg33<span class="sc">$</span>copynumber[, dg33<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb105-14"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-14" aria-hidden="true" tabindex="-1"></a>dg39_cn <span class="ot">=</span> dg39<span class="sc">$</span>copynumber[, dg39<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb105-15"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-15" aria-hidden="true" tabindex="-1"></a>dg40_cn <span class="ot">=</span> dg40<span class="sc">$</span>copynumber[, dg40<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb105-16"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-17"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Add prefix to generate unique colnames</span></span>
<span id="cb105-18"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-18" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dg20_cn, <span class="fu">paste0</span>(<span class="st">&quot;dg20_&quot;</span>, <span class="fu">colnames</span>(dg20_cn)))</span>
<span id="cb105-19"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-19" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dg21_cn, <span class="fu">paste0</span>(<span class="st">&quot;dg21_&quot;</span>, <span class="fu">colnames</span>(dg21_cn)))</span>
<span id="cb105-20"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-20" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dg22_cn, <span class="fu">paste0</span>(<span class="st">&quot;dg22_&quot;</span>, <span class="fu">colnames</span>(dg22_cn)))</span>
<span id="cb105-21"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-21" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dg33_cn, <span class="fu">paste0</span>(<span class="st">&quot;dg33_&quot;</span>, <span class="fu">colnames</span>(dg33_cn)))</span>
<span id="cb105-22"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-22" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dg39_cn, <span class="fu">paste0</span>(<span class="st">&quot;dg39_&quot;</span>, <span class="fu">colnames</span>(dg39_cn)))</span>
<span id="cb105-23"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-23" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dg40_cn, <span class="fu">paste0</span>(<span class="st">&quot;dg40_&quot;</span>, <span class="fu">colnames</span>(dg40_cn)))</span>
<span id="cb105-24"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-25"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine libraries with same cell type</span></span>
<span id="cb105-26"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-26" aria-hidden="true" tabindex="-1"></a>donor1_neun <span class="ot">=</span> <span class="fu">cbind</span>(dg20_cn, dg21_cn)</span>
<span id="cb105-27"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-27" aria-hidden="true" tabindex="-1"></a>donor1_skmusc <span class="ot">=</span> dg22_cn</span>
<span id="cb105-28"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-28" aria-hidden="true" tabindex="-1"></a>donor2_neun <span class="ot">=</span> <span class="fu">cbind</span>(dg33_cn, dg39_cn)</span>
<span id="cb105-29"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-29" aria-hidden="true" tabindex="-1"></a>donor2_skmusc <span class="ot">=</span> dg40_cn</span>
<span id="cb105-30"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-31"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Make annotation for brain cells</span></span>
<span id="cb105-32"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-32" aria-hidden="true" tabindex="-1"></a>donor1_annot <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(donor1_neun),</span>
<span id="cb105-33"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-33" aria-hidden="true" tabindex="-1"></a>                          <span class="at">variable =</span> <span class="st">&quot;celltype&quot;</span>,</span>
<span id="cb105-34"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-34" aria-hidden="true" tabindex="-1"></a>                          <span class="at">value =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;NeuN+&quot;</span>, <span class="fu">ncol</span>(dg20_cn)), <span class="fu">rep</span>(<span class="st">&quot;NeuN-&quot;</span>, <span class="fu">ncol</span>(dg21_cn))))</span>
<span id="cb105-35"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-36"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-36" aria-hidden="true" tabindex="-1"></a>donor2_annot <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(donor2_neun),</span>
<span id="cb105-37"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-37" aria-hidden="true" tabindex="-1"></a>                          <span class="at">variable =</span> <span class="st">&quot;celltype&quot;</span>,</span>
<span id="cb105-38"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-38" aria-hidden="true" tabindex="-1"></a>                          <span class="at">value =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;NeuN&quot;</span>, <span class="fu">ncol</span>(dg33_cn)), <span class="fu">rep</span>(<span class="st">&quot;NeuN+&quot;</span>, <span class="fu">ncol</span>(dg39_cn))))</span>
<span id="cb105-39"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-40"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmaps for donor 1</span></span>
<span id="cb105-41"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb105-41" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(donor1_neun, dg20<span class="sc">$</span>bins, <span class="at">annotation =</span> donor1_annot, <span class="at">dendrogram =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/load%20data-1.png" width="672" /></p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(donor1_skmusc, dg20<span class="sc">$</span>bins, <span class="at">dendrogram =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/load%20data-2.png" width="672" /></p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmaps for donor 2</span></span>
<span id="cb107-2"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(donor2_neun, dg20<span class="sc">$</span>bins, <span class="at">annotation =</span> donor2_annot, <span class="at">dendrogram =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/load%20data-3.png" width="672" /></p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(donor2_skmusc, dg20<span class="sc">$</span>bins, <span class="at">dendrogram =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/load%20data-4.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="random-forest-classification.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="copy-number-profiles-of-other-prostate-cancer-patients.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
