<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>11 Copy number profiles of other prostate cancer patients | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="11 Copy number profiles of other prostate cancer patients | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="11 Copy number profiles of other prostate cancer patients | _main.knit" />
  
  
  

<meta name="author" content="Luuk Harbers" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"/>
<link rel="next" href="nebnext-versus-sccutseq.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>4</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>4.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="4.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>4.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="4.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>4.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>5</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>5.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="5.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>5.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="5.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>5.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="5.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>5.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="5.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>5.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>6</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>6.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="6.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>6.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="6.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>6.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="6.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>6.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>7</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="7.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>7.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="7.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>7.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="7.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>7.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="7.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>7.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="7.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>7.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="7.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>7.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="7.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>7.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="7.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>7.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>8</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="8.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>8.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="8.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>8.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="8.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>8.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="8.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>8.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>9</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>9.1</b> Loading in data</a></li>
<li class="chapter" data-level="9.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>9.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="9.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>9.3</b> Training the model</a></li>
<li class="chapter" data-level="9.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>9.4</b> Validating the model</a></li>
<li class="chapter" data-level="9.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>9.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>10</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="10.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>10.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>11</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="11.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>11.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="11.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>11.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>12</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="12.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>12.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>13</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>13.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>14</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="14.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>14.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="14.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>14.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>16</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="16.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>16.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>17</b> Monster cells</a>
<ul>
<li class="chapter" data-level="17.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>17.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="17.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>17.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="17.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>17.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html"><i class="fa fa-check"></i><b>18</b> Deletions and targeted panel</a>
<ul>
<li class="chapter" data-level="18.1" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>18.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="18.2" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>18.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a>
<ul>
<li class="chapter" data-level="19.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>19.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="copy-number-profiles-of-other-prostate-cancer-patients" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">11</span> Copy number profiles of other prostate cancer patients<a href="copy-number-profiles-of-other-prostate-cancer-patients.html#copy-number-profiles-of-other-prostate-cancer-patients" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This sections produces all the figures used in Supplementary Figure 7.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb109-2"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span>
<span id="cb109-3"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb109-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-4"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb109-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load functions</span></span>
<span id="cb109-5"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb109-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotHeatmap.R&quot;</span>)</span></code></pre></div>
<div id="plot-genomewide-heatmap" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Plot genomewide heatmap<a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First we look at the overall copynumber profiles of the other 4 prostate cancer patients. We sequenced 2 sections (96 cells input) for each patient.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb110-2"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-2" aria-hidden="true" tabindex="-1"></a>P2 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P2_cnv.rds&quot;</span>)</span>
<span id="cb110-3"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-3" aria-hidden="true" tabindex="-1"></a>P4 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P4_cnv.rds&quot;</span>)</span>
<span id="cb110-4"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-4" aria-hidden="true" tabindex="-1"></a>P5 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P5_cnv.rds&quot;</span>)</span>
<span id="cb110-5"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-5" aria-hidden="true" tabindex="-1"></a>P7 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P7_cnv.rds&quot;</span>)</span>
<span id="cb110-6"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-7"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Select HQ cells</span></span>
<span id="cb110-8"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-8" aria-hidden="true" tabindex="-1"></a>p2_profiles <span class="ot">=</span> P2<span class="sc">$</span>copynumber[, P2<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb110-9"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-9" aria-hidden="true" tabindex="-1"></a>p4_profiles <span class="ot">=</span> P4<span class="sc">$</span>copynumber[, P4<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb110-10"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-10" aria-hidden="true" tabindex="-1"></a>p5_profiles <span class="ot">=</span> P5<span class="sc">$</span>copynumber[, P5<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb110-11"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-11" aria-hidden="true" tabindex="-1"></a>p7_profiles <span class="ot">=</span> P7<span class="sc">$</span>copynumber[, P7<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb110-12"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-13"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Set blocks to block 1 and 2</span></span>
<span id="cb110-14"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-14" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(p2_profiles, <span class="fu">gsub</span>(<span class="st">&quot;L6C3&quot;</span>, <span class="st">&quot;P2_block1&quot;</span>, <span class="fu">colnames</span>(p2_profiles)))</span>
<span id="cb110-15"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-15" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(p2_profiles, <span class="fu">gsub</span>(<span class="st">&quot;P2L2C3&quot;</span>, <span class="st">&quot;P2_block2&quot;</span>, <span class="fu">colnames</span>(p2_profiles)))</span>
<span id="cb110-16"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-17"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-17" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(p4_profiles, <span class="fu">gsub</span>(<span class="st">&quot;L6C4&quot;</span>, <span class="st">&quot;P4_block1&quot;</span>, <span class="fu">colnames</span>(p4_profiles)))</span>
<span id="cb110-18"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-18" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(p4_profiles, <span class="fu">gsub</span>(<span class="st">&quot;L2C3&quot;</span>, <span class="st">&quot;P4_block2&quot;</span>, <span class="fu">colnames</span>(p4_profiles)))</span>
<span id="cb110-19"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-20"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-20" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(p5_profiles, <span class="fu">gsub</span>(<span class="st">&quot;L7C5&quot;</span>, <span class="st">&quot;P5_block1&quot;</span>, <span class="fu">colnames</span>(p5_profiles)))</span>
<span id="cb110-21"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-21" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(p5_profiles, <span class="fu">gsub</span>(<span class="st">&quot;L5C5&quot;</span>, <span class="st">&quot;P5_block2&quot;</span>, <span class="fu">colnames</span>(p5_profiles)))</span>
<span id="cb110-22"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-23"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-23" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(p7_profiles, <span class="fu">gsub</span>(<span class="st">&quot;L7C3&quot;</span>, <span class="st">&quot;P7_block1&quot;</span>, <span class="fu">colnames</span>(p7_profiles)))</span>
<span id="cb110-24"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-24" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(p7_profiles, <span class="fu">gsub</span>(<span class="st">&quot;L3C3&quot;</span>, <span class="st">&quot;P7_block2&quot;</span>, <span class="fu">colnames</span>(p7_profiles)))</span>
<span id="cb110-25"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-26"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine data</span></span>
<span id="cb110-27"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-27" aria-hidden="true" tabindex="-1"></a>total <span class="ot">=</span> <span class="fu">do.call</span>(cbind, <span class="fu">list</span>(p2_profiles, p4_profiles, p5_profiles, p7_profiles))</span>
<span id="cb110-28"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-28" aria-hidden="true" tabindex="-1"></a>bins <span class="ot">=</span> P2<span class="sc">$</span>bins</span>
<span id="cb110-29"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-30"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Get annotation</span></span>
<span id="cb110-31"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-31" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(total),</span>
<span id="cb110-32"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">patient =</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(total)),</span>
<span id="cb110-33"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">block =</span> <span class="fu">gsub</span>(<span class="st">&quot;P._|_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(total)))</span>
<span id="cb110-34"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-34" aria-hidden="true" tabindex="-1"></a>annot_m <span class="ot">=</span> <span class="fu">melt</span>(annot, <span class="at">id.vars =</span> <span class="st">&quot;sample&quot;</span>)</span>
<span id="cb110-35"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-36"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb110-37"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb110-37" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(total, bins, <span class="at">annotation =</span> annot_m, <span class="at">linesize =</span> <span class="fl">1.5</span>, <span class="at">rasterize =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20genomewide%20heatmap-1.png" width="672" /></p>
</div>
<div id="plot-cell-type-distribution" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Plot cell type distribution<a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we plot the distribution of the three different cell types that we detected earlier in the other 2 prostate cancer samples.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set diploid vector to compare against</span></span>
<span id="cb111-2"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-2" aria-hidden="true" tabindex="-1"></a>diploid_vector <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">2</span>, <span class="fu">sum</span>(bins<span class="sc">$</span>chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>)), <span class="fu">rep</span>(<span class="dv">1</span>, <span class="fu">sum</span>(bins<span class="sc">$</span>chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>)))</span>
<span id="cb111-3"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-4"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep these profiles</span></span>
<span id="cb111-5"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-5" aria-hidden="true" tabindex="-1"></a>diploid_check <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">colnames</span>(total), <span class="cf">function</span>(cell) {</span>
<span id="cb111-6"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sum</span>(total[[cell]] <span class="sc">!=</span> diploid_vector)</span>
<span id="cb111-7"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-7" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb111-8"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-9"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-9" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="fu">names</span>(diploid_check), <span class="at">num_altered =</span> diploid_check)</span>
<span id="cb111-10"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-11"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotate celltype</span></span>
<span id="cb111-12"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-12" aria-hidden="true" tabindex="-1"></a>dt[num_altered <span class="sc">==</span> <span class="dv">0</span>, celltype <span class="sc">:</span><span class="er">=</span> <span class="st">&quot;diploid&quot;</span>]</span>
<span id="cb111-13"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-13" aria-hidden="true" tabindex="-1"></a>dt[num_altered <span class="sc">!=</span> <span class="dv">0</span> <span class="sc">&amp;</span> num_altered <span class="sc">&lt;=</span> <span class="fl">0.25</span> <span class="sc">*</span> <span class="fu">nrow</span>(bins), celltype <span class="sc">:</span><span class="er">=</span> <span class="st">&quot;pseudo-diploid&quot;</span>]</span>
<span id="cb111-14"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-14" aria-hidden="true" tabindex="-1"></a>dt[num_altered <span class="sc">&gt;</span> <span class="fl">0.25</span> <span class="sc">*</span> <span class="fu">nrow</span>(bins), celltype <span class="sc">:</span><span class="er">=</span> <span class="st">&quot;monster&quot;</span>]</span>
<span id="cb111-15"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-16"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Get patient info</span></span>
<span id="cb111-17"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-17" aria-hidden="true" tabindex="-1"></a>dt[, patient <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb111-18"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-18" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> dt[, .N, by <span class="ot">=</span> .(celltype, patient)]</span>
<span id="cb111-19"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-19" aria-hidden="true" tabindex="-1"></a>counts_patient <span class="ot">=</span> dt[, .N, by <span class="ot">=</span> patient]</span>
<span id="cb111-20"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-21"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Get fraction</span></span>
<span id="cb111-22"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-22" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">merge</span>(counts, counts_patient, <span class="at">by =</span> <span class="st">&quot;patient&quot;</span>)</span>
<span id="cb111-23"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-23" aria-hidden="true" tabindex="-1"></a>counts[, fraction <span class="sc">:</span><span class="er">=</span> N.x <span class="sc">/</span> N.y]</span>
<span id="cb111-24"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-25"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Set order</span></span>
<span id="cb111-26"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-26" aria-hidden="true" tabindex="-1"></a>counts[, celltype <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(celltype, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;diploid&quot;</span>, <span class="st">&quot;pseudo-diploid&quot;</span>, <span class="st">&quot;monster&quot;</span>))]</span>
<span id="cb111-27"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-28"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(counts, <span class="fu">aes</span>(<span class="at">x =</span> patient, <span class="at">y =</span> fraction, <span class="at">fill =</span> celltype, <span class="at">label =</span> N.x)) <span class="sc">+</span></span>
<span id="cb111-29"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb111-30"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>()) <span class="sc">+</span></span>
<span id="cb111-31"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Percentage of cell type&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Cell type&quot;</span>) <span class="sc">+</span></span>
<span id="cb111-32"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">paste0</span>(<span class="fu">round</span>(fraction <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">&quot;%</span><span class="sc">\n</span><span class="st">(n = &quot;</span>, N.x, <span class="st">&quot;)&quot;</span>)),</span>
<span id="cb111-33"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-33" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> .<span class="dv">5</span>), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb111-34"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_npg</span>() <span class="sc">+</span></span>
<span id="cb111-35"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#cb111-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20cell%20type%20distribution-1.png" width="576" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="nebnext-versus-sccutseq.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
