<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>18 Deletions and targeted panel | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="18 Deletions and targeted panel | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="18 Deletions and targeted panel | _main.knit" />
  
  
  

<meta name="author" content="Luuk Harbers" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="monster-cells.html"/>
<link rel="next" href="session-info.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>4</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>4.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="4.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>4.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="4.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>4.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>5</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>5.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="5.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>5.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="5.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>5.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="5.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>5.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="5.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>5.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>6</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>6.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="6.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>6.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="6.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>6.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="6.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>6.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>7</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="7.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>7.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="7.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>7.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="7.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>7.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="7.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>7.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="7.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>7.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="7.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>7.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="7.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>7.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="7.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>7.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>8</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="8.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>8.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="8.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>8.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="8.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>8.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="8.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>8.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>9</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>9.1</b> Loading in data</a></li>
<li class="chapter" data-level="9.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>9.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="9.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>9.3</b> Training the model</a></li>
<li class="chapter" data-level="9.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>9.4</b> Validating the model</a></li>
<li class="chapter" data-level="9.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>9.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>10</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="10.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>10.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>11</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="11.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>11.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="11.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>11.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>12</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="12.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>12.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>13</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>13.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>14</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="14.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>14.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="14.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>14.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>16</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="16.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>16.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>17</b> Monster cells</a>
<ul>
<li class="chapter" data-level="17.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>17.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="17.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>17.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="17.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>17.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html"><i class="fa fa-check"></i><b>18</b> Deletions and targeted panel</a>
<ul>
<li class="chapter" data-level="18.1" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>18.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="18.2" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>18.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a>
<ul>
<li class="chapter" data-level="19.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>19.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="deletions-and-targeted-panel" class="section level1 hasAnchor" number="18">
<h1><span class="header-section-number">18</span> Deletions and targeted panel<a href="deletions-and-targeted-panel.html#deletions-and-targeted-panel" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This sections produces all the figures used in Supplementary Figure 19.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="deletions-and-targeted-panel.html#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb139-2"><a href="deletions-and-targeted-panel.html#cb139-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span></code></pre></div>
<div id="trrfer-specific-subclones" class="section level2 hasAnchor" number="18.1">
<h2><span class="header-section-number">18.1</span> TRR/FER specific subclones<a href="deletions-and-targeted-panel.html#trrfer-specific-subclones" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First we look at subclones (identified based on copynumber profiles previously) that are exclusively present in Tumour-Rich Regions (TRRs) or Focally Enriched regions (FERs). For these subclones we plot the genes that are altered. The spatial distributions of cells from these subclones can be found in the plots created previously (<code>scCUTseq_subclone-distributions.Rmd</code>).</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="deletions-and-targeted-panel.html#cb140-1" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P6_median_cn.tsv&quot;</span>)</span>
<span id="cb140-2"><a href="deletions-and-targeted-panel.html#cb140-2" aria-hidden="true" tabindex="-1"></a>clones <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P6_clones.tsv&quot;</span>)</span>
<span id="cb140-3"><a href="deletions-and-targeted-panel.html#cb140-3" aria-hidden="true" tabindex="-1"></a>cosmic <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/genelists/cosmic_gene-census.tsv&quot;</span>)</span>
<span id="cb140-4"><a href="deletions-and-targeted-panel.html#cb140-4" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P6.tsv&quot;</span>, <span class="at">header =</span> F)</span>
<span id="cb140-5"><a href="deletions-and-targeted-panel.html#cb140-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-6"><a href="deletions-and-targeted-panel.html#cb140-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge clones with annot</span></span>
<span id="cb140-7"><a href="deletions-and-targeted-panel.html#cb140-7" aria-hidden="true" tabindex="-1"></a>clones[, library <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample_id)]</span>
<span id="cb140-8"><a href="deletions-and-targeted-panel.html#cb140-8" aria-hidden="true" tabindex="-1"></a>clones_annot <span class="ot">=</span> <span class="fu">merge</span>(clones, annot, <span class="at">by.x =</span> <span class="st">&quot;library&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;V1&quot;</span>)</span>
<span id="cb140-9"><a href="deletions-and-targeted-panel.html#cb140-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-10"><a href="deletions-and-targeted-panel.html#cb140-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Select clones that are only present in cancer/focal</span></span>
<span id="cb140-11"><a href="deletions-and-targeted-panel.html#cb140-11" aria-hidden="true" tabindex="-1"></a>total <span class="ot">=</span> clones_annot[, .(<span class="fu">paste</span>(<span class="fu">unique</span>(V3), <span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)), by <span class="ot">=</span> cluster]</span>
<span id="cb140-12"><a href="deletions-and-targeted-panel.html#cb140-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-13"><a href="deletions-and-targeted-panel.html#cb140-13" aria-hidden="true" tabindex="-1"></a>clones_selected <span class="ot">=</span> total[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;Normal&quot;</span>, V1)]</span>
<span id="cb140-14"><a href="deletions-and-targeted-panel.html#cb140-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-15"><a href="deletions-and-targeted-panel.html#cb140-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Get selected</span></span>
<span id="cb140-16"><a href="deletions-and-targeted-panel.html#cb140-16" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> profiles[cluster <span class="sc">%in%</span> clones_selected<span class="sc">$</span>cluster]</span>
<span id="cb140-17"><a href="deletions-and-targeted-panel.html#cb140-17" aria-hidden="true" tabindex="-1"></a>profiles <span class="ot">=</span> profiles[(total_cn <span class="sc">!=</span> <span class="dv">2</span> <span class="sc">&amp;</span> chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>) <span class="sc">|</span> (total_cn <span class="sc">!=</span> <span class="dv">1</span> <span class="sc">&amp;</span> chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>)]</span>
<span id="cb140-18"><a href="deletions-and-targeted-panel.html#cb140-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-19"><a href="deletions-and-targeted-panel.html#cb140-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Get overlaps</span></span>
<span id="cb140-20"><a href="deletions-and-targeted-panel.html#cb140-20" aria-hidden="true" tabindex="-1"></a><span class="fu">setkey</span>(profiles, chr, start, end)</span>
<span id="cb140-21"><a href="deletions-and-targeted-panel.html#cb140-21" aria-hidden="true" tabindex="-1"></a><span class="fu">setkey</span>(cosmic, chr, start, end)</span>
<span id="cb140-22"><a href="deletions-and-targeted-panel.html#cb140-22" aria-hidden="true" tabindex="-1"></a>overlaps <span class="ot">=</span> <span class="fu">foverlaps</span>(cosmic, profiles)[<span class="sc">!</span><span class="fu">is.na</span>(total_cn)]</span>
<span id="cb140-23"><a href="deletions-and-targeted-panel.html#cb140-23" aria-hidden="true" tabindex="-1"></a>overlaps[, alteration <span class="sc">:</span><span class="er">=</span> <span class="fu">ifelse</span>(total_cn <span class="sc">&lt;</span> <span class="dv">2</span>, <span class="st">&quot;Deleted&quot;</span>, <span class="st">&quot;Amplified&quot;</span>)]</span>
<span id="cb140-24"><a href="deletions-and-targeted-panel.html#cb140-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-25"><a href="deletions-and-targeted-panel.html#cb140-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique</span></span>
<span id="cb140-26"><a href="deletions-and-targeted-panel.html#cb140-26" aria-hidden="true" tabindex="-1"></a>unique_overlaps <span class="ot">=</span> <span class="fu">unique</span>(overlaps[, .(name, gene, alteration)])</span>
<span id="cb140-27"><a href="deletions-and-targeted-panel.html#cb140-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-28"><a href="deletions-and-targeted-panel.html#cb140-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Make oncoprint</span></span>
<span id="cb140-29"><a href="deletions-and-targeted-panel.html#cb140-29" aria-hidden="true" tabindex="-1"></a>unique_overlaps_wide <span class="ot">=</span> <span class="fu">dcast</span>(unique_overlaps, name <span class="sc">~</span> gene, <span class="at">value.var =</span> <span class="st">&quot;alteration&quot;</span>)</span>
<span id="cb140-30"><a href="deletions-and-targeted-panel.html#cb140-30" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">=</span> <span class="fu">as.matrix</span>(unique_overlaps_wide[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(unique_overlaps_wide)])</span>
<span id="cb140-31"><a href="deletions-and-targeted-panel.html#cb140-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">=</span> unique_overlaps_wide<span class="sc">$</span>name</span>
<span id="cb140-32"><a href="deletions-and-targeted-panel.html#cb140-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-33"><a href="deletions-and-targeted-panel.html#cb140-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot oncoprint</span></span>
<span id="cb140-34"><a href="deletions-and-targeted-panel.html#cb140-34" aria-hidden="true" tabindex="-1"></a><span class="co"># colors</span></span>
<span id="cb140-35"><a href="deletions-and-targeted-panel.html#cb140-35" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Set1&quot;</span>)[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)]</span>
<span id="cb140-36"><a href="deletions-and-targeted-panel.html#cb140-36" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cols) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Amplified&quot;</span>, <span class="st">&quot;Deleted&quot;</span>)</span>
<span id="cb140-37"><a href="deletions-and-targeted-panel.html#cb140-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-38"><a href="deletions-and-targeted-panel.html#cb140-38" aria-hidden="true" tabindex="-1"></a><span class="fu">oncoPrint</span>(<span class="fu">t</span>(mat),</span>
<span id="cb140-39"><a href="deletions-and-targeted-panel.html#cb140-39" aria-hidden="true" tabindex="-1"></a>          <span class="at">alter_fun =</span> <span class="fu">list</span>(</span>
<span id="cb140-40"><a href="deletions-and-targeted-panel.html#cb140-40" aria-hidden="true" tabindex="-1"></a>            <span class="at">background =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.9</span>, <span class="at">height =</span> <span class="fl">0.9</span>, <span class="at">fill =</span> <span class="st">&quot;#FFFFFF&quot;</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> .<span class="dv">1</span>),</span>
<span id="cb140-41"><a href="deletions-and-targeted-panel.html#cb140-41" aria-hidden="true" tabindex="-1"></a>            <span class="at">Amplified =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.85</span>, <span class="at">height =</span> <span class="fl">0.85</span>, <span class="at">fill =</span> cols[<span class="st">&quot;Amplified&quot;</span>]),</span>
<span id="cb140-42"><a href="deletions-and-targeted-panel.html#cb140-42" aria-hidden="true" tabindex="-1"></a>            <span class="at">Deleted =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.85</span>, <span class="at">height =</span> <span class="fl">0.85</span>, <span class="at">fill =</span> cols[<span class="st">&quot;Deleted&quot;</span>])),</span>
<span id="cb140-43"><a href="deletions-and-targeted-panel.html#cb140-43" aria-hidden="true" tabindex="-1"></a>          <span class="at">col =</span> cols, <span class="at">border =</span> <span class="st">&quot;black&quot;</span>, <span class="at">show_column_names =</span> T, <span class="at">show_row_names =</span> T, <span class="at">remove_empty_rows =</span> T,</span>
<span id="cb140-44"><a href="deletions-and-targeted-panel.html#cb140-44" aria-hidden="true" tabindex="-1"></a>          <span class="at">show_pct =</span> F, <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">17</span>), <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">22</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/Select%20subclones-1.png" width="384" /></p>
</div>
<div id="variant-allele-frequency-snvs" class="section level2 hasAnchor" number="18.2">
<h2><span class="header-section-number">18.2</span> Variant allele frequency SNVs<a href="deletions-and-targeted-panel.html#variant-allele-frequency-snvs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next we wanted to see the distribution of the Variant Allele Frequencies in both of our patients. First we plot P3.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="deletions-and-targeted-panel.html#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in SNV data</span></span>
<span id="cb141-2"><a href="deletions-and-targeted-panel.html#cb141-2" aria-hidden="true" tabindex="-1"></a>snv <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/mutations/P3_filtered_SNVs.tsv&quot;</span>)</span>
<span id="cb141-3"><a href="deletions-and-targeted-panel.html#cb141-3" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>,</span>
<span id="cb141-4"><a href="deletions-and-targeted-panel.html#cb141-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">header =</span> F, </span>
<span id="cb141-5"><a href="deletions-and-targeted-panel.html#cb141-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span>  <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;pathology&quot;</span>))</span>
<span id="cb141-6"><a href="deletions-and-targeted-panel.html#cb141-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-7"><a href="deletions-and-targeted-panel.html#cb141-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-8"><a href="deletions-and-targeted-panel.html#cb141-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove underscore in sample</span></span>
<span id="cb141-9"><a href="deletions-and-targeted-panel.html#cb141-9" aria-hidden="true" tabindex="-1"></a>snv[, SAMPLE <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, SAMPLE)]</span>
<span id="cb141-10"><a href="deletions-and-targeted-panel.html#cb141-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-11"><a href="deletions-and-targeted-panel.html#cb141-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove LOC/2nd genes</span></span>
<span id="cb141-12"><a href="deletions-and-targeted-panel.html#cb141-12" aria-hidden="true" tabindex="-1"></a>snv[, GeneName <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;LOC.*:&quot;</span>, <span class="st">&quot;&quot;</span>, GeneName)]</span>
<span id="cb141-13"><a href="deletions-and-targeted-panel.html#cb141-13" aria-hidden="true" tabindex="-1"></a>snv[, GeneName <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>, <span class="st">&quot;&quot;</span>, GeneName)]</span>
<span id="cb141-14"><a href="deletions-and-targeted-panel.html#cb141-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-15"><a href="deletions-and-targeted-panel.html#cb141-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Get VAFs</span></span>
<span id="cb141-16"><a href="deletions-and-targeted-panel.html#cb141-16" aria-hidden="true" tabindex="-1"></a>muts_vaf <span class="ot">=</span> snv[, .(<span class="at">mean_vaf =</span> <span class="fu">mean</span>(VAF), <span class="at">count =</span> .N), by <span class="ot">=</span> GeneName]</span>
<span id="cb141-17"><a href="deletions-and-targeted-panel.html#cb141-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-18"><a href="deletions-and-targeted-panel.html#cb141-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb141-19"><a href="deletions-and-targeted-panel.html#cb141-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(muts_vaf, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> mean_vaf, <span class="at">label =</span> GeneName, <span class="at">color =</span> count)) <span class="sc">+</span></span>
<span id="cb141-20"><a href="deletions-and-targeted-panel.html#cb141-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb141-21"><a href="deletions-and-targeted-panel.html#cb141-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">position =</span> <span class="fu">position_quasirandom</span>()) <span class="sc">+</span></span>
<span id="cb141-22"><a href="deletions-and-targeted-panel.html#cb141-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">name =</span> <span class="st">&quot;Number of sections</span><span class="sc">\n</span><span class="st">with gene mutated&quot;</span>, <span class="at">option =</span> <span class="st">&quot;D&quot;</span>) <span class="sc">+</span></span>
<span id="cb141-23"><a href="deletions-and-targeted-panel.html#cb141-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Mean VAF&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">color =</span> <span class="st">&quot;Number of sections</span><span class="sc">\n</span><span class="st">with gene mutated&quot;</span>) <span class="sc">+</span></span>
<span id="cb141-24"><a href="deletions-and-targeted-panel.html#cb141-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/VAF%20distribution%20P3-1.png" width="672" /></p>
<p>Following this, we do the same for P6.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="deletions-and-targeted-panel.html#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in SNV data</span></span>
<span id="cb142-2"><a href="deletions-and-targeted-panel.html#cb142-2" aria-hidden="true" tabindex="-1"></a>snv <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/mutations/P6_filtered_SNVs.tsv&quot;</span>)</span>
<span id="cb142-3"><a href="deletions-and-targeted-panel.html#cb142-3" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P6.tsv&quot;</span>,</span>
<span id="cb142-4"><a href="deletions-and-targeted-panel.html#cb142-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">header =</span> F, </span>
<span id="cb142-5"><a href="deletions-and-targeted-panel.html#cb142-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span>  <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;pathology&quot;</span>))</span>
<span id="cb142-6"><a href="deletions-and-targeted-panel.html#cb142-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-7"><a href="deletions-and-targeted-panel.html#cb142-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-8"><a href="deletions-and-targeted-panel.html#cb142-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove underscore in sample</span></span>
<span id="cb142-9"><a href="deletions-and-targeted-panel.html#cb142-9" aria-hidden="true" tabindex="-1"></a>snv[, SAMPLE <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, SAMPLE)]</span>
<span id="cb142-10"><a href="deletions-and-targeted-panel.html#cb142-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-11"><a href="deletions-and-targeted-panel.html#cb142-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove LOC/2nd genes</span></span>
<span id="cb142-12"><a href="deletions-and-targeted-panel.html#cb142-12" aria-hidden="true" tabindex="-1"></a>snv[, GeneName <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;LOC.*:&quot;</span>, <span class="st">&quot;&quot;</span>, GeneName)]</span>
<span id="cb142-13"><a href="deletions-and-targeted-panel.html#cb142-13" aria-hidden="true" tabindex="-1"></a>snv[, GeneName <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>, <span class="st">&quot;&quot;</span>, GeneName)]</span>
<span id="cb142-14"><a href="deletions-and-targeted-panel.html#cb142-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-15"><a href="deletions-and-targeted-panel.html#cb142-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Get VAFs</span></span>
<span id="cb142-16"><a href="deletions-and-targeted-panel.html#cb142-16" aria-hidden="true" tabindex="-1"></a>muts_vaf <span class="ot">=</span> snv[, .(<span class="at">mean_vaf =</span> <span class="fu">mean</span>(VAF), <span class="at">count =</span> .N), by <span class="ot">=</span> GeneName]</span>
<span id="cb142-17"><a href="deletions-and-targeted-panel.html#cb142-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-18"><a href="deletions-and-targeted-panel.html#cb142-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb142-19"><a href="deletions-and-targeted-panel.html#cb142-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(muts_vaf, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> mean_vaf, <span class="at">label =</span> GeneName, <span class="at">color =</span> count)) <span class="sc">+</span></span>
<span id="cb142-20"><a href="deletions-and-targeted-panel.html#cb142-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb142-21"><a href="deletions-and-targeted-panel.html#cb142-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">position =</span> <span class="fu">position_quasirandom</span>()) <span class="sc">+</span></span>
<span id="cb142-22"><a href="deletions-and-targeted-panel.html#cb142-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">name =</span> <span class="st">&quot;Number of sections</span><span class="sc">\n</span><span class="st">with gene mutated&quot;</span>, <span class="at">option =</span> <span class="st">&quot;D&quot;</span>) <span class="sc">+</span></span>
<span id="cb142-23"><a href="deletions-and-targeted-panel.html#cb142-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Mean VAF&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">color =</span> <span class="st">&quot;Number of sections</span><span class="sc">\n</span><span class="st">with gene mutated&quot;</span>) <span class="sc">+</span></span>
<span id="cb142-24"><a href="deletions-and-targeted-panel.html#cb142-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/VAF%20distribution%20P6-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="monster-cells.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
