<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>17 Monster cells | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</title>
  <meta name="description" content="17 Monster cells | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="17 Monster cells | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="17 Monster cells | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  
  
  

<meta name="author" content="Luuk Harbers" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="individual-subclone-distribution-in-prostate.html"/>
<link rel="next" href="deletions-and-targeted-panel.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>4</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>4.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="4.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>4.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="4.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>4.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>5</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>5.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="5.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>5.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="5.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>5.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="5.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>5.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="5.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>5.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>6</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>6.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="6.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>6.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="6.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>6.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="6.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>6.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>7</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="7.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>7.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="7.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>7.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="7.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>7.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="7.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>7.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="7.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>7.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="7.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>7.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="7.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>7.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="7.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>7.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>8</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="8.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>8.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="8.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>8.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="8.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>8.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="8.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>8.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>9</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>9.1</b> Loading in data</a></li>
<li class="chapter" data-level="9.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>9.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="9.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>9.3</b> Training the model</a></li>
<li class="chapter" data-level="9.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>9.4</b> Validating the model</a></li>
<li class="chapter" data-level="9.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>9.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>10</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="10.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>10.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>11</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="11.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>11.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="11.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>11.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>12</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="12.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>12.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>13</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>13.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>14</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="14.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>14.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="14.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>14.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>16</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="16.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>16.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>17</b> Monster cells</a>
<ul>
<li class="chapter" data-level="17.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>17.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="17.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>17.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="17.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>17.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html"><i class="fa fa-check"></i><b>18</b> Deletions and targeted panel</a>
<ul>
<li class="chapter" data-level="18.1" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>18.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="18.2" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>18.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a>
<ul>
<li class="chapter" data-level="19.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>19.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="monster-cells" class="section level1 hasAnchor" number="17">
<h1><span class="header-section-number">17</span> Monster cells<a href="monster-cells.html#monster-cells" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This sections produces all the figures used in Supplementary Figure 16 and 17.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="monster-cells.html#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb132-2"><a href="monster-cells.html#cb132-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span>
<span id="cb132-3"><a href="monster-cells.html#cb132-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-4"><a href="monster-cells.html#cb132-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load functions</span></span>
<span id="cb132-5"><a href="monster-cells.html#cb132-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotHeatmap.R&quot;</span>)</span></code></pre></div>
<div id="clustering-copynumber-profiles" class="section level2 hasAnchor" number="17.1">
<h2><span class="header-section-number">17.1</span> Clustering copynumber profiles<a href="monster-cells.html#clustering-copynumber-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First we cluster the copynumber profiles of the previously identified monster cells. Note that the results might slightly differ from the plots in the manuscript due to random seeds being different.</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="monster-cells.html#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb133-2"><a href="monster-cells.html#cb133-2" aria-hidden="true" tabindex="-1"></a>annot_p3 <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>, <span class="at">header =</span> F, </span>
<span id="cb133-3"><a href="monster-cells.html#cb133-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;pathology&quot;</span>))</span>
<span id="cb133-4"><a href="monster-cells.html#cb133-4" aria-hidden="true" tabindex="-1"></a>profiles_p3 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_monsters_500kb.rds&quot;</span>)</span>
<span id="cb133-5"><a href="monster-cells.html#cb133-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-6"><a href="monster-cells.html#cb133-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Run UMAP</span></span>
<span id="cb133-7"><a href="monster-cells.html#cb133-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">678</span>) <span class="co">#678</span></span>
<span id="cb133-8"><a href="monster-cells.html#cb133-8" aria-hidden="true" tabindex="-1"></a>umap_res_p3 <span class="ot">=</span> <span class="fu">umap</span>(<span class="fu">t</span>(profiles_p3[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(profiles_p3)]), </span>
<span id="cb133-9"><a href="monster-cells.html#cb133-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">metric =</span> <span class="st">&quot;manhattan&quot;</span>, </span>
<span id="cb133-10"><a href="monster-cells.html#cb133-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">min_dist =</span> <span class="dv">0</span>, </span>
<span id="cb133-11"><a href="monster-cells.html#cb133-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">n_neighbors =</span> <span class="dv">3</span>,</span>
<span id="cb133-12"><a href="monster-cells.html#cb133-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">spread =</span> <span class="dv">1</span>,</span>
<span id="cb133-13"><a href="monster-cells.html#cb133-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">n_components =</span> <span class="dv">2</span>)</span>
<span id="cb133-14"><a href="monster-cells.html#cb133-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-15"><a href="monster-cells.html#cb133-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform into dt</span></span>
<span id="cb133-16"><a href="monster-cells.html#cb133-16" aria-hidden="true" tabindex="-1"></a>umap_dt_p3 <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">x =</span> umap_res_p3[, <span class="dv">1</span>],</span>
<span id="cb133-17"><a href="monster-cells.html#cb133-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y =</span> umap_res_p3[, <span class="dv">2</span>],</span>
<span id="cb133-18"><a href="monster-cells.html#cb133-18" aria-hidden="true" tabindex="-1"></a>                     <span class="at">cell =</span> <span class="fu">as.character</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(umap_res_p3)),</span>
<span id="cb133-19"><a href="monster-cells.html#cb133-19" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sample =</span> <span class="fu">colnames</span>(profiles_p3[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(profiles_p3)]))</span>
<span id="cb133-20"><a href="monster-cells.html#cb133-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-21"><a href="monster-cells.html#cb133-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-22"><a href="monster-cells.html#cb133-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Run clustering and remove cluster 0 (= noise points) if present</span></span>
<span id="cb133-23"><a href="monster-cells.html#cb133-23" aria-hidden="true" tabindex="-1"></a>clusters_p3 <span class="ot">=</span> dbscan<span class="sc">::</span><span class="fu">hdbscan</span>(umap_dt_p3[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)], <span class="at">minPts =</span> <span class="dv">10</span>)</span>
<span id="cb133-24"><a href="monster-cells.html#cb133-24" aria-hidden="true" tabindex="-1"></a>umap_dt_p3[, cluster <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(clusters_p3<span class="sc">$</span>cluster)]</span>
<span id="cb133-25"><a href="monster-cells.html#cb133-25" aria-hidden="true" tabindex="-1"></a>umap_dt_p3 <span class="ot">=</span> umap_dt_p3[cluster <span class="sc">!=</span> <span class="st">&quot;0&quot;</span>, ]</span>
<span id="cb133-26"><a href="monster-cells.html#cb133-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-27"><a href="monster-cells.html#cb133-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(umap_dt_p3, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> cluster)) <span class="sc">+</span></span>
<span id="cb133-28"><a href="monster-cells.html#cb133-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="fu">aes</span>(<span class="at">color =</span> cluster)) <span class="sc">+</span></span>
<span id="cb133-29"><a href="monster-cells.html#cb133-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb133-30"><a href="monster-cells.html#cb133-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x  =</span> <span class="st">&quot;UMAP 1&quot;</span>, <span class="at">y =</span> <span class="st">&quot;UMAP 2&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/generate%20UMAP%20and%20cluster%20for%20P3-1.png" width="672" /></p>
<p>Run the same for P6</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="monster-cells.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb134-2"><a href="monster-cells.html#cb134-2" aria-hidden="true" tabindex="-1"></a>annot_p6 <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P6.tsv&quot;</span>, <span class="at">header =</span> F, </span>
<span id="cb134-3"><a href="monster-cells.html#cb134-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;pathology&quot;</span>))</span>
<span id="cb134-4"><a href="monster-cells.html#cb134-4" aria-hidden="true" tabindex="-1"></a>profiles_p6 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P6_monsters_500kb.rds&quot;</span>)</span>
<span id="cb134-5"><a href="monster-cells.html#cb134-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-6"><a href="monster-cells.html#cb134-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Run UMAP</span></span>
<span id="cb134-7"><a href="monster-cells.html#cb134-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">678</span>) <span class="co">#678</span></span>
<span id="cb134-8"><a href="monster-cells.html#cb134-8" aria-hidden="true" tabindex="-1"></a>umap_res_p6 <span class="ot">=</span> <span class="fu">umap</span>(<span class="fu">t</span>(profiles_p6[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(profiles_p6)]), </span>
<span id="cb134-9"><a href="monster-cells.html#cb134-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">metric =</span> <span class="st">&quot;manhattan&quot;</span>, </span>
<span id="cb134-10"><a href="monster-cells.html#cb134-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">min_dist =</span> <span class="dv">0</span>, </span>
<span id="cb134-11"><a href="monster-cells.html#cb134-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">n_neighbors =</span> <span class="dv">3</span>,</span>
<span id="cb134-12"><a href="monster-cells.html#cb134-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">spread =</span> <span class="dv">1</span>,</span>
<span id="cb134-13"><a href="monster-cells.html#cb134-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">n_components =</span> <span class="dv">2</span>)</span>
<span id="cb134-14"><a href="monster-cells.html#cb134-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-15"><a href="monster-cells.html#cb134-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform into dt</span></span>
<span id="cb134-16"><a href="monster-cells.html#cb134-16" aria-hidden="true" tabindex="-1"></a>umap_dt_p6 <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">x =</span> umap_res_p6[, <span class="dv">1</span>],</span>
<span id="cb134-17"><a href="monster-cells.html#cb134-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y =</span> umap_res_p6[, <span class="dv">2</span>],</span>
<span id="cb134-18"><a href="monster-cells.html#cb134-18" aria-hidden="true" tabindex="-1"></a>                     <span class="at">cell =</span> <span class="fu">as.character</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(umap_res_p6)),</span>
<span id="cb134-19"><a href="monster-cells.html#cb134-19" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sample =</span> <span class="fu">colnames</span>(profiles_p6[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(profiles_p6)]))</span>
<span id="cb134-20"><a href="monster-cells.html#cb134-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-21"><a href="monster-cells.html#cb134-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-22"><a href="monster-cells.html#cb134-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Run clustering and remove cluster 0 (= noise points) if present</span></span>
<span id="cb134-23"><a href="monster-cells.html#cb134-23" aria-hidden="true" tabindex="-1"></a>clusters_p6 <span class="ot">=</span> dbscan<span class="sc">::</span><span class="fu">hdbscan</span>(umap_dt_p6[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)], <span class="at">minPts =</span> <span class="dv">10</span>)</span>
<span id="cb134-24"><a href="monster-cells.html#cb134-24" aria-hidden="true" tabindex="-1"></a>umap_dt_p6[, cluster <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(clusters_p6<span class="sc">$</span>cluster)]</span>
<span id="cb134-25"><a href="monster-cells.html#cb134-25" aria-hidden="true" tabindex="-1"></a>umap_dt_p6 <span class="ot">=</span> umap_dt_p6[cluster <span class="sc">!=</span> <span class="st">&quot;0&quot;</span>, ]</span>
<span id="cb134-26"><a href="monster-cells.html#cb134-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-27"><a href="monster-cells.html#cb134-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(umap_dt_p6, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> cluster)) <span class="sc">+</span></span>
<span id="cb134-28"><a href="monster-cells.html#cb134-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="fu">aes</span>(<span class="at">color =</span> cluster)) <span class="sc">+</span></span>
<span id="cb134-29"><a href="monster-cells.html#cb134-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb134-30"><a href="monster-cells.html#cb134-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x  =</span> <span class="st">&quot;UMAP 1&quot;</span>, <span class="at">y =</span> <span class="st">&quot;UMAP 2&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/generate%20UMAP%20and%20cluster%20for%20P6-1.png" width="672" /></p>
</div>
<div id="plotting-genomewide-heatmaps" class="section level2 hasAnchor" number="17.2">
<h2><span class="header-section-number">17.2</span> Plotting genomewide heatmaps<a href="monster-cells.html#plotting-genomewide-heatmaps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now we plot, for each cluster, the genomewide heatmap to show the copynumber profiles of these cells.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="monster-cells.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co"># For P3</span></span>
<span id="cb135-2"><a href="monster-cells.html#cb135-2" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">unique</span>(umap_dt_p3<span class="sc">$</span>cluster), <span class="cf">function</span>(clust) {</span>
<span id="cb135-3"><a href="monster-cells.html#cb135-3" aria-hidden="true" tabindex="-1"></a>  bins <span class="ot">=</span> profiles_p3[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb135-4"><a href="monster-cells.html#cb135-4" aria-hidden="true" tabindex="-1"></a>  profiles <span class="ot">=</span> profiles_p3[, umap_dt_p3[cluster <span class="sc">==</span> clust, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb135-5"><a href="monster-cells.html#cb135-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotHeatmap</span>(profiles, bins, <span class="at">dendrogram =</span> <span class="cn">FALSE</span>)</span>
<span id="cb135-6"><a href="monster-cells.html#cb135-6" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb135-7"><a href="monster-cells.html#cb135-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-8"><a href="monster-cells.html#cb135-8" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(plots, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="dv">1</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20genomewide-1.png" width="1920" /></p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="monster-cells.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="co"># For P6</span></span>
<span id="cb136-2"><a href="monster-cells.html#cb136-2" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">unique</span>(umap_dt_p6<span class="sc">$</span>cluster), <span class="cf">function</span>(clust) {</span>
<span id="cb136-3"><a href="monster-cells.html#cb136-3" aria-hidden="true" tabindex="-1"></a>  bins <span class="ot">=</span> profiles_p6[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb136-4"><a href="monster-cells.html#cb136-4" aria-hidden="true" tabindex="-1"></a>  profiles <span class="ot">=</span> profiles_p6[, umap_dt_p6[cluster <span class="sc">==</span> clust, sample], with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb136-5"><a href="monster-cells.html#cb136-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotHeatmap</span>(profiles, bins, <span class="at">dendrogram =</span> <span class="cn">FALSE</span>)</span>
<span id="cb136-6"><a href="monster-cells.html#cb136-6" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb136-7"><a href="monster-cells.html#cb136-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-8"><a href="monster-cells.html#cb136-8" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(plots, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.6</span>, <span class="fl">0.1</span>, <span class="fl">0.05</span>, <span class="fl">0.05</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20genomewide-2.png" width="1920" /></p>
</div>
<div id="plot-distribution" class="section level2 hasAnchor" number="17.3">
<h2><span class="header-section-number">17.3</span> Plot distribution<a href="monster-cells.html#plot-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Finally, we plot the distribution of where these cells are located in the prostate. First we run it for P3.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="monster-cells.html#cb137-1" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> umap_dt_p3[, <span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb137-2"><a href="monster-cells.html#cb137-2" aria-hidden="true" tabindex="-1"></a>locations[, library <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb137-3"><a href="monster-cells.html#cb137-3" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> <span class="fu">merge</span>(locations, annot_p3, <span class="at">by =</span> <span class="st">&quot;library&quot;</span>)</span>
<span id="cb137-4"><a href="monster-cells.html#cb137-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-5"><a href="monster-cells.html#cb137-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Get counts and fractions</span></span>
<span id="cb137-6"><a href="monster-cells.html#cb137-6" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> locations[, .(<span class="at">count =</span> .N), by <span class="ot">=</span> .(section, cluster)]</span>
<span id="cb137-7"><a href="monster-cells.html#cb137-7" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> locations[, .(<span class="at">count =</span> count, <span class="at">fraction =</span> count <span class="sc">/</span> <span class="fu">sum</span>(count), <span class="at">section =</span> section), by <span class="ot">=</span> .(cluster)]</span>
<span id="cb137-8"><a href="monster-cells.html#cb137-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-9"><a href="monster-cells.html#cb137-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill</span></span>
<span id="cb137-10"><a href="monster-cells.html#cb137-10" aria-hidden="true" tabindex="-1"></a>locations[, x <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb137-11"><a href="monster-cells.html#cb137-11" aria-hidden="true" tabindex="-1"></a>locations[, y <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb137-12"><a href="monster-cells.html#cb137-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-13"><a href="monster-cells.html#cb137-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill NAs</span></span>
<span id="cb137-14"><a href="monster-cells.html#cb137-14" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> <span class="fu">complete</span>(locations, tidyr<span class="sc">::</span><span class="fu">expand</span>(locations, x, y))</span>
<span id="cb137-15"><a href="monster-cells.html#cb137-15" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(locations)</span>
<span id="cb137-16"><a href="monster-cells.html#cb137-16" aria-hidden="true" tabindex="-1"></a>locations[, section <span class="sc">:</span><span class="er">=</span> <span class="fu">paste0</span>(<span class="st">&quot;L&quot;</span>, x, <span class="st">&quot;C&quot;</span>, y)]</span>
<span id="cb137-17"><a href="monster-cells.html#cb137-17" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> locations[<span class="sc">!</span><span class="fu">is.na</span>(cluster), ]</span>
<span id="cb137-18"><a href="monster-cells.html#cb137-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-19"><a href="monster-cells.html#cb137-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb137-20"><a href="monster-cells.html#cb137-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(locations, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> fraction, <span class="at">label =</span> count)) <span class="sc">+</span></span>
<span id="cb137-21"><a href="monster-cells.html#cb137-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb137-22"><a href="monster-cells.html#cb137-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>() <span class="sc">+</span></span>
<span id="cb137-23"><a href="monster-cells.html#cb137-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>cluster) <span class="sc">+</span></span>
<span id="cb137-24"><a href="monster-cells.html#cb137-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;Hopeful-Monster</span><span class="sc">\n</span><span class="st">Fraction&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="at">direction =</span> <span class="dv">1</span>, <span class="at">na.value =</span> <span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb137-25"><a href="monster-cells.html#cb137-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(locations<span class="sc">$</span>y), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb137-26"><a href="monster-cells.html#cb137-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(locations<span class="sc">$</span>x), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb137-27"><a href="monster-cells.html#cb137-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(locations<span class="sc">$</span>y)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(locations<span class="sc">$</span>y))) <span class="sc">+</span> </span>
<span id="cb137-28"><a href="monster-cells.html#cb137-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(locations<span class="sc">$</span>x)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(locations<span class="sc">$</span>x))) <span class="sc">+</span></span>
<span id="cb137-29"><a href="monster-cells.html#cb137-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20distribution%20P3-1.png" width="672" /></p>
<p>Lastly, we run it for P6.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="monster-cells.html#cb138-1" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> umap_dt_p6[, <span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb138-2"><a href="monster-cells.html#cb138-2" aria-hidden="true" tabindex="-1"></a>locations[, library <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb138-3"><a href="monster-cells.html#cb138-3" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> <span class="fu">merge</span>(locations, annot_p6, <span class="at">by =</span> <span class="st">&quot;library&quot;</span>)</span>
<span id="cb138-4"><a href="monster-cells.html#cb138-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-5"><a href="monster-cells.html#cb138-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Get counts and fractions</span></span>
<span id="cb138-6"><a href="monster-cells.html#cb138-6" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> locations[, .(<span class="at">count =</span> .N), by <span class="ot">=</span> .(section, cluster)]</span>
<span id="cb138-7"><a href="monster-cells.html#cb138-7" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> locations[, .(<span class="at">count =</span> count, <span class="at">fraction =</span> count <span class="sc">/</span> <span class="fu">sum</span>(count), <span class="at">section =</span> section), by <span class="ot">=</span> .(cluster)]</span>
<span id="cb138-8"><a href="monster-cells.html#cb138-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-9"><a href="monster-cells.html#cb138-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill</span></span>
<span id="cb138-10"><a href="monster-cells.html#cb138-10" aria-hidden="true" tabindex="-1"></a>locations[, x <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb138-11"><a href="monster-cells.html#cb138-11" aria-hidden="true" tabindex="-1"></a>locations[, y <span class="sc">:</span><span class="er">=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb138-12"><a href="monster-cells.html#cb138-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-13"><a href="monster-cells.html#cb138-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill NAs</span></span>
<span id="cb138-14"><a href="monster-cells.html#cb138-14" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> <span class="fu">complete</span>(locations, tidyr<span class="sc">::</span><span class="fu">expand</span>(locations, x, y))</span>
<span id="cb138-15"><a href="monster-cells.html#cb138-15" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(locations)</span>
<span id="cb138-16"><a href="monster-cells.html#cb138-16" aria-hidden="true" tabindex="-1"></a>locations[, section <span class="sc">:</span><span class="er">=</span> <span class="fu">paste0</span>(<span class="st">&quot;L&quot;</span>, x, <span class="st">&quot;C&quot;</span>, y)]</span>
<span id="cb138-17"><a href="monster-cells.html#cb138-17" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">=</span> locations[<span class="sc">!</span><span class="fu">is.na</span>(cluster), ]</span>
<span id="cb138-18"><a href="monster-cells.html#cb138-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-19"><a href="monster-cells.html#cb138-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb138-20"><a href="monster-cells.html#cb138-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(locations, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> fraction, <span class="at">label =</span> count)) <span class="sc">+</span></span>
<span id="cb138-21"><a href="monster-cells.html#cb138-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb138-22"><a href="monster-cells.html#cb138-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>() <span class="sc">+</span></span>
<span id="cb138-23"><a href="monster-cells.html#cb138-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>cluster) <span class="sc">+</span></span>
<span id="cb138-24"><a href="monster-cells.html#cb138-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;Hopeful-Monster</span><span class="sc">\n</span><span class="st">Fraction&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="at">direction =</span> <span class="dv">1</span>, <span class="at">na.value =</span> <span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb138-25"><a href="monster-cells.html#cb138-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(locations<span class="sc">$</span>y), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb138-26"><a href="monster-cells.html#cb138-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(locations<span class="sc">$</span>x), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb138-27"><a href="monster-cells.html#cb138-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(locations<span class="sc">$</span>y)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(locations<span class="sc">$</span>y))) <span class="sc">+</span> </span>
<span id="cb138-28"><a href="monster-cells.html#cb138-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(locations<span class="sc">$</span>x)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(locations<span class="sc">$</span>x))) <span class="sc">+</span></span>
<span id="cb138-29"><a href="monster-cells.html#cb138-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20distribution%20P6-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="individual-subclone-distribution-in-prostate.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="deletions-and-targeted-panel.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
