<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Random forest classification | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</title>
  <meta name="description" content="7 Random forest classification | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Random forest classification | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Random forest classification | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"/>
<link rel="next" href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>4</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="4.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>4.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="4.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>4.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="4.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>4.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="4.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>4.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="4.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>4.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="4.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>4.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="4.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>4.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="4.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>4.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>5</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="5.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>5.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="5.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>5.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="5.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>5.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="5.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>5.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>6</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="6.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>6.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="6.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>6.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="6.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>6.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>7</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="7.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>7.1</b> Loading in data</a></li>
<li class="chapter" data-level="7.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>7.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="7.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>7.3</b> Training the model</a></li>
<li class="chapter" data-level="7.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>7.4</b> Validating the model</a></li>
<li class="chapter" data-level="7.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>7.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>8</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="8.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>8.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>9</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="9.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>9.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="9.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>9.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>10</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>10.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>11</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="11.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>11.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>12</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="12.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>12.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="12.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>12.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="12.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>12.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="12.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>12.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="12.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>12.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>13</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="13.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>13.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="13.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>13.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>14</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="14.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>14.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>16</b> Monster cells</a>
<ul>
<li class="chapter" data-level="16.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>16.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="16.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>16.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="16.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>16.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="validation-of-pseudodiploid-deletions.html"><a href="validation-of-pseudodiploid-deletions.html"><i class="fa fa-check"></i><b>17</b> Validation of pseudodiploid deletions</a></li>
<li class="chapter" data-level="18" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>18</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="18.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>18.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="18.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>18.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="18.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>18.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="18.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>18.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html"><i class="fa fa-check"></i><b>19</b> COSMIC deletions and targeted panel mutations</a>
<ul>
<li class="chapter" data-level="19.1" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>19.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="19.2" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>19.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>20</b> Session info</a>
<ul>
<li class="chapter" data-level="20.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>20.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="random-forest-classification" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Random forest classification<a href="random-forest-classification.html#random-forest-classification" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This sections produces all the figures used in Supplementary Figure 5.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="random-forest-classification.html#cb61-1" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb61-2"><a href="random-forest-classification.html#cb61-2" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span></code></pre></div>
<div id="loading-in-data" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Loading in data<a href="random-forest-classification.html#loading-in-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After annotating the (2000ish) profiles using the shiny app from the previous section, we now load in all the metrics and remove sample/library names to not include that information in the training. Do note, because we recreate the RF model here, the results might differ slightly from the results and plots shown in the manuscript.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="random-forest-classification.html#cb62-1" tabindex="-1"></a><span class="co"># Load in metrics and user classification data</span></span>
<span id="cb62-2"><a href="random-forest-classification.html#cb62-2" tabindex="-1"></a>metrics <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/RandomForest/metrics.csv&quot;</span>)</span>
<span id="cb62-3"><a href="random-forest-classification.html#cb62-3" tabindex="-1"></a></span>
<span id="cb62-4"><a href="random-forest-classification.html#cb62-4" tabindex="-1"></a><span class="co"># Set factor and remove sample and library information since we do not want to include that in the training</span></span>
<span id="cb62-5"><a href="random-forest-classification.html#cb62-5" tabindex="-1"></a>metrics[, user_quality <span class="sc">:=</span> <span class="fu">factor</span>(user_quality, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;good&quot;</span>, <span class="st">&quot;bad&quot;</span>))]</span>
<span id="cb62-6"><a href="random-forest-classification.html#cb62-6" tabindex="-1"></a>metrics <span class="ot">=</span> metrics[<span class="sc">!</span><span class="fu">is.na</span>(user_quality), <span class="sc">!</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;library&quot;</span>)]</span></code></pre></div>
</div>
<div id="obtain-training-set-and-validation-set" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Obtain training set and validation set<a href="random-forest-classification.html#obtain-training-set-and-validation-set" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We used 80% of our data to train our model on and then evaluated the RF model on the other 20% of the samples that the RF model has not encountered during training.
It is important to check that the distribution of good/bad profiles is relatively equal in training and validation set and is a reflection of the full data. If this is not the case you can rerun the sampling (you can also set a seed to ensure you obtain the same sampling results)</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="random-forest-classification.html#cb63-1" tabindex="-1"></a><span class="co"># Set size (in fraction of total) of the training set</span></span>
<span id="cb63-2"><a href="random-forest-classification.html#cb63-2" tabindex="-1"></a>size_training <span class="ot">=</span> <span class="fl">0.8</span></span>
<span id="cb63-3"><a href="random-forest-classification.html#cb63-3" tabindex="-1"></a></span>
<span id="cb63-4"><a href="random-forest-classification.html#cb63-4" tabindex="-1"></a><span class="co"># Randomly sample the training set and put the remaining samples in the validation set</span></span>
<span id="cb63-5"><a href="random-forest-classification.html#cb63-5" tabindex="-1"></a>training_set <span class="ot">=</span> metrics[<span class="fu">sample</span>(.N, <span class="fu">round</span>(<span class="fu">nrow</span>(metrics) <span class="sc">*</span> size_training))]</span>
<span id="cb63-6"><a href="random-forest-classification.html#cb63-6" tabindex="-1"></a>validation_set <span class="ot">=</span> <span class="fu">fsetdiff</span>(metrics, training_set)</span>
<span id="cb63-7"><a href="random-forest-classification.html#cb63-7" tabindex="-1"></a></span>
<span id="cb63-8"><a href="random-forest-classification.html#cb63-8" tabindex="-1"></a><span class="co"># Verify that the training and validation set have a good distribution of good/bad quality profiles (based on user annotation)</span></span>
<span id="cb63-9"><a href="random-forest-classification.html#cb63-9" tabindex="-1"></a><span class="fu">table</span>(metrics<span class="sc">$</span>user_quality)[<span class="dv">1</span>] <span class="sc">/</span> </span>
<span id="cb63-10"><a href="random-forest-classification.html#cb63-10" tabindex="-1"></a>  (<span class="fu">table</span>(metrics<span class="sc">$</span>user_quality)[<span class="dv">1</span>] <span class="sc">+</span> <span class="fu">table</span>(metrics<span class="sc">$</span>user_quality)[<span class="dv">2</span>])  <span class="co"># ratio of good vs bad</span></span></code></pre></div>
<pre><code>##      good 
## 0.3793403</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="random-forest-classification.html#cb65-1" tabindex="-1"></a><span class="fu">table</span>(training_set<span class="sc">$</span>user_quality)[<span class="dv">1</span>] <span class="sc">/</span> </span>
<span id="cb65-2"><a href="random-forest-classification.html#cb65-2" tabindex="-1"></a>  (<span class="fu">table</span>(training_set<span class="sc">$</span>user_quality)[<span class="dv">1</span>] <span class="sc">+</span> <span class="fu">table</span>(training_set<span class="sc">$</span>user_quality)[<span class="dv">2</span>])  <span class="co"># ratio of good vs bad</span></span></code></pre></div>
<pre><code>##      good 
## 0.3776451</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="random-forest-classification.html#cb67-1" tabindex="-1"></a><span class="fu">table</span>(validation_set<span class="sc">$</span>user_quality)[<span class="dv">1</span>] <span class="sc">/</span> </span>
<span id="cb67-2"><a href="random-forest-classification.html#cb67-2" tabindex="-1"></a>  (<span class="fu">table</span>(validation_set<span class="sc">$</span>user_quality)[<span class="dv">1</span>] <span class="sc">+</span> <span class="fu">table</span>(validation_set<span class="sc">$</span>user_quality)[<span class="dv">2</span>])  <span class="co"># ratio of good vs bad</span></span></code></pre></div>
<pre><code>##      good 
## 0.3861171</code></pre>
</div>
<div id="training-the-model" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Training the model<a href="random-forest-classification.html#training-the-model" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We trained the RF model with default settings since we obtained a high classification accuracy</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="random-forest-classification.html#cb69-1" tabindex="-1"></a><span class="co"># Train model</span></span>
<span id="cb69-2"><a href="random-forest-classification.html#cb69-2" tabindex="-1"></a>rf <span class="ot">=</span> <span class="fu">randomForest</span>(user_quality <span class="sc">~</span> ., </span>
<span id="cb69-3"><a href="random-forest-classification.html#cb69-3" tabindex="-1"></a>                  <span class="at">data =</span> training_set, </span>
<span id="cb69-4"><a href="random-forest-classification.html#cb69-4" tabindex="-1"></a>                  <span class="at">importance =</span> T)</span></code></pre></div>
</div>
<div id="validating-the-model" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Validating the model<a href="random-forest-classification.html#validating-the-model" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Following the training we need to validate the models performance using the validation set. First we run the model on the validation set using predict and add columns with information about the predicted quality and the actual quality. Following this we plot the Receiver operating characteristic (ROC) curve and check the Area Under the Curve (AUC). Depending on randomness (or which seed is set) the curve and AUC can vary slightly. But in our experience it has always been &gt;0.98</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="random-forest-classification.html#cb70-1" tabindex="-1"></a><span class="co"># Predict on validation set</span></span>
<span id="cb70-2"><a href="random-forest-classification.html#cb70-2" tabindex="-1"></a>prediction <span class="ot">=</span> <span class="fu">as.data.table</span>(<span class="fu">predict</span>(rf, validation_set, <span class="at">type=</span><span class="st">&quot;prob&quot;</span>))</span>
<span id="cb70-3"><a href="random-forest-classification.html#cb70-3" tabindex="-1"></a>prediction[, response <span class="sc">:=</span> <span class="fu">ifelse</span>(good <span class="sc">&gt;</span> bad, <span class="st">&quot;good&quot;</span>, <span class="st">&quot;bad&quot;</span>)]</span>
<span id="cb70-4"><a href="random-forest-classification.html#cb70-4" tabindex="-1"></a>prediction[, observation <span class="sc">:=</span> validation_set<span class="sc">$</span>user_quality]</span>
<span id="cb70-5"><a href="random-forest-classification.html#cb70-5" tabindex="-1"></a></span>
<span id="cb70-6"><a href="random-forest-classification.html#cb70-6" tabindex="-1"></a><span class="co"># Plot receiver operator characteristics curve</span></span>
<span id="cb70-7"><a href="random-forest-classification.html#cb70-7" tabindex="-1"></a>roc_curve <span class="ot">=</span> <span class="fu">roc</span>(prediction<span class="sc">$</span>observation, prediction<span class="sc">$</span>good)</span>
<span id="cb70-8"><a href="random-forest-classification.html#cb70-8" tabindex="-1"></a><span class="fu">ggroc</span>(roc_curve, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb70-9"><a href="random-forest-classification.html#cb70-9" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">y =</span> <span class="fl">0.1</span>, <span class="at">x =</span> <span class="fl">0.1</span>, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;AUC = &quot;</span>, <span class="fu">round</span>(roc_curve<span class="sc">$</span>auc[[<span class="dv">1</span>]], <span class="dv">3</span>)))</span></code></pre></div>
<p><img src="_main_files/figure-html/validating%20the%20RF%20model-1.png" width="672" /></p>
<p>Next, we wanted to see which variables (sequencing/copy number calling metrics) are the most important for the RF classification.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="random-forest-classification.html#cb71-1" tabindex="-1"></a><span class="co"># Get variable iomportance values</span></span>
<span id="cb71-2"><a href="random-forest-classification.html#cb71-2" tabindex="-1"></a>var_imps <span class="ot">=</span> <span class="fu">data.frame</span>(rf<span class="sc">$</span>importance)</span>
<span id="cb71-3"><a href="random-forest-classification.html#cb71-3" tabindex="-1"></a>var_imps<span class="sc">$</span>feature <span class="ot">=</span> <span class="fu">rownames</span>(var_imps)</span>
<span id="cb71-4"><a href="random-forest-classification.html#cb71-4" tabindex="-1"></a><span class="fu">setDT</span>(var_imps)</span>
<span id="cb71-5"><a href="random-forest-classification.html#cb71-5" tabindex="-1"></a><span class="fu">setorder</span>(var_imps, MeanDecreaseAccuracy)</span>
<span id="cb71-6"><a href="random-forest-classification.html#cb71-6" tabindex="-1"></a>var_imps[, feature <span class="sc">:=</span> <span class="fu">factor</span>(feature, <span class="at">levels =</span> feature)]</span>
<span id="cb71-7"><a href="random-forest-classification.html#cb71-7" tabindex="-1"></a></span>
<span id="cb71-8"><a href="random-forest-classification.html#cb71-8" tabindex="-1"></a><span class="co"># Plot variable importances</span></span>
<span id="cb71-9"><a href="random-forest-classification.html#cb71-9" tabindex="-1"></a><span class="fu">ggplot</span>(var_imps, <span class="fu">aes</span>(<span class="at">x =</span> MeanDecreaseAccuracy, <span class="at">y =</span> feature)) <span class="sc">+</span></span>
<span id="cb71-10"><a href="random-forest-classification.html#cb71-10" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">xend =</span> <span class="dv">0</span>, <span class="at">yend =</span> feature), <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb71-11"><a href="random-forest-classification.html#cb71-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="at">color =</span> <span class="st">&quot;orange&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-12"><a href="random-forest-classification.html#cb71-12" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Feature&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Mean decrease in accuracy&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/variable%20importance-1.png" width="672" /></p>
<p>Finally, we calculate other measures such as the F1-score, Positive and Negative Predictive Value, etc.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="random-forest-classification.html#cb72-1" tabindex="-1"></a><span class="co"># Calculate metrics using caret::confusionMatrix</span></span>
<span id="cb72-2"><a href="random-forest-classification.html#cb72-2" tabindex="-1"></a><span class="fu">confusionMatrix</span>(<span class="fu">factor</span>(prediction<span class="sc">$</span>response, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;good&quot;</span>, <span class="st">&quot;bad&quot;</span>)), <span class="fu">factor</span>(prediction<span class="sc">$</span>observation, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;good&quot;</span>, <span class="st">&quot;bad&quot;</span>)), <span class="at">mode =</span> <span class="st">&quot;everything&quot;</span>)</span></code></pre></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction good bad
##       good  170  14
##       bad     8 269
##                                           
##                Accuracy : 0.9523          
##                  95% CI : (0.9286, 0.9699)
##     No Information Rate : 0.6139          
##     P-Value [Acc &gt; NIR] : &lt;2e-16          
##                                           
##                   Kappa : 0.9             
##                                           
##  Mcnemar&#39;s Test P-Value : 0.2864          
##                                           
##             Sensitivity : 0.9551          
##             Specificity : 0.9505          
##          Pos Pred Value : 0.9239          
##          Neg Pred Value : 0.9711          
##               Precision : 0.9239          
##                  Recall : 0.9551          
##                      F1 : 0.9392          
##              Prevalence : 0.3861          
##          Detection Rate : 0.3688          
##    Detection Prevalence : 0.3991          
##       Balanced Accuracy : 0.9528          
##                                           
##        &#39;Positive&#39; Class : good            
## </code></pre>
</div>
<div id="plotting-number-of-cells-that-pass-qc" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Plotting number of cells that pass QC<a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we plot the number of cells in the prostate samples that pass this RandomForest QC. First we plot Patient 3.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="random-forest-classification.html#cb74-1" tabindex="-1"></a><span class="co"># Load all profiles</span></span>
<span id="cb74-2"><a href="random-forest-classification.html#cb74-2" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_cnv.rds&quot;</span>)</span>
<span id="cb74-3"><a href="random-forest-classification.html#cb74-3" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>,</span>
<span id="cb74-4"><a href="random-forest-classification.html#cb74-4" tabindex="-1"></a>              <span class="at">header =</span> <span class="cn">FALSE</span>, </span>
<span id="cb74-5"><a href="random-forest-classification.html#cb74-5" tabindex="-1"></a>              <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;tissue_type&quot;</span>))</span>
<span id="cb74-6"><a href="random-forest-classification.html#cb74-6" tabindex="-1"></a></span>
<span id="cb74-7"><a href="random-forest-classification.html#cb74-7" tabindex="-1"></a><span class="co"># Get number of counts per region</span></span>
<span id="cb74-8"><a href="random-forest-classification.html#cb74-8" tabindex="-1"></a>profiles<span class="sc">$</span>stats[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb74-9"><a href="random-forest-classification.html#cb74-9" tabindex="-1"></a>counts <span class="ot">=</span> profiles<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, .N, by <span class="ot">=</span> library]</span>
<span id="cb74-10"><a href="random-forest-classification.html#cb74-10" tabindex="-1"></a></span>
<span id="cb74-11"><a href="random-forest-classification.html#cb74-11" tabindex="-1"></a><span class="co"># Merge with annotation</span></span>
<span id="cb74-12"><a href="random-forest-classification.html#cb74-12" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">merge</span>(counts, annot, <span class="at">by =</span> <span class="st">&quot;library&quot;</span>)</span>
<span id="cb74-13"><a href="random-forest-classification.html#cb74-13" tabindex="-1"></a></span>
<span id="cb74-14"><a href="random-forest-classification.html#cb74-14" tabindex="-1"></a><span class="co"># Add coordinates</span></span>
<span id="cb74-15"><a href="random-forest-classification.html#cb74-15" tabindex="-1"></a>counts[, x <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb74-16"><a href="random-forest-classification.html#cb74-16" tabindex="-1"></a>counts[, y <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb74-17"><a href="random-forest-classification.html#cb74-17" tabindex="-1"></a></span>
<span id="cb74-18"><a href="random-forest-classification.html#cb74-18" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb74-19"><a href="random-forest-classification.html#cb74-19" tabindex="-1"></a><span class="fu">ggplot</span>(counts, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> N, <span class="at">label =</span> N)) <span class="sc">+</span></span>
<span id="cb74-20"><a href="random-forest-classification.html#cb74-20" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb74-21"><a href="random-forest-classification.html#cb74-21" tabindex="-1"></a>  <span class="fu">geom_text</span>() <span class="sc">+</span></span>
<span id="cb74-22"><a href="random-forest-classification.html#cb74-22" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;Number of cells</span><span class="sc">\n</span><span class="st">pass QC&quot;</span>,</span>
<span id="cb74-23"><a href="random-forest-classification.html#cb74-23" tabindex="-1"></a>                       <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>,</span>
<span id="cb74-24"><a href="random-forest-classification.html#cb74-24" tabindex="-1"></a>                       <span class="at">direction =</span> <span class="dv">1</span>, <span class="at">na.value =</span> <span class="st">&quot;grey&quot;</span>,</span>
<span id="cb74-25"><a href="random-forest-classification.html#cb74-25" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">384</span>)) <span class="sc">+</span></span>
<span id="cb74-26"><a href="random-forest-classification.html#cb74-26" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(counts<span class="sc">$</span>y), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb74-27"><a href="random-forest-classification.html#cb74-27" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(counts<span class="sc">$</span>x), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb74-28"><a href="random-forest-classification.html#cb74-28" tabindex="-1"></a>  <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(counts<span class="sc">$</span>y)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(counts<span class="sc">$</span>y))) <span class="sc">+</span> </span>
<span id="cb74-29"><a href="random-forest-classification.html#cb74-29" tabindex="-1"></a>  <span class="fu">scale_x_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(counts<span class="sc">$</span>x)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(counts<span class="sc">$</span>x))) <span class="sc">+</span></span>
<span id="cb74-30"><a href="random-forest-classification.html#cb74-30" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/load%20in%20prostate%20data%20for%20P3-1.png" width="672" /></p>
<p>Now, plot the same for Patient 6.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="random-forest-classification.html#cb75-1" tabindex="-1"></a><span class="co"># Load all profiles</span></span>
<span id="cb75-2"><a href="random-forest-classification.html#cb75-2" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P6_cnv.rds&quot;</span>)</span>
<span id="cb75-3"><a href="random-forest-classification.html#cb75-3" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P6.tsv&quot;</span>,</span>
<span id="cb75-4"><a href="random-forest-classification.html#cb75-4" tabindex="-1"></a>              <span class="at">header =</span> <span class="cn">FALSE</span>, </span>
<span id="cb75-5"><a href="random-forest-classification.html#cb75-5" tabindex="-1"></a>              <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;tissue_type&quot;</span>))</span>
<span id="cb75-6"><a href="random-forest-classification.html#cb75-6" tabindex="-1"></a></span>
<span id="cb75-7"><a href="random-forest-classification.html#cb75-7" tabindex="-1"></a><span class="co"># Get number of counts per region</span></span>
<span id="cb75-8"><a href="random-forest-classification.html#cb75-8" tabindex="-1"></a>profiles<span class="sc">$</span>stats[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb75-9"><a href="random-forest-classification.html#cb75-9" tabindex="-1"></a>counts <span class="ot">=</span> profiles<span class="sc">$</span>stats[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, .N, by <span class="ot">=</span> library]</span>
<span id="cb75-10"><a href="random-forest-classification.html#cb75-10" tabindex="-1"></a></span>
<span id="cb75-11"><a href="random-forest-classification.html#cb75-11" tabindex="-1"></a><span class="co"># Merge with annotation</span></span>
<span id="cb75-12"><a href="random-forest-classification.html#cb75-12" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">merge</span>(counts, annot, <span class="at">by =</span> <span class="st">&quot;library&quot;</span>)</span>
<span id="cb75-13"><a href="random-forest-classification.html#cb75-13" tabindex="-1"></a></span>
<span id="cb75-14"><a href="random-forest-classification.html#cb75-14" tabindex="-1"></a><span class="co"># Add coordinates</span></span>
<span id="cb75-15"><a href="random-forest-classification.html#cb75-15" tabindex="-1"></a>counts[, x <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb75-16"><a href="random-forest-classification.html#cb75-16" tabindex="-1"></a>counts[, y <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb75-17"><a href="random-forest-classification.html#cb75-17" tabindex="-1"></a></span>
<span id="cb75-18"><a href="random-forest-classification.html#cb75-18" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb75-19"><a href="random-forest-classification.html#cb75-19" tabindex="-1"></a><span class="fu">ggplot</span>(counts, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> N, <span class="at">label =</span> N)) <span class="sc">+</span></span>
<span id="cb75-20"><a href="random-forest-classification.html#cb75-20" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb75-21"><a href="random-forest-classification.html#cb75-21" tabindex="-1"></a>  <span class="fu">geom_text</span>() <span class="sc">+</span></span>
<span id="cb75-22"><a href="random-forest-classification.html#cb75-22" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;Number of cells</span><span class="sc">\n</span><span class="st">pass QC&quot;</span>,</span>
<span id="cb75-23"><a href="random-forest-classification.html#cb75-23" tabindex="-1"></a>                       <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>,</span>
<span id="cb75-24"><a href="random-forest-classification.html#cb75-24" tabindex="-1"></a>                       <span class="at">direction =</span> <span class="dv">1</span>, <span class="at">na.value =</span> <span class="st">&quot;grey&quot;</span>,</span>
<span id="cb75-25"><a href="random-forest-classification.html#cb75-25" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">384</span>)) <span class="sc">+</span></span>
<span id="cb75-26"><a href="random-forest-classification.html#cb75-26" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(counts<span class="sc">$</span>y), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb75-27"><a href="random-forest-classification.html#cb75-27" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(counts<span class="sc">$</span>x), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb75-28"><a href="random-forest-classification.html#cb75-28" tabindex="-1"></a>  <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(counts<span class="sc">$</span>y)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(counts<span class="sc">$</span>y))) <span class="sc">+</span> </span>
<span id="cb75-29"><a href="random-forest-classification.html#cb75-29" tabindex="-1"></a>  <span class="fu">scale_x_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(counts<span class="sc">$</span>x)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(counts<span class="sc">$</span>x))) <span class="sc">+</span></span>
<span id="cb75-30"><a href="random-forest-classification.html#cb75-30" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/load%20in%20prostate%20data%20for%20P6-1.png" width="672" /></p>
<p>Next, we plot the number of QC-pass cells for the ACT and scCUTseq comparison.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="random-forest-classification.html#cb76-1" tabindex="-1"></a><span class="co"># Load files</span></span>
<span id="cb76-2"><a href="random-forest-classification.html#cb76-2" tabindex="-1"></a>l9c1 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD5p_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-3"><a href="random-forest-classification.html#cb76-3" tabindex="-1"></a>l9c2 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD7p_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-4"><a href="random-forest-classification.html#cb76-4" tabindex="-1"></a>l8c1 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD4p_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-5"><a href="random-forest-classification.html#cb76-5" tabindex="-1"></a>l2c2 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD2p_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-6"><a href="random-forest-classification.html#cb76-6" tabindex="-1"></a>l8c2 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD6p_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-7"><a href="random-forest-classification.html#cb76-7" tabindex="-1"></a>l3c4 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD3p_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-8"><a href="random-forest-classification.html#cb76-8" tabindex="-1"></a></span>
<span id="cb76-9"><a href="random-forest-classification.html#cb76-9" tabindex="-1"></a>p7l3c4_cut <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD27.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-10"><a href="random-forest-classification.html#cb76-10" tabindex="-1"></a>p7l3c4_act <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/CD1p.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-11"><a href="random-forest-classification.html#cb76-11" tabindex="-1"></a></span>
<span id="cb76-12"><a href="random-forest-classification.html#cb76-12" tabindex="-1"></a><span class="co"># Add information and rbind</span></span>
<span id="cb76-13"><a href="random-forest-classification.html#cb76-13" tabindex="-1"></a>l9c1[, library <span class="sc">:=</span> <span class="st">&quot;L9C1&quot;</span>]</span>
<span id="cb76-14"><a href="random-forest-classification.html#cb76-14" tabindex="-1"></a>l9c2[, library <span class="sc">:=</span> <span class="st">&quot;L9C2&quot;</span>]</span>
<span id="cb76-15"><a href="random-forest-classification.html#cb76-15" tabindex="-1"></a>l8c1[, library <span class="sc">:=</span> <span class="st">&quot;L8C1&quot;</span>]</span>
<span id="cb76-16"><a href="random-forest-classification.html#cb76-16" tabindex="-1"></a>l2c2[, library <span class="sc">:=</span> <span class="st">&quot;L2C2&quot;</span>]</span>
<span id="cb76-17"><a href="random-forest-classification.html#cb76-17" tabindex="-1"></a>l8c2[, library <span class="sc">:=</span> <span class="st">&quot;L8C2&quot;</span>]</span>
<span id="cb76-18"><a href="random-forest-classification.html#cb76-18" tabindex="-1"></a>l3c4[, library <span class="sc">:=</span> <span class="st">&quot;L3C4&quot;</span>]</span>
<span id="cb76-19"><a href="random-forest-classification.html#cb76-19" tabindex="-1"></a>p7l3c4_act[, library <span class="sc">:=</span> <span class="st">&quot;P7L3C4&quot;</span>]</span>
<span id="cb76-20"><a href="random-forest-classification.html#cb76-20" tabindex="-1"></a>p7l3c4_cut[, library <span class="sc">:=</span> <span class="st">&quot;P7L3C4&quot;</span>]</span>
<span id="cb76-21"><a href="random-forest-classification.html#cb76-21" tabindex="-1"></a>p7l3c4_cut[, method <span class="sc">:=</span> <span class="st">&quot;scCUTseq&quot;</span>]</span>
<span id="cb76-22"><a href="random-forest-classification.html#cb76-22" tabindex="-1"></a></span>
<span id="cb76-23"><a href="random-forest-classification.html#cb76-23" tabindex="-1"></a>act <span class="ot">=</span> <span class="fu">rbindlist</span>(<span class="fu">list</span>(l9c1, l9c2, l8c1, l2c2, l8c2, l3c4, p7l3c4_act))</span>
<span id="cb76-24"><a href="random-forest-classification.html#cb76-24" tabindex="-1"></a>act[, method <span class="sc">:=</span> <span class="st">&quot;ACT&quot;</span>]</span>
<span id="cb76-25"><a href="random-forest-classification.html#cb76-25" tabindex="-1"></a></span>
<span id="cb76-26"><a href="random-forest-classification.html#cb76-26" tabindex="-1"></a><span class="co"># Load scCUTseq</span></span>
<span id="cb76-27"><a href="random-forest-classification.html#cb76-27" tabindex="-1"></a>sccutseq <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb76-28"><a href="random-forest-classification.html#cb76-28" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>, </span>
<span id="cb76-29"><a href="random-forest-classification.html#cb76-29" tabindex="-1"></a>              <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;lib&quot;</span>, <span class="st">&quot;library&quot;</span>, <span class="st">&quot;Pathology&quot;</span>))</span>
<span id="cb76-30"><a href="random-forest-classification.html#cb76-30" tabindex="-1"></a></span>
<span id="cb76-31"><a href="random-forest-classification.html#cb76-31" tabindex="-1"></a><span class="co"># Extract scCUTseq libraries</span></span>
<span id="cb76-32"><a href="random-forest-classification.html#cb76-32" tabindex="-1"></a>sccutseq[, lib <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb76-33"><a href="random-forest-classification.html#cb76-33" tabindex="-1"></a></span>
<span id="cb76-34"><a href="random-forest-classification.html#cb76-34" tabindex="-1"></a><span class="co"># Merge with annotation</span></span>
<span id="cb76-35"><a href="random-forest-classification.html#cb76-35" tabindex="-1"></a>sccutseq <span class="ot">=</span> <span class="fu">merge</span>(sccutseq, annot, <span class="at">by =</span> <span class="st">&quot;lib&quot;</span>)</span>
<span id="cb76-36"><a href="random-forest-classification.html#cb76-36" tabindex="-1"></a>sccutseq_subset <span class="ot">=</span> sccutseq[library <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;L9C1&quot;</span>, <span class="st">&quot;L9C2&quot;</span>, <span class="st">&quot;L8C1&quot;</span>, <span class="st">&quot;L2C2&quot;</span>, <span class="st">&quot;L8C2&quot;</span>, <span class="st">&quot;L3C4&quot;</span>)]</span>
<span id="cb76-37"><a href="random-forest-classification.html#cb76-37" tabindex="-1"></a>sccutseq_subset[, method <span class="sc">:=</span> <span class="st">&quot;scCUTseq&quot;</span>]</span>
<span id="cb76-38"><a href="random-forest-classification.html#cb76-38" tabindex="-1"></a></span>
<span id="cb76-39"><a href="random-forest-classification.html#cb76-39" tabindex="-1"></a><span class="co"># Combine all data</span></span>
<span id="cb76-40"><a href="random-forest-classification.html#cb76-40" tabindex="-1"></a>total <span class="ot">=</span> <span class="fu">rbindlist</span>(<span class="fu">list</span>(act[, .(classifier_prediction, library, method)],</span>
<span id="cb76-41"><a href="random-forest-classification.html#cb76-41" tabindex="-1"></a>                       sccutseq_subset[, .(classifier_prediction, library, method)],</span>
<span id="cb76-42"><a href="random-forest-classification.html#cb76-42" tabindex="-1"></a>                       p7l3c4_cut[, .(classifier_prediction, library, method)]))</span>
<span id="cb76-43"><a href="random-forest-classification.html#cb76-43" tabindex="-1"></a></span>
<span id="cb76-44"><a href="random-forest-classification.html#cb76-44" tabindex="-1"></a>counts <span class="ot">=</span> total[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, .N <span class="sc">/</span> <span class="dv">384</span>, by <span class="ot">=</span> .(library, method)]</span>
<span id="cb76-45"><a href="random-forest-classification.html#cb76-45" tabindex="-1"></a></span>
<span id="cb76-46"><a href="random-forest-classification.html#cb76-46" tabindex="-1"></a><span class="fu">ggplot</span>(counts, <span class="fu">aes</span>(<span class="at">x =</span> method, <span class="at">y =</span> V1, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb76-47"><a href="random-forest-classification.html#cb76-47" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb76-48"><a href="random-forest-classification.html#cb76-48" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb76-49"><a href="random-forest-classification.html#cb76-49" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>()) <span class="sc">+</span></span>
<span id="cb76-50"><a href="random-forest-classification.html#cb76-50" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Percentage of</span><span class="sc">\n</span><span class="st">high quality cells&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Method used&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/QC-pass%20cells%20ACT%20versus%20scCUTseq-1.png" width="672" /></p>
<p>Finally, we plot the number of QC-pass cells in different samples.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="random-forest-classification.html#cb77-1" tabindex="-1"></a><span class="co"># Prostate</span></span>
<span id="cb77-2"><a href="random-forest-classification.html#cb77-2" tabindex="-1"></a>p3 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-3"><a href="random-forest-classification.html#cb77-3" tabindex="-1"></a>p6 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P6_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-4"><a href="random-forest-classification.html#cb77-4" tabindex="-1"></a></span>
<span id="cb77-5"><a href="random-forest-classification.html#cb77-5" tabindex="-1"></a><span class="co"># BRCA</span></span>
<span id="cb77-6"><a href="random-forest-classification.html#cb77-6" tabindex="-1"></a>brca1 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/brca1_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-7"><a href="random-forest-classification.html#cb77-7" tabindex="-1"></a>brca2 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/brca2_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-8"><a href="random-forest-classification.html#cb77-8" tabindex="-1"></a></span>
<span id="cb77-9"><a href="random-forest-classification.html#cb77-9" tabindex="-1"></a><span class="co"># Brain</span></span>
<span id="cb77-10"><a href="random-forest-classification.html#cb77-10" tabindex="-1"></a>dg20 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG20_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-11"><a href="random-forest-classification.html#cb77-11" tabindex="-1"></a>dg21 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG21_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-12"><a href="random-forest-classification.html#cb77-12" tabindex="-1"></a>dg22 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG22_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-13"><a href="random-forest-classification.html#cb77-13" tabindex="-1"></a>dg33 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG33_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-14"><a href="random-forest-classification.html#cb77-14" tabindex="-1"></a>dg39 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG39_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-15"><a href="random-forest-classification.html#cb77-15" tabindex="-1"></a>dg40 <span class="ot">=</span>  <span class="fu">readRDS</span>(<span class="st">&quot;./data/DG40_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb77-16"><a href="random-forest-classification.html#cb77-16" tabindex="-1"></a></span>
<span id="cb77-17"><a href="random-forest-classification.html#cb77-17" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">library =</span> <span class="fu">c</span>(<span class="st">&quot;dg20&quot;</span>, <span class="st">&quot;dg21&quot;</span>, <span class="st">&quot;dg22&quot;</span>,</span>
<span id="cb77-18"><a href="random-forest-classification.html#cb77-18" tabindex="-1"></a>                               <span class="st">&quot;dg33&quot;</span>, <span class="st">&quot;dg39&quot;</span>, <span class="st">&quot;dg40&quot;</span>),</span>
<span id="cb77-19"><a href="random-forest-classification.html#cb77-19" tabindex="-1"></a>                   <span class="at">celltype =</span> <span class="fu">c</span>(<span class="st">&quot;neun+&quot;</span>, <span class="st">&quot;neun-&quot;</span>, <span class="st">&quot;skmuscle&quot;</span>,</span>
<span id="cb77-20"><a href="random-forest-classification.html#cb77-20" tabindex="-1"></a>                                <span class="st">&quot;neun-&quot;</span>, <span class="st">&quot;neun+&quot;</span>, <span class="st">&quot;skmuscle&quot;</span>))</span>
<span id="cb77-21"><a href="random-forest-classification.html#cb77-21" tabindex="-1"></a></span>
<span id="cb77-22"><a href="random-forest-classification.html#cb77-22" tabindex="-1"></a><span class="co"># Prepare brain samples</span></span>
<span id="cb77-23"><a href="random-forest-classification.html#cb77-23" tabindex="-1"></a>dg20[, library <span class="sc">:=</span> <span class="st">&quot;dg20&quot;</span>]</span>
<span id="cb77-24"><a href="random-forest-classification.html#cb77-24" tabindex="-1"></a>dg21[, library <span class="sc">:=</span> <span class="st">&quot;dg21&quot;</span>]</span>
<span id="cb77-25"><a href="random-forest-classification.html#cb77-25" tabindex="-1"></a>dg22[, library <span class="sc">:=</span> <span class="st">&quot;dg22&quot;</span>]</span>
<span id="cb77-26"><a href="random-forest-classification.html#cb77-26" tabindex="-1"></a>dg33[, library <span class="sc">:=</span> <span class="st">&quot;dg33&quot;</span>]</span>
<span id="cb77-27"><a href="random-forest-classification.html#cb77-27" tabindex="-1"></a>dg39[, library <span class="sc">:=</span> <span class="st">&quot;dg39&quot;</span>]</span>
<span id="cb77-28"><a href="random-forest-classification.html#cb77-28" tabindex="-1"></a>dg40[, library <span class="sc">:=</span> <span class="st">&quot;dg40&quot;</span>]</span>
<span id="cb77-29"><a href="random-forest-classification.html#cb77-29" tabindex="-1"></a></span>
<span id="cb77-30"><a href="random-forest-classification.html#cb77-30" tabindex="-1"></a>brain <span class="ot">=</span> <span class="fu">rbindlist</span>(<span class="fu">list</span>(dg20, dg21, dg22, dg33, dg39, dg40))</span>
<span id="cb77-31"><a href="random-forest-classification.html#cb77-31" tabindex="-1"></a></span>
<span id="cb77-32"><a href="random-forest-classification.html#cb77-32" tabindex="-1"></a><span class="co"># Preare brain and prostate</span></span>
<span id="cb77-33"><a href="random-forest-classification.html#cb77-33" tabindex="-1"></a>p3[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb77-34"><a href="random-forest-classification.html#cb77-34" tabindex="-1"></a>p6[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb77-35"><a href="random-forest-classification.html#cb77-35" tabindex="-1"></a></span>
<span id="cb77-36"><a href="random-forest-classification.html#cb77-36" tabindex="-1"></a>brca1[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb77-37"><a href="random-forest-classification.html#cb77-37" tabindex="-1"></a>brca2[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb77-38"><a href="random-forest-classification.html#cb77-38" tabindex="-1"></a></span>
<span id="cb77-39"><a href="random-forest-classification.html#cb77-39" tabindex="-1"></a><span class="co"># Loop through samples and get counts</span></span>
<span id="cb77-40"><a href="random-forest-classification.html#cb77-40" tabindex="-1"></a>list_samples <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;p3&quot;</span>, <span class="st">&quot;p6&quot;</span>, <span class="st">&quot;brca1&quot;</span>, <span class="st">&quot;brca2&quot;</span>,<span class="st">&quot;brain&quot;</span>)</span>
<span id="cb77-41"><a href="random-forest-classification.html#cb77-41" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">lapply</span>(list_samples, <span class="cf">function</span>(sample) {</span>
<span id="cb77-42"><a href="random-forest-classification.html#cb77-42" tabindex="-1"></a>  dt <span class="ot">=</span> <span class="fu">get</span>(sample)</span>
<span id="cb77-43"><a href="random-forest-classification.html#cb77-43" tabindex="-1"></a>  counts <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> sample,</span>
<span id="cb77-44"><a href="random-forest-classification.html#cb77-44" tabindex="-1"></a>                      dt[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, .N, <span class="at">by =</span> library])</span>
<span id="cb77-45"><a href="random-forest-classification.html#cb77-45" tabindex="-1"></a>  counts[, fraction <span class="sc">:=</span> N <span class="sc">/</span> <span class="dv">384</span>]</span>
<span id="cb77-46"><a href="random-forest-classification.html#cb77-46" tabindex="-1"></a>})</span>
<span id="cb77-47"><a href="random-forest-classification.html#cb77-47" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">rbindlist</span>(res)</span>
<span id="cb77-48"><a href="random-forest-classification.html#cb77-48" tabindex="-1"></a></span>
<span id="cb77-49"><a href="random-forest-classification.html#cb77-49" tabindex="-1"></a><span class="co"># TK6</span></span>
<span id="cb77-50"><a href="random-forest-classification.html#cb77-50" tabindex="-1"></a>tk6 <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/tk6_hq_stats.tsv&quot;</span>)</span>
<span id="cb77-51"><a href="random-forest-classification.html#cb77-51" tabindex="-1"></a></span>
<span id="cb77-52"><a href="random-forest-classification.html#cb77-52" tabindex="-1"></a>all_counts <span class="ot">=</span> <span class="fu">rbind</span>(res, tk6)</span>
<span id="cb77-53"><a href="random-forest-classification.html#cb77-53" tabindex="-1"></a></span>
<span id="cb77-54"><a href="random-forest-classification.html#cb77-54" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb77-55"><a href="random-forest-classification.html#cb77-55" tabindex="-1"></a><span class="fu">ggplot</span>(all_counts, <span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> fraction)) <span class="sc">+</span> </span>
<span id="cb77-56"><a href="random-forest-classification.html#cb77-56" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb77-57"><a href="random-forest-classification.html#cb77-57" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> .<span class="dv">25</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb77-58"><a href="random-forest-classification.html#cb77-58" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Fraction of high quality cells&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/QC-pass%20cells%20in%20different%20sequencing%20datasets-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
