<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>19 Session info | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="19 Session info | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="19 Session info | _main.knit" />
  
  
  

<meta name="author" content="Luuk Harbers" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="deletions-and-targeted-panel.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>4</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>4.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="4.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>4.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="4.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>4.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>5</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>5.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="5.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>5.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="5.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>5.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="5.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>5.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="5.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>5.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>6</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>6.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="6.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>6.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="6.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>6.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="6.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>6.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>7</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="7.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>7.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="7.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>7.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="7.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>7.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="7.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>7.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="7.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>7.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="7.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>7.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="7.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>7.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="7.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>7.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>8</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="8.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>8.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="8.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>8.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="8.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>8.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="8.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>8.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>9</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>9.1</b> Loading in data</a></li>
<li class="chapter" data-level="9.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>9.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="9.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>9.3</b> Training the model</a></li>
<li class="chapter" data-level="9.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>9.4</b> Validating the model</a></li>
<li class="chapter" data-level="9.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>9.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>10</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="10.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>10.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>11</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="11.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>11.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="11.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>11.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>12</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="12.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>12.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>13</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>13.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>14</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="14.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>14.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="14.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>14.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>16</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="16.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>16.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>17</b> Monster cells</a>
<ul>
<li class="chapter" data-level="17.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>17.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="17.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>17.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="17.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>17.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html"><i class="fa fa-check"></i><b>18</b> Deletions and targeted panel</a>
<ul>
<li class="chapter" data-level="18.1" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>18.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="18.2" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>18.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a>
<ul>
<li class="chapter" data-level="19.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>19.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-info" class="section level1 hasAnchor" number="19">
<h1><span class="header-section-number">19</span> Session info<a href="session-info.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="session-info.html#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.1 (2023-06-16)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.6 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=sv_SE.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=sv_SE.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=sv_SE.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=sv_SE.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: Europe/Stockholm
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats4    grid      stats     graphics  grDevices
## [6] utils     datasets  methods   base     
## 
## other attached packages:
##  [1] containerit_0.6.0.9004 lubridate_1.9.2       
##  [3] forcats_1.0.0          purrr_1.0.1           
##  [5] tibble_3.2.1           tidyverse_2.0.0       
##  [7] caret_6.0-94           lattice_0.21-8        
##  [9] pROC_1.18.4            randomForest_4.7-1.1  
## [11] TCGAbiolinks_2.28.3    maftools_2.16.0       
## [13] DescTools_0.99.49      uwot_0.1.16           
## [15] Matrix_1.6-0           ineq_0.2-13           
## [17] readr_2.1.4            stringr_1.5.0         
## [19] treeio_1.24.1          GenomicRanges_1.52.0  
## [21] GenomeInfoDb_1.36.1    IRanges_2.34.1        
## [23] S4Vectors_0.38.1       BiocGenerics_0.46.0   
## [25] tidyr_1.3.0            data.table_1.14.8     
## [27] ggrastr_1.0.2          Polychrome_1.5.1      
## [29] circlize_0.4.15        ComplexHeatmap_2.16.0 
## [31] ggrepel_0.9.3          ggbeeswarm_0.7.2      
## [33] ggtree_3.8.0           patchwork_1.1.2       
## [35] ggdendro_0.1.23        ggpubr_0.6.0          
## [37] ggsci_3.0.0            viridis_0.6.3         
## [39] viridisLite_0.4.2      RColorBrewer_1.1-3    
## [41] dplyr_1.1.2            cowplot_1.1.1         
## [43] ggplot2_3.4.2         
## 
## loaded via a namespace (and not attached):
##   [1] fs_1.6.2                   
##   [2] matrixStats_1.0.0          
##   [3] bitops_1.0-7               
##   [4] httr_1.4.6                 
##   [5] doParallel_1.0.17          
##   [6] tools_4.3.1                
##   [7] backports_1.4.1            
##   [8] utf8_1.2.3                 
##   [9] R6_2.5.1                   
##  [10] lazyeval_0.2.2             
##  [11] GetoptLong_1.0.5           
##  [12] withr_2.5.0                
##  [13] prettyunits_1.1.1          
##  [14] gridExtra_2.3              
##  [15] versions_0.3               
##  [16] cli_3.6.1                  
##  [17] Biobase_2.60.0             
##  [18] formatR_1.14               
##  [19] Cairo_1.6-0                
##  [20] labeling_0.4.2             
##  [21] sass_0.4.6                 
##  [22] mvtnorm_1.2-2              
##  [23] proxy_0.4-27               
##  [24] yulab.utils_0.0.6          
##  [25] dbscan_1.1-11              
##  [26] parallelly_1.36.0          
##  [27] readxl_1.4.3               
##  [28] rstudioapi_0.15.0          
##  [29] RSQLite_2.3.1              
##  [30] FNN_1.1.3.2                
##  [31] generics_0.1.3             
##  [32] gridGraphics_0.5-1         
##  [33] shape_1.4.6                
##  [34] gtools_3.9.4               
##  [35] vroom_1.6.3                
##  [36] car_3.1-2                  
##  [37] futile.logger_1.4.3        
##  [38] fansi_1.0.4                
##  [39] abind_1.4-5                
##  [40] lifecycle_1.0.3            
##  [41] scatterplot3d_0.3-44       
##  [42] yaml_2.3.7                 
##  [43] carData_3.0-5              
##  [44] SummarizedExperiment_1.30.2
##  [45] recipes_1.0.6              
##  [46] BiocFileCache_2.8.0        
##  [47] blob_1.2.4                 
##  [48] promises_1.2.0.1           
##  [49] crayon_1.5.2               
##  [50] miniUI_0.1.1.1             
##  [51] KEGGREST_1.40.0            
##  [52] pillar_1.9.0               
##  [53] knitr_1.43                 
##  [54] rjson_0.2.21               
##  [55] boot_1.3-28.1              
##  [56] gld_2.6.6                  
##  [57] TCGAbiolinksGUI.data_1.20.0
##  [58] future.apply_1.11.0        
##  [59] codetools_0.2-19           
##  [60] glue_1.6.2                 
##  [61] downloader_0.4             
##  [62] ggfun_0.1.1                
##  [63] vctrs_0.6.3                
##  [64] png_0.1-8                  
##  [65] cellranger_1.1.0           
##  [66] gtable_0.3.3               
##  [67] cachem_1.0.8               
##  [68] gower_1.0.1                
##  [69] xfun_0.39                  
##  [70] mime_0.12                  
##  [71] S4Arrays_1.0.4             
##  [72] prodlim_2023.03.31         
##  [73] survival_3.5-5             
##  [74] timeDate_4022.108          
##  [75] iterators_1.0.14           
##  [76] hardhat_1.3.0              
##  [77] lava_1.7.2.1               
##  [78] ellipsis_0.3.2             
##  [79] ipred_0.9-14               
##  [80] nlme_3.1-162               
##  [81] bit64_4.0.5                
##  [82] RcppAnnoy_0.0.21           
##  [83] progress_1.2.2             
##  [84] filelock_1.0.2             
##  [85] rprojroot_2.0.3            
##  [86] bslib_0.5.0                
##  [87] irlba_2.3.5.1              
##  [88] vipor_0.4.5                
##  [89] rpart_4.1.19               
##  [90] colorspace_2.1-0           
##  [91] DBI_1.1.3                  
##  [92] nnet_7.3-19                
##  [93] DNAcopy_1.74.1             
##  [94] Exact_3.2                  
##  [95] tidyselect_1.2.0           
##  [96] bit_4.0.5                  
##  [97] compiler_4.3.1             
##  [98] curl_5.0.1                 
##  [99] rvest_1.0.3                
## [100] expm_0.999-7               
## [101] xml2_1.3.5                 
## [102] desc_1.4.2                 
## [103] DelayedArray_0.26.6        
## [104] bookdown_0.34              
## [105] scales_1.2.1               
## [106] rappdirs_0.3.3             
## [107] digest_0.6.33              
## [108] rmarkdown_2.23             
## [109] XVector_0.40.0             
## [110] htmltools_0.5.5            
## [111] pkgconfig_2.0.3            
## [112] MatrixGenerics_1.12.2      
## [113] highr_0.10                 
## [114] dbplyr_2.3.3               
## [115] fastmap_1.1.1              
## [116] rlang_1.1.1                
## [117] GlobalOptions_0.1.2        
## [118] shiny_1.7.4.1              
## [119] farver_2.1.1               
## [120] jquerylib_0.1.4            
## [121] jsonlite_1.8.7             
## [122] ModelMetrics_1.2.2.2       
## [123] RCurl_1.98-1.12            
## [124] magrittr_2.0.3             
## [125] GenomeInfoDbData_1.2.10    
## [126] ggplotify_0.1.1            
## [127] munsell_0.5.0              
## [128] Rcpp_1.0.11                
## [129] ape_5.7-1                  
## [130] stringi_1.7.12             
## [131] rootSolve_1.8.2.3          
## [132] zlibbioc_1.46.0            
## [133] MASS_7.3-60                
## [134] plyr_1.8.8                 
## [135] shinyFiles_0.9.3           
## [136] parallel_4.3.1             
## [137] listenv_0.9.0              
## [138] lmom_2.9                   
## [139] semver_0.2.0               
## [140] Biostrings_2.68.1          
## [141] splines_4.3.1              
## [142] hms_1.1.3                  
## [143] ggsignif_0.6.4             
## [144] reshape2_1.4.4             
## [145] biomaRt_2.56.1             
## [146] futile.options_1.0.1       
## [147] XML_3.99-0.14              
## [148] evaluate_0.21              
## [149] stevedore_0.9.6            
## [150] BiocManager_1.30.21        
## [151] lambda.r_1.2.4             
## [152] httpuv_1.6.11              
## [153] tzdb_0.4.0                 
## [154] foreach_1.5.2              
## [155] future_1.33.0              
## [156] clue_0.3-64                
## [157] xtable_1.8-4               
## [158] broom_1.0.5                
## [159] e1071_1.7-13               
## [160] tidytree_0.4.2             
## [161] later_1.3.1                
## [162] rstatix_0.7.2              
## [163] class_7.3-22               
## [164] aplot_0.1.10               
## [165] memoise_2.0.1              
## [166] beeswarm_0.4.0             
## [167] AnnotationDbi_1.62.2       
## [168] cluster_2.1.4              
## [169] timechange_0.2.0           
## [170] globals_0.16.2</code></pre>
<div id="dockerfile" class="section level2 hasAnchor" number="19.1">
<h2><span class="header-section-number">19.1</span> Dockerfile<a href="session-info.html#dockerfile" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="session-info.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(containerit)</span>
<span id="cb145-2"><a href="session-info.html#cb145-2" aria-hidden="true" tabindex="-1"></a>dockerfile <span class="ot">=</span> <span class="fu">dockerfile</span>(<span class="at">from =</span> utils<span class="sc">::</span><span class="fu">sessionInfo</span>(),</span>
<span id="cb145-3"><a href="session-info.html#cb145-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">versioned_libs =</span> <span class="cn">TRUE</span>,</span>
<span id="cb145-4"><a href="session-info.html#cb145-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">versioned_packages =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## INFO [2023-07-11 16:05:12] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:12] Trying to determine system requirements for the package(s) &#39;BiocGenerics&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:13] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:13] Trying to determine system requirements for the package(s) &#39;caret&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:14] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:14] Trying to determine system requirements for the package(s) &#39;circlize&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:14] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:14] Trying to determine system requirements for the package(s) &#39;ComplexHeatmap&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:15] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:15] Trying to determine system requirements for the package(s) &#39;cowplot&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:16] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:16] Trying to determine system requirements for the package(s) &#39;data.table&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:17] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:17] Trying to determine system requirements for the package(s) &#39;DescTools&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:18] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:18] Trying to determine system requirements for the package(s) &#39;dplyr&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:19] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:19] Trying to determine system requirements for the package(s) &#39;forcats&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:19] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:20] Trying to determine system requirements for the package(s) &#39;GenomeInfoDb&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:20] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:20] Trying to determine system requirements for the package(s) &#39;GenomicRanges&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:21] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:21] Trying to determine system requirements for the package(s) &#39;ggbeeswarm&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:22] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:22] Trying to determine system requirements for the package(s) &#39;ggdendro&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:23] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:23] Trying to determine system requirements for the package(s) &#39;ggplot2&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:24] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:24] Trying to determine system requirements for the package(s) &#39;ggpubr&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:25] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:25] Trying to determine system requirements for the package(s) &#39;ggrastr&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:25] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:25] Trying to determine system requirements for the package(s) &#39;ggrepel&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:26] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:26] Trying to determine system requirements for the package(s) &#39;ggsci&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:27] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:27] Trying to determine system requirements for the package(s) &#39;ggtree&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:28] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:28] Trying to determine system requirements for the package(s) &#39;ineq&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:29] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:29] Trying to determine system requirements for the package(s) &#39;IRanges&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:30] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:30] Trying to determine system requirements for the package(s) &#39;lattice&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:31] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:31] Trying to determine system requirements for the package(s) &#39;lubridate&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:32] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:32] Trying to determine system requirements for the package(s) &#39;maftools&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:32] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:32] Trying to determine system requirements for the package(s) &#39;Matrix&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:33] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:33] Trying to determine system requirements for the package(s) &#39;patchwork&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:34] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:34] Trying to determine system requirements for the package(s) &#39;Polychrome&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:35] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:35] Trying to determine system requirements for the package(s) &#39;pROC&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:36] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:36] Trying to determine system requirements for the package(s) &#39;purrr&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:37] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:37] Trying to determine system requirements for the package(s) &#39;randomForest&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:38] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:38] Trying to determine system requirements for the package(s) &#39;RColorBrewer&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:38] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:38] Trying to determine system requirements for the package(s) &#39;readr&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:39] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:39] Trying to determine system requirements for the package(s) &#39;S4Vectors&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:40] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:40] Trying to determine system requirements for the package(s) &#39;stringr&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:41] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:41] Trying to determine system requirements for the package(s) &#39;TCGAbiolinks&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:42] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:42] Trying to determine system requirements for the package(s) &#39;tibble&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:43] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:43] Trying to determine system requirements for the package(s) &#39;tidyr&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:44] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:44] Trying to determine system requirements for the package(s) &#39;tidyverse&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:45] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:45] Trying to determine system requirements for the package(s) &#39;treeio&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:45] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:45] Trying to determine system requirements for the package(s) &#39;uwot&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:46] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:46] Trying to determine system requirements for the package(s) &#39;viridis&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:47] Going online? TRUE  ... to retrieve system dependencies (sysreq-api)
## INFO [2023-07-11 16:05:47] Trying to determine system requirements for the package(s) &#39;viridisLite&#39; from sysreqs online DB
## INFO [2023-07-11 16:05:48] Versioned packages enabled, installing &#39;versions&#39;
## INFO [2023-07-11 16:05:48] Adding versioned CRAN packages: lubridate, forcats, purrr, tibble, tidyverse, caret, lattice, pROC, randomForest, DescTools, uwot, Matrix, ineq, readr, stringr, tidyr, data.table, ggrastr, Polychrome, circlize, ggrepel, ggbeeswarm, patchwork, ggdendro, ggpubr, ggsci, viridis, viridisLite, RColorBrewer, dplyr, cowplot, ggplot2
## WARN [2023-07-11 16:05:48] Adding versioned Bioconductor packages not supported: TCGAbiolinks, maftools, treeio, GenomicRanges, GenomeInfoDb, IRanges, S4Vectors, BiocGenerics, ComplexHeatmap, ggtree
## INFO [2023-07-11 16:05:48] Adding Bioconductor packages: BiocGenerics, ComplexHeatmap, GenomeInfoDb, GenomicRanges, ggtree, IRanges, maftools, S4Vectors, TCGAbiolinks, treeio
## INFO [2023-07-11 16:05:48] Created Dockerfile-Object based on sessionInfo</code></pre>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="session-info.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(dockerfile)</span></code></pre></div>
<pre><code>## FROM rocker/r-ver:4.3.1
## LABEL maintainer=&quot;luukharbers&quot;
## RUN export DEBIAN_FRONTEND=noninteractive; apt-get -y update \
##   &amp;&amp; apt-get install -y zlib1g-dev
## RUN [&quot;install2.r&quot;, &quot;versions&quot;]
## RUN [&quot;Rscript&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;caret&#39;, &#39;6.0-94&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;circlize&#39;, &#39;0.4.15&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;cowplot&#39;, &#39;1.1.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;data.table&#39;, &#39;1.14.8&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;DescTools&#39;, &#39;0.99.49&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;dplyr&#39;, &#39;1.1.2&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;forcats&#39;, &#39;1.0.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggbeeswarm&#39;, &#39;0.7.2&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggdendro&#39;, &#39;0.1.23&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggplot2&#39;, &#39;3.4.2&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggpubr&#39;, &#39;0.6.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggrastr&#39;, &#39;1.0.2&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggrepel&#39;, &#39;0.9.3&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ggsci&#39;, &#39;3.0.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;ineq&#39;, &#39;0.2-13&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;lattice&#39;, &#39;0.21-8&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;lubridate&#39;, &#39;1.9.2&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;Matrix&#39;, &#39;1.6-0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;patchwork&#39;, &#39;1.1.2&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;Polychrome&#39;, &#39;1.5.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;pROC&#39;, &#39;1.18.4&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;purrr&#39;, &#39;1.0.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;randomForest&#39;, &#39;4.7-1.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;RColorBrewer&#39;, &#39;1.1-3&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;readr&#39;, &#39;2.1.4&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;stringr&#39;, &#39;1.5.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;tibble&#39;, &#39;3.2.1&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;tidyr&#39;, &#39;1.3.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;tidyverse&#39;, &#39;2.0.0&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;uwot&#39;, &#39;0.1.16&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;viridis&#39;, &#39;0.6.3&#39;)&quot;, &quot;-e&quot;, &quot;versions::install.versions(&#39;viridisLite&#39;, &#39;0.4.2&#39;)&quot;]
## RUN [&quot;install2.r&quot;, &quot;-r https://bioconductor.org/packages/3.17/bioc -r https://bioconductor.org/packages/3.17/data/annotation -r https://bioconductor.org/packages/3.17/data/experiment -r https://bioconductor.org/packages/3.17/workflows&quot;, &quot;BiocGenerics&quot;, &quot;ComplexHeatmap&quot;, &quot;GenomeInfoDb&quot;, &quot;GenomicRanges&quot;, &quot;ggtree&quot;, &quot;IRanges&quot;, &quot;maftools&quot;, &quot;S4Vectors&quot;, &quot;TCGAbiolinks&quot;, &quot;treeio&quot;]
## WORKDIR /payload/
## CMD [&quot;R&quot;]</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="deletions-and-targeted-panel.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
