<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Technical validation of scCUTseq | scCUTseq paper</title>
  <meta name="description" content="7 Technical validation of scCUTseq | scCUTseq paper" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Technical validation of scCUTseq | scCUTseq paper" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Technical validation of scCUTseq | scCUTseq paper" />
  
  
  

<meta name="author" content="Luuk Harbers" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="trrfer-subclone-analysis.html"/>
<link rel="next" href="comparison-of-sccutseq-and-act.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>4</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>4.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="4.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>4.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="4.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>4.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>5</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>5.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="5.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>5.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="5.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>5.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="5.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>5.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="5.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>5.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>6</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>6.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="6.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>6.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="6.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>6.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="6.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>6.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>7</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="7.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>7.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="7.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>7.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="7.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>7.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="7.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>7.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="7.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>7.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="7.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>7.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="7.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>7.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="7.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>7.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>8</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="8.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>8.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="8.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>8.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="8.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>8.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="8.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>8.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>9</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>9.1</b> Loading in data</a></li>
<li class="chapter" data-level="9.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>9.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="9.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>9.3</b> Training the model</a></li>
<li class="chapter" data-level="9.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>9.4</b> Validating the model</a></li>
<li class="chapter" data-level="9.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>9.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>10</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="10.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>10.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>11</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="11.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>11.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="11.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>11.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>12</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="12.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>12.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>13</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>13.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>14</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="14.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>14.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="14.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>14.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>16</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="16.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>16.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>17</b> Monster cells</a>
<ul>
<li class="chapter" data-level="17.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>17.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="17.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>17.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="17.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>17.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html"><i class="fa fa-check"></i><b>18</b> Deletions and targeted panel</a>
<ul>
<li class="chapter" data-level="18.1" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>18.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="18.2" data-path="deletions-and-targeted-panel.html"><a href="deletions-and-targeted-panel.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>18.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a>
<ul>
<li class="chapter" data-level="19.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>19.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scCUTseq paper</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="technical-validation-of-sccutseq" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Technical validation of scCUTseq<a href="technical-validation-of-sccutseq.html#technical-validation-of-sccutseq" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This sections produces all the figures used in Supplementary Figure 1 and 2.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="technical-validation-of-sccutseq.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb62-2"><a href="technical-validation-of-sccutseq.html#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span>
<span id="cb62-3"><a href="technical-validation-of-sccutseq.html#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="technical-validation-of-sccutseq.html#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Source plotting functions</span></span>
<span id="cb62-5"><a href="technical-validation-of-sccutseq.html#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotProfile.R&quot;</span>)</span>
<span id="cb62-6"><a href="technical-validation-of-sccutseq.html#cb62-6" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotHeatmap.R&quot;</span>)</span></code></pre></div>
<div id="duplication-rates-of-cutseq-vs-malbac" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Duplication rates of CUTseq vs MALBAC<a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Duplication rate of CUTseq performed on a single cell and MALBAC with different volume scaling.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="technical-validation-of-sccutseq.html#cb63-1" aria-hidden="true" tabindex="-1"></a>dup_rate <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/CUTseq_MALBAC_duplication_rate.rds&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="technical-validation-of-sccutseq.html#cb64-1" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> dup_rate[, .N, by <span class="ot">=</span> library]</span>
<span id="cb64-2"><a href="technical-validation-of-sccutseq.html#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(dup_rate, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> duplication)) <span class="sc">+</span></span>
<span id="cb64-3"><a href="technical-validation-of-sccutseq.html#cb64-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">fill =</span> library, <span class="at">color =</span> library)) <span class="sc">+</span> </span>
<span id="cb64-4"><a href="technical-validation-of-sccutseq.html#cb64-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> .<span class="dv">075</span>) <span class="sc">+</span></span>
<span id="cb64-5"><a href="technical-validation-of-sccutseq.html#cb64-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">begin =</span> .<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb64-6"><a href="technical-validation-of-sccutseq.html#cb64-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_d</span>(<span class="at">begin =</span> .<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb64-7"><a href="technical-validation-of-sccutseq.html#cb64-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb64-8"><a href="technical-validation-of-sccutseq.html#cb64-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> counts, <span class="fu">aes</span>(<span class="at">y =</span> <span class="fl">0.1</span>, <span class="at">x =</span> library, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;n = &quot;</span>, N))) <span class="sc">+</span></span>
<span id="cb64-9"><a href="technical-validation-of-sccutseq.html#cb64-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Duplication rate&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb64-10"><a href="technical-validation-of-sccutseq.html#cb64-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20duplication%20rate%20of%20CUTseq%20versus%20scale-down%20MALBAC-1.png" width="672" /></p>
</div>
<div id="pearson-correlation-malbac-scaling" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Pearson correlation MALBAC scaling<a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Pairwise Pearson correlation between scCUTseq on SKBR3 cells using different MALBAC scalings</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="technical-validation-of-sccutseq.html#cb65-1" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/cnv_SKBR_bulk_CUTseq.rds&quot;</span>)</span>
<span id="cb65-2"><a href="technical-validation-of-sccutseq.html#cb65-2" aria-hidden="true" tabindex="-1"></a>mb50 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/cnv_MALBAC_50.rds&quot;</span>)</span>
<span id="cb65-3"><a href="technical-validation-of-sccutseq.html#cb65-3" aria-hidden="true" tabindex="-1"></a>mb100 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/cnv_MALBAC_100.rds&quot;</span>)</span>
<span id="cb65-4"><a href="technical-validation-of-sccutseq.html#cb65-4" aria-hidden="true" tabindex="-1"></a>mb200 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/cnv_MALBAC_200.rds&quot;</span>)</span>
<span id="cb65-5"><a href="technical-validation-of-sccutseq.html#cb65-5" aria-hidden="true" tabindex="-1"></a>mb500 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/cnv_MALBAC_500.rds&quot;</span>)</span>
<span id="cb65-6"><a href="technical-validation-of-sccutseq.html#cb65-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-7"><a href="technical-validation-of-sccutseq.html#cb65-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Select copynumbers</span></span>
<span id="cb65-8"><a href="technical-validation-of-sccutseq.html#cb65-8" aria-hidden="true" tabindex="-1"></a>ref_cn <span class="ot">=</span> ref<span class="sc">$</span>copynumber<span class="sc">$</span>TGATGCGC</span>
<span id="cb65-9"><a href="technical-validation-of-sccutseq.html#cb65-9" aria-hidden="true" tabindex="-1"></a>mb50_cn <span class="ot">=</span> mb50<span class="sc">$</span>copynumber</span>
<span id="cb65-10"><a href="technical-validation-of-sccutseq.html#cb65-10" aria-hidden="true" tabindex="-1"></a>mb100_cn <span class="ot">=</span> mb100<span class="sc">$</span>copynumber</span>
<span id="cb65-11"><a href="technical-validation-of-sccutseq.html#cb65-11" aria-hidden="true" tabindex="-1"></a>mb200_cn <span class="ot">=</span> mb200<span class="sc">$</span>copynumber</span>
<span id="cb65-12"><a href="technical-validation-of-sccutseq.html#cb65-12" aria-hidden="true" tabindex="-1"></a>mb500_cn <span class="ot">=</span> mb500<span class="sc">$</span>copynumber</span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="technical-validation-of-sccutseq.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pairwise correlation</span></span>
<span id="cb66-2"><a href="technical-validation-of-sccutseq.html#cb66-2" aria-hidden="true" tabindex="-1"></a>mb50_pw <span class="ot">=</span> <span class="fu">cor</span>(mb50_cn)</span>
<span id="cb66-3"><a href="technical-validation-of-sccutseq.html#cb66-3" aria-hidden="true" tabindex="-1"></a>mb50_pw <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="st">&quot;1:50&quot;</span>, </span>
<span id="cb66-4"><a href="technical-validation-of-sccutseq.html#cb66-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">V1 =</span> <span class="fu">rownames</span>(mb50_pw)[<span class="fu">row</span>(mb50_pw)[<span class="fu">upper.tri</span>(mb50_pw, <span class="at">diag =</span> F)]], </span>
<span id="cb66-5"><a href="technical-validation-of-sccutseq.html#cb66-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">V2 =</span> <span class="fu">colnames</span>(mb50_pw)[<span class="fu">col</span>(mb50_pw)[<span class="fu">upper.tri</span>(mb50_pw, <span class="at">diag =</span> F)]], </span>
<span id="cb66-6"><a href="technical-validation-of-sccutseq.html#cb66-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pearson =</span> <span class="fu">c</span>(mb50_pw[<span class="fu">upper.tri</span>(mb50_pw, <span class="at">diag =</span> F)]))</span>
<span id="cb66-7"><a href="technical-validation-of-sccutseq.html#cb66-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-8"><a href="technical-validation-of-sccutseq.html#cb66-8" aria-hidden="true" tabindex="-1"></a>mb100_pw <span class="ot">=</span> <span class="fu">cor</span>(mb100_cn)</span>
<span id="cb66-9"><a href="technical-validation-of-sccutseq.html#cb66-9" aria-hidden="true" tabindex="-1"></a>mb100_pw <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="st">&quot;1:100&quot;</span>, </span>
<span id="cb66-10"><a href="technical-validation-of-sccutseq.html#cb66-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">V1 =</span> <span class="fu">rownames</span>(mb100_pw)[<span class="fu">row</span>(mb100_pw)[<span class="fu">upper.tri</span>(mb100_pw, <span class="at">diag =</span> F)]], </span>
<span id="cb66-11"><a href="technical-validation-of-sccutseq.html#cb66-11" aria-hidden="true" tabindex="-1"></a>                      <span class="at">V2 =</span> <span class="fu">colnames</span>(mb100_pw)[<span class="fu">col</span>(mb100_pw)[<span class="fu">upper.tri</span>(mb100_pw, <span class="at">diag =</span> F)]], </span>
<span id="cb66-12"><a href="technical-validation-of-sccutseq.html#cb66-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pearson =</span> <span class="fu">c</span>(mb100_pw[<span class="fu">upper.tri</span>(mb100_pw, <span class="at">diag =</span> F)]))</span>
<span id="cb66-13"><a href="technical-validation-of-sccutseq.html#cb66-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-14"><a href="technical-validation-of-sccutseq.html#cb66-14" aria-hidden="true" tabindex="-1"></a>mb200_pw <span class="ot">=</span> <span class="fu">cor</span>(mb200_cn)</span>
<span id="cb66-15"><a href="technical-validation-of-sccutseq.html#cb66-15" aria-hidden="true" tabindex="-1"></a>mb200_pw <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="st">&quot;1:200&quot;</span>, </span>
<span id="cb66-16"><a href="technical-validation-of-sccutseq.html#cb66-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">V1 =</span> <span class="fu">rownames</span>(mb200_pw)[<span class="fu">row</span>(mb200_pw)[<span class="fu">upper.tri</span>(mb200_pw, <span class="at">diag =</span> F)]], </span>
<span id="cb66-17"><a href="technical-validation-of-sccutseq.html#cb66-17" aria-hidden="true" tabindex="-1"></a>                      <span class="at">V2 =</span> <span class="fu">colnames</span>(mb200_pw)[<span class="fu">col</span>(mb200_pw)[<span class="fu">upper.tri</span>(mb200_pw, <span class="at">diag =</span> F)]], </span>
<span id="cb66-18"><a href="technical-validation-of-sccutseq.html#cb66-18" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pearson =</span> <span class="fu">c</span>(mb200_pw[<span class="fu">upper.tri</span>(mb200_pw, <span class="at">diag =</span> F)]))</span>
<span id="cb66-19"><a href="technical-validation-of-sccutseq.html#cb66-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-20"><a href="technical-validation-of-sccutseq.html#cb66-20" aria-hidden="true" tabindex="-1"></a>mb500_pw <span class="ot">=</span> <span class="fu">cor</span>(mb500_cn)</span>
<span id="cb66-21"><a href="technical-validation-of-sccutseq.html#cb66-21" aria-hidden="true" tabindex="-1"></a>mb500_pw <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="st">&quot;1:500&quot;</span>, </span>
<span id="cb66-22"><a href="technical-validation-of-sccutseq.html#cb66-22" aria-hidden="true" tabindex="-1"></a>                     <span class="at">V1 =</span> <span class="fu">rownames</span>(mb500_pw)[<span class="fu">row</span>(mb500_pw)[<span class="fu">upper.tri</span>(mb500_pw, <span class="at">diag =</span> F)]], </span>
<span id="cb66-23"><a href="technical-validation-of-sccutseq.html#cb66-23" aria-hidden="true" tabindex="-1"></a>                     <span class="at">V2 =</span> <span class="fu">colnames</span>(mb500_pw)[<span class="fu">col</span>(mb500_pw)[<span class="fu">upper.tri</span>(mb500_pw, <span class="at">diag =</span> F)]], </span>
<span id="cb66-24"><a href="technical-validation-of-sccutseq.html#cb66-24" aria-hidden="true" tabindex="-1"></a>                     <span class="at">pearson =</span> <span class="fu">c</span>(mb500_pw[<span class="fu">upper.tri</span>(mb500_pw, <span class="at">diag =</span> F)]))</span>
<span id="cb66-25"><a href="technical-validation-of-sccutseq.html#cb66-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-26"><a href="technical-validation-of-sccutseq.html#cb66-26" aria-hidden="true" tabindex="-1"></a>res_pw <span class="ot">=</span> <span class="fu">rbindlist</span>(<span class="fu">list</span>(mb50_pw, mb100_pw, mb200_pw, mb500_pw))</span>
<span id="cb66-27"><a href="technical-validation-of-sccutseq.html#cb66-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-28"><a href="technical-validation-of-sccutseq.html#cb66-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare for plotting</span></span>
<span id="cb66-29"><a href="technical-validation-of-sccutseq.html#cb66-29" aria-hidden="true" tabindex="-1"></a>res_pw[, sample <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(sample, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;1:50&quot;</span>, <span class="st">&quot;1:100&quot;</span>, <span class="st">&quot;1:200&quot;</span>, <span class="st">&quot;1:500&quot;</span>))]</span>
<span id="cb66-30"><a href="technical-validation-of-sccutseq.html#cb66-30" aria-hidden="true" tabindex="-1"></a>obs <span class="ot">=</span> res_pw[, .N, by <span class="ot">=</span> sample]</span>
<span id="cb66-31"><a href="technical-validation-of-sccutseq.html#cb66-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-32"><a href="technical-validation-of-sccutseq.html#cb66-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb66-33"><a href="technical-validation-of-sccutseq.html#cb66-33" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res_pw, <span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> pearson)) <span class="sc">+</span></span>
<span id="cb66-34"><a href="technical-validation-of-sccutseq.html#cb66-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">fill =</span> sample)) <span class="sc">+</span></span>
<span id="cb66-35"><a href="technical-validation-of-sccutseq.html#cb66-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> .<span class="dv">04</span>, <span class="at">outlier.size =</span> .<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb66-36"><a href="technical-validation-of-sccutseq.html#cb66-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> obs, <span class="fu">aes</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;n =&quot;</span>, N))) <span class="sc">+</span></span>
<span id="cb66-37"><a href="technical-validation-of-sccutseq.html#cb66-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">begin =</span> .<span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb66-38"><a href="technical-validation-of-sccutseq.html#cb66-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, .<span class="dv">2</span>), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, .<span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb66-39"><a href="technical-validation-of-sccutseq.html#cb66-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Pairwise Pearson&#39;s Correlation&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb66-40"><a href="technical-validation-of-sccutseq.html#cb66-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/Calculating%20and%20plotting%20pairwise%20pearson%20correlation%20within%20each%20group-1.png" width="672" /></p>
<p>Pearson correlation between scCUTseq with different MALBAC scalings and bulk CUTseq</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="technical-validation-of-sccutseq.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlation against ref bulk CUTseq</span></span>
<span id="cb67-2"><a href="technical-validation-of-sccutseq.html#cb67-2" aria-hidden="true" tabindex="-1"></a>mb50_cor <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="st">&quot;1:50&quot;</span>, <span class="at">pearson =</span> <span class="fu">as.vector</span>(<span class="fu">cor</span>(mb50_cn, ref_cn)))</span>
<span id="cb67-3"><a href="technical-validation-of-sccutseq.html#cb67-3" aria-hidden="true" tabindex="-1"></a>mb100_cor <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="st">&quot;1:100&quot;</span>, <span class="at">pearson =</span> <span class="fu">as.vector</span>(<span class="fu">cor</span>(mb100_cn, ref_cn)))</span>
<span id="cb67-4"><a href="technical-validation-of-sccutseq.html#cb67-4" aria-hidden="true" tabindex="-1"></a>mb200_cor <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="st">&quot;1:200&quot;</span>, <span class="at">pearson =</span> <span class="fu">as.vector</span>(<span class="fu">cor</span>(mb200_cn, ref_cn)))</span>
<span id="cb67-5"><a href="technical-validation-of-sccutseq.html#cb67-5" aria-hidden="true" tabindex="-1"></a>mb500_cor <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="st">&quot;1:500&quot;</span>, <span class="at">pearson =</span> <span class="fu">as.vector</span>(<span class="fu">cor</span>(mb500_cn, ref_cn)))</span>
<span id="cb67-6"><a href="technical-validation-of-sccutseq.html#cb67-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-7"><a href="technical-validation-of-sccutseq.html#cb67-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">rbindlist</span>(<span class="fu">list</span>(mb50_cor, mb100_cor, mb200_cor, mb500_cor))</span>
<span id="cb67-8"><a href="technical-validation-of-sccutseq.html#cb67-8" aria-hidden="true" tabindex="-1"></a>res[, sample <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(sample, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;1:50&quot;</span>, <span class="st">&quot;1:100&quot;</span>, <span class="st">&quot;1:200&quot;</span>, <span class="st">&quot;1:500&quot;</span>))]</span>
<span id="cb67-9"><a href="technical-validation-of-sccutseq.html#cb67-9" aria-hidden="true" tabindex="-1"></a>obs <span class="ot">=</span> res[, .N, by <span class="ot">=</span> sample]</span>
<span id="cb67-10"><a href="technical-validation-of-sccutseq.html#cb67-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-11"><a href="technical-validation-of-sccutseq.html#cb67-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb67-12"><a href="technical-validation-of-sccutseq.html#cb67-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res, <span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> pearson)) <span class="sc">+</span></span>
<span id="cb67-13"><a href="technical-validation-of-sccutseq.html#cb67-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">fill =</span> sample)) <span class="sc">+</span></span>
<span id="cb67-14"><a href="technical-validation-of-sccutseq.html#cb67-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> .<span class="dv">04</span>, <span class="at">outlier.size =</span> .<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb67-15"><a href="technical-validation-of-sccutseq.html#cb67-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> obs, <span class="fu">aes</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;n =&quot;</span>, N))) <span class="sc">+</span></span>
<span id="cb67-16"><a href="technical-validation-of-sccutseq.html#cb67-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">begin =</span> .<span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb67-17"><a href="technical-validation-of-sccutseq.html#cb67-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limit =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, .<span class="dv">2</span>), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, .<span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb67-18"><a href="technical-validation-of-sccutseq.html#cb67-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Pearson&#39;s Correlation to bulk CUTseq&quot;</span>, <span class="at">x =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb67-19"><a href="technical-validation-of-sccutseq.html#cb67-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/Calculating%20and%20plotting%20pearson%20correlation%20to%20bulk%20CUTseq-1.png" width="672" /></p>
</div>
<div id="representative-sccutseq-and-smalbac-profiles-of-skbr3-cells" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Representative scCUTseq and sMALBAC profiles of SKBR3 cells<a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="technical-validation-of-sccutseq.html#cb68-1" aria-hidden="true" tabindex="-1"></a>smalbac_profile <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/smalbac_bc221.rds&quot;</span>)</span>
<span id="cb68-2"><a href="technical-validation-of-sccutseq.html#cb68-2" aria-hidden="true" tabindex="-1"></a>scCUTseq_profile <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/scCUTseq_NZ40.rds&quot;</span>)</span>
<span id="cb68-3"><a href="technical-validation-of-sccutseq.html#cb68-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-4"><a href="technical-validation-of-sccutseq.html#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the profiles</span></span>
<span id="cb68-5"><a href="technical-validation-of-sccutseq.html#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plotProfile</span>(smalbac_profile<span class="sc">$</span>segments_read[[<span class="st">&quot;NZ58&quot;</span>]], smalbac_profile<span class="sc">$</span>counts_gc[[<span class="st">&quot;NZ58&quot;</span>]], smalbac_profile<span class="sc">$</span>bins)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20example%20profiles%20of%20SKBR3%20for%20scCUTseq%20and%20sMALBAC-1.png" width="672" /></p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="technical-validation-of-sccutseq.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotProfile</span>(scCUTseq_profile<span class="sc">$</span>segments_read[[<span class="st">&quot;ACTGAGAT&quot;</span>]], scCUTseq_profile<span class="sc">$</span>counts_gc[[<span class="st">&quot;ACTGAGAT&quot;</span>]], scCUTseq_profile<span class="sc">$</span>bins)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20example%20profiles%20of%20SKBR3%20for%20scCUTseq%20and%20sMALBAC-2.png" width="672" /></p>
</div>
<div id="lorenz-curves-of-sccutseq-and-scaled-malbac" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Lorenz curves of (sc)CUTseq and (scaled) MALBAC<a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Lorenz curves of Bulk CUTseq, MALBAC (fixed), MALBAC (live), sMALBAC (fixed), sMALBAC (live) and two fixed scCUTseq cells</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="technical-validation-of-sccutseq.html#cb70-1" aria-hidden="true" tabindex="-1"></a>raw <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/lorenz-counts-500kb.rds&quot;</span>)</span>
<span id="cb70-2"><a href="technical-validation-of-sccutseq.html#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="technical-validation-of-sccutseq.html#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Give names</span></span>
<span id="cb70-4"><a href="technical-validation-of-sccutseq.html#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(raw, <span class="fu">c</span>(<span class="st">&quot;NEBNext - live 1:1&quot;</span>, <span class="st">&quot;NEBNext - fixed 1:1&quot;</span>, <span class="st">&quot;NEBNext - live 1:200&quot;</span>, </span>
<span id="cb70-5"><a href="technical-validation-of-sccutseq.html#cb70-5" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;NEBNext - fixed 1:200&quot;</span>, <span class="st">&quot;scCUTseq - Cell 1&quot;</span>, <span class="st">&quot;scCUTseq - Cell 2&quot;</span>, <span class="st">&quot;Bulk CUTseq&quot;</span>))</span>
<span id="cb70-6"><a href="technical-validation-of-sccutseq.html#cb70-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-7"><a href="technical-validation-of-sccutseq.html#cb70-7" aria-hidden="true" tabindex="-1"></a>lorenz <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">colnames</span>(raw), <span class="cf">function</span>(sample) {</span>
<span id="cb70-8"><a href="technical-validation-of-sccutseq.html#cb70-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get lorenz curve points</span></span>
<span id="cb70-9"><a href="technical-validation-of-sccutseq.html#cb70-9" aria-hidden="true" tabindex="-1"></a>  lc <span class="ot">=</span> <span class="fu">Lc</span>(raw[[sample]])</span>
<span id="cb70-10"><a href="technical-validation-of-sccutseq.html#cb70-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb70-11"><a href="technical-validation-of-sccutseq.html#cb70-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.table</span>(<span class="at">l =</span> lc<span class="sc">$</span>L, <span class="at">p =</span> lc<span class="sc">$</span>p, <span class="at">sample =</span> sample))</span>
<span id="cb70-12"><a href="technical-validation-of-sccutseq.html#cb70-12" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb70-13"><a href="technical-validation-of-sccutseq.html#cb70-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-14"><a href="technical-validation-of-sccutseq.html#cb70-14" aria-hidden="true" tabindex="-1"></a>lorenz <span class="ot">=</span> <span class="fu">rbindlist</span>(lorenz)</span>
<span id="cb70-15"><a href="technical-validation-of-sccutseq.html#cb70-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-16"><a href="technical-validation-of-sccutseq.html#cb70-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(lorenz, <span class="fu">aes</span>(<span class="at">x =</span> p, <span class="at">y =</span> l, <span class="at">color =</span> sample)) <span class="sc">+</span></span>
<span id="cb70-17"><a href="technical-validation-of-sccutseq.html#cb70-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="fl">1.25</span>) <span class="sc">+</span></span>
<span id="cb70-18"><a href="technical-validation-of-sccutseq.html#cb70-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>(<span class="fu">aes</span>(<span class="at">group =</span> sample), <span class="at">size =</span> <span class="fl">1.25</span>) <span class="sc">+</span></span>
<span id="cb70-19"><a href="technical-validation-of-sccutseq.html#cb70-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb70-20"><a href="technical-validation-of-sccutseq.html#cb70-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb70-21"><a href="technical-validation-of-sccutseq.html#cb70-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb70-22"><a href="technical-validation-of-sccutseq.html#cb70-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Cumulative fraction of total reads&quot;</span>, </span>
<span id="cb70-23"><a href="technical-validation-of-sccutseq.html#cb70-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Cumulative fraction of genome&quot;</span>, <span class="at">color =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb70-24"><a href="technical-validation-of-sccutseq.html#cb70-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_equal</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/calculate%20lorenz%20curves%20and%20plot-1.png" width="672" /></p>
</div>
<div id="pearson-correlation-of-fixed-versus-live-skbr3-cells" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Pearson correlation of fixed versus live SKBR3 cells<a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Pearson correlation of fixed and live SKBR3 libraries prepared with either scaled down (1:200) MALBAC or standard MALBAC followed by commercial library preparation.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="technical-validation-of-sccutseq.html#cb71-1" aria-hidden="true" tabindex="-1"></a>files <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;./data/technical_validation/smalbac_bc221.rds&quot;</span>,</span>
<span id="cb71-2"><a href="technical-validation-of-sccutseq.html#cb71-2" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;./data/technical_validation/malbac_bc229.rds&quot;</span>)</span>
<span id="cb71-3"><a href="technical-validation-of-sccutseq.html#cb71-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-4"><a href="technical-validation-of-sccutseq.html#cb71-4" aria-hidden="true" tabindex="-1"></a>total <span class="ot">=</span> <span class="fu">lapply</span>(files, <span class="cf">function</span>(i) {</span>
<span id="cb71-5"><a href="technical-validation-of-sccutseq.html#cb71-5" aria-hidden="true" tabindex="-1"></a>  rds <span class="ot">=</span> <span class="fu">readRDS</span>(i)</span>
<span id="cb71-6"><a href="technical-validation-of-sccutseq.html#cb71-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(rds<span class="sc">$</span>copynumber)</span>
<span id="cb71-7"><a href="technical-validation-of-sccutseq.html#cb71-7" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb71-8"><a href="technical-validation-of-sccutseq.html#cb71-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-9"><a href="technical-validation-of-sccutseq.html#cb71-9" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">do.call</span>(cbind, total)</span>
<span id="cb71-10"><a href="technical-validation-of-sccutseq.html#cb71-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-11"><a href="technical-validation-of-sccutseq.html#cb71-11" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(dt, <span class="fu">c</span>(<span class="st">&quot;MALBAC 1:200 - fixed (cell 3)&quot;</span>, <span class="st">&quot;MALBAC 1:200 - live (cell 1)&quot;</span>,</span>
<span id="cb71-12"><a href="technical-validation-of-sccutseq.html#cb71-12" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;MALBAC 1:200 - fixed (cell 1)&quot;</span>, <span class="st">&quot;MALBAC 1:200 - live (cell 2)&quot;</span>,</span>
<span id="cb71-13"><a href="technical-validation-of-sccutseq.html#cb71-13" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;MALBAC 1:200 - live (cell 3)&quot;</span>, <span class="st">&quot;MALBAC 1:200 - fixed (cell 2)&quot;</span>,</span>
<span id="cb71-14"><a href="technical-validation-of-sccutseq.html#cb71-14" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;MALBAC 1:1 - live (cell 4)&quot;</span>, <span class="st">&quot;MALBAC 1:1 - fixed (cell 4)&quot;</span>,</span>
<span id="cb71-15"><a href="technical-validation-of-sccutseq.html#cb71-15" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;MALBAC 1:1 - fixed (cell 2)&quot;</span>, <span class="st">&quot;MALBAC 1:1 - live (cell 1)&quot;</span>,</span>
<span id="cb71-16"><a href="technical-validation-of-sccutseq.html#cb71-16" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;MALBAC 1:1 - fixed (cell 3)&quot;</span>, <span class="st">&quot;MALBAC 1:1 - fixed (cell 1)&quot;</span>,</span>
<span id="cb71-17"><a href="technical-validation-of-sccutseq.html#cb71-17" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;MALBAC 1:1 - live (cell 3)&quot;</span>, <span class="st">&quot;MALBAC 1:1 - live (cell 2)&quot;</span>))</span>
<span id="cb71-18"><a href="technical-validation-of-sccutseq.html#cb71-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-19"><a href="technical-validation-of-sccutseq.html#cb71-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot correlations</span></span>
<span id="cb71-20"><a href="technical-validation-of-sccutseq.html#cb71-20" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">cor</span>(dt)</span>
<span id="cb71-21"><a href="technical-validation-of-sccutseq.html#cb71-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-22"><a href="technical-validation-of-sccutseq.html#cb71-22" aria-hidden="true" tabindex="-1"></a>res_m <span class="ot">=</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(res, <span class="at">na.rm =</span> T)</span>
<span id="cb71-23"><a href="technical-validation-of-sccutseq.html#cb71-23" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(res_m)</span>
<span id="cb71-24"><a href="technical-validation-of-sccutseq.html#cb71-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-25"><a href="technical-validation-of-sccutseq.html#cb71-25" aria-hidden="true" tabindex="-1"></a>res_m <span class="ot">=</span> res_m[<span class="fu">grepl</span>(<span class="st">&quot;live&quot;</span>, Var1) <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">&quot;fixed&quot;</span>, Var2), ]</span>
<span id="cb71-26"><a href="technical-validation-of-sccutseq.html#cb71-26" aria-hidden="true" tabindex="-1"></a>res_m[, Var1 <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(Var1, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;MALBAC 1:200 - live (cell 1)&quot;</span>, <span class="st">&quot;MALBAC 1:200 - live (cell 2)&quot;</span>,</span>
<span id="cb71-27"><a href="technical-validation-of-sccutseq.html#cb71-27" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">&quot;MALBAC 1:200 - live (cell 3)&quot;</span>, <span class="st">&quot;MALBAC 1:1 - live (cell 1)&quot;</span>,</span>
<span id="cb71-28"><a href="technical-validation-of-sccutseq.html#cb71-28" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">&quot;MALBAC 1:1 - live (cell 2)&quot;</span>, <span class="st">&quot;MALBAC 1:1 - live (cell 3)&quot;</span>,</span>
<span id="cb71-29"><a href="technical-validation-of-sccutseq.html#cb71-29" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">&quot;MALBAC 1:1 - live (cell 4)&quot;</span>))]</span>
<span id="cb71-30"><a href="technical-validation-of-sccutseq.html#cb71-30" aria-hidden="true" tabindex="-1"></a>res_m[, Var2 <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(Var2, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;MALBAC 1:200 - fixed (cell 1)&quot;</span>, <span class="st">&quot;MALBAC 1:200 - fixed (cell 2)&quot;</span>,</span>
<span id="cb71-31"><a href="technical-validation-of-sccutseq.html#cb71-31" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">&quot;MALBAC 1:200 - fixed (cell 3)&quot;</span>, <span class="st">&quot;MALBAC 1:1 - fixed (cell 1)&quot;</span>,</span>
<span id="cb71-32"><a href="technical-validation-of-sccutseq.html#cb71-32" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">&quot;MALBAC 1:1 - fixed (cell 2)&quot;</span>, <span class="st">&quot;MALBAC 1:1 - fixed (cell 3)&quot;</span>,</span>
<span id="cb71-33"><a href="technical-validation-of-sccutseq.html#cb71-33" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">&quot;MALBAC 1:1 - fixed (cell 4)&quot;</span>)))]</span>
<span id="cb71-34"><a href="technical-validation-of-sccutseq.html#cb71-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-35"><a href="technical-validation-of-sccutseq.html#cb71-35" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res_m, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb71-36"><a href="technical-validation-of-sccutseq.html#cb71-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb71-37"><a href="technical-validation-of-sccutseq.html#cb71-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span> </span>
<span id="cb71-38"><a href="technical-validation-of-sccutseq.html#cb71-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="st">&quot;&quot;</span>, <span class="at">position =</span> <span class="st">&quot;top&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-39"><a href="technical-validation-of-sccutseq.html#cb71-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(value, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb71-40"><a href="technical-validation-of-sccutseq.html#cb71-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="st">&quot;Pearson&#39;s</span><span class="sc">\n</span><span class="st">correlation&quot;</span>, <span class="at">option=</span><span class="st">&quot;B&quot;</span>, <span class="at">begin =</span> <span class="fl">0.75</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb71-41"><a href="technical-validation-of-sccutseq.html#cb71-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb71-42"><a href="technical-validation-of-sccutseq.html#cb71-42" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.line =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/scale%20down%20versus%20standard%20MALBAC%20live%20and%20fixed-1.png" width="672" /></p>
</div>
<div id="genomewide-copy-number-profiles-of-cell-lines" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> Genomewide copy number profiles of cell lines<a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Heatmap of genomewide copy number profiles of SKBR3 (live and fixed), IMR90 and MCF10A cell lines</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="technical-validation-of-sccutseq.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in cnv.rds and extract HQ profiles</span></span>
<span id="cb72-2"><a href="technical-validation-of-sccutseq.html#cb72-2" aria-hidden="true" tabindex="-1"></a>cn <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/cell_lines_heatmap.rds&quot;</span>)</span>
<span id="cb72-3"><a href="technical-validation-of-sccutseq.html#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="technical-validation-of-sccutseq.html#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get annotation</span></span>
<span id="cb72-5"><a href="technical-validation-of-sccutseq.html#cb72-5" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">samples =</span> <span class="fu">factor</span>(<span class="fu">colnames</span>(cn[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(cn)])),</span>
<span id="cb72-6"><a href="technical-validation-of-sccutseq.html#cb72-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">variable =</span> <span class="st">&quot;cell_type&quot;</span>,</span>
<span id="cb72-7"><a href="technical-validation-of-sccutseq.html#cb72-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">value =</span> <span class="fu">gsub</span>(<span class="st">&quot;-.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(cn[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(cn)])))</span>
<span id="cb72-8"><a href="technical-validation-of-sccutseq.html#cb72-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-9"><a href="technical-validation-of-sccutseq.html#cb72-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb72-10"><a href="technical-validation-of-sccutseq.html#cb72-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(cn[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(cn)], cn[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">annotation =</span> annot, <span class="at">linesize =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/genomewide%20heatmap%20of%20cell%20lines-1.png" width="1920" style="display: block; margin: auto auto auto 0;" /></p>
</div>
<div id="cross-contamination-experiment-with-dm-and-hs" class="section level2 hasAnchor" number="7.7">
<h2><span class="header-section-number">7.7</span> Cross-contamination experiment with Dm and Hs<a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="technical-validation-of-sccutseq.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in total number of reads (pre and post deduplication)</span></span>
<span id="cb73-2"><a href="technical-validation-of-sccutseq.html#cb73-2" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/technical_validation/hs_dm_counts.rds&quot;</span>)</span>
<span id="cb73-3"><a href="technical-validation-of-sccutseq.html#cb73-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-4"><a href="technical-validation-of-sccutseq.html#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Melt datatable</span></span>
<span id="cb73-5"><a href="technical-validation-of-sccutseq.html#cb73-5" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">melt</span>(counts)</span>
<span id="cb73-6"><a href="technical-validation-of-sccutseq.html#cb73-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-7"><a href="technical-validation-of-sccutseq.html#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Format for plotting</span></span>
<span id="cb73-8"><a href="technical-validation-of-sccutseq.html#cb73-8" aria-hidden="true" tabindex="-1"></a>counts[, value <span class="sc">:</span><span class="er">=</span> value <span class="sc">/</span> <span class="fl">1e6</span>]</span>
<span id="cb73-9"><a href="technical-validation-of-sccutseq.html#cb73-9" aria-hidden="true" tabindex="-1"></a>counts[, cell <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot; POS&quot;</span>, <span class="st">&quot;&quot;</span>, cell)]</span>
<span id="cb73-10"><a href="technical-validation-of-sccutseq.html#cb73-10" aria-hidden="true" tabindex="-1"></a>counts[, replicate <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*_&quot;</span>, <span class="st">&quot;&quot;</span>, variable)]</span>
<span id="cb73-11"><a href="technical-validation-of-sccutseq.html#cb73-11" aria-hidden="true" tabindex="-1"></a>counts[, variable <span class="sc">:</span><span class="er">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_rep.*&quot;</span>, <span class="st">&quot;&quot;</span>, variable)]</span>
<span id="cb73-12"><a href="technical-validation-of-sccutseq.html#cb73-12" aria-hidden="true" tabindex="-1"></a>counts[, variable <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(variable, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;all_hs&quot;</span>, <span class="st">&quot;all_dm&quot;</span>))]</span>
<span id="cb73-13"><a href="technical-validation-of-sccutseq.html#cb73-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-14"><a href="technical-validation-of-sccutseq.html#cb73-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(counts, <span class="fu">aes</span>(<span class="at">x=</span>variable, <span class="at">y=</span>value, <span class="at">color =</span> cell)) <span class="sc">+</span></span>
<span id="cb73-15"><a href="technical-validation-of-sccutseq.html#cb73-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb73-16"><a href="technical-validation-of-sccutseq.html#cb73-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_jitterdodge</span>(<span class="at">jitter.width =</span> .<span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb73-17"><a href="technical-validation-of-sccutseq.html#cb73-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>replicate) <span class="sc">+</span></span>
<span id="cb73-18"><a href="technical-validation-of-sccutseq.html#cb73-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Reads per cell (M)&quot;</span>,</span>
<span id="cb73-19"><a href="technical-validation-of-sccutseq.html#cb73-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb73-20"><a href="technical-validation-of-sccutseq.html#cb73-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb73-21"><a href="technical-validation-of-sccutseq.html#cb73-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set1&quot;</span>)) <span class="sc">+</span></span>
<span id="cb73-22"><a href="technical-validation-of-sccutseq.html#cb73-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;top&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20cross%20contamination%20DM%20and%20HS-1.png" width="672" /></p>
</div>
<div id="detection-of-crispr-induced-deletions" class="section level2 hasAnchor" number="7.8">
<h2><span class="header-section-number">7.8</span> Detection of CRISPR induced deletions<a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Select cells that have some type of alteration in the CRISPR area and then plot a zoom-in heatmap of this region.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="technical-validation-of-sccutseq.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load TK6 data</span></span>
<span id="cb74-2"><a href="technical-validation-of-sccutseq.html#cb74-2" aria-hidden="true" tabindex="-1"></a>treated <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/TK6_treated.rds&quot;</span>)</span>
<span id="cb74-3"><a href="technical-validation-of-sccutseq.html#cb74-3" aria-hidden="true" tabindex="-1"></a>untreated <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/TK6_untreated.rds&quot;</span>)</span>
<span id="cb74-4"><a href="technical-validation-of-sccutseq.html#cb74-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-5"><a href="technical-validation-of-sccutseq.html#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set bins</span></span>
<span id="cb74-6"><a href="technical-validation-of-sccutseq.html#cb74-6" aria-hidden="true" tabindex="-1"></a>bins <span class="ot">=</span> treated[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb74-7"><a href="technical-validation-of-sccutseq.html#cb74-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-8"><a href="technical-validation-of-sccutseq.html#cb74-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Set region of interest for heatmap</span></span>
<span id="cb74-9"><a href="technical-validation-of-sccutseq.html#cb74-9" aria-hidden="true" tabindex="-1"></a>roi <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">chr =</span> <span class="st">&quot;11&quot;</span>, <span class="at">start =</span> <span class="dv">118307205</span>, <span class="at">end =</span> <span class="dv">125770541</span>) <span class="co"># This is the CRISPR targeted region</span></span>
<span id="cb74-10"><a href="technical-validation-of-sccutseq.html#cb74-10" aria-hidden="true" tabindex="-1"></a>roi_bins <span class="ot">=</span> <span class="fu">foverlaps</span>(roi, bins, <span class="at">which =</span> T)<span class="sc">$</span>yid</span>
<span id="cb74-11"><a href="technical-validation-of-sccutseq.html#cb74-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-12"><a href="technical-validation-of-sccutseq.html#cb74-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Select cells that have any part of the ROI altered (cells of interest; COI)</span></span>
<span id="cb74-13"><a href="technical-validation-of-sccutseq.html#cb74-13" aria-hidden="true" tabindex="-1"></a>treated_coi <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">sapply</span>(<span class="fu">colnames</span>(treated[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(treated)]), <span class="cf">function</span>(x) {</span>
<span id="cb74-14"><a href="technical-validation-of-sccutseq.html#cb74-14" aria-hidden="true" tabindex="-1"></a>  count <span class="ot">=</span> <span class="fu">sum</span>(treated[roi_bins, ..x] <span class="sc">!=</span> <span class="dv">2</span>)</span>
<span id="cb74-15"><a href="technical-validation-of-sccutseq.html#cb74-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(count <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="fu">return</span>(x)</span>
<span id="cb74-16"><a href="technical-validation-of-sccutseq.html#cb74-16" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb74-17"><a href="technical-validation-of-sccutseq.html#cb74-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-18"><a href="technical-validation-of-sccutseq.html#cb74-18" aria-hidden="true" tabindex="-1"></a>untreated_coi <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">sapply</span>(<span class="fu">colnames</span>(untreated[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(untreated)]), <span class="cf">function</span>(x) {</span>
<span id="cb74-19"><a href="technical-validation-of-sccutseq.html#cb74-19" aria-hidden="true" tabindex="-1"></a>  count <span class="ot">=</span> <span class="fu">sum</span>(untreated[roi_bins, ..x] <span class="sc">!=</span> <span class="dv">2</span>)</span>
<span id="cb74-20"><a href="technical-validation-of-sccutseq.html#cb74-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(count <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="fu">return</span>(x)</span>
<span id="cb74-21"><a href="technical-validation-of-sccutseq.html#cb74-21" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb74-22"><a href="technical-validation-of-sccutseq.html#cb74-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-23"><a href="technical-validation-of-sccutseq.html#cb74-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Get bins of -10mb and +10 mb of gRNAs</span></span>
<span id="cb74-24"><a href="technical-validation-of-sccutseq.html#cb74-24" aria-hidden="true" tabindex="-1"></a>closest <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">chr =</span> <span class="st">&quot;11&quot;</span>, <span class="at">start =</span> <span class="fu">c</span>(roi<span class="sc">$</span>start[<span class="dv">1</span>] <span class="sc">-</span> <span class="fl">1e7</span>, roi<span class="sc">$</span>end[<span class="dv">1</span>] <span class="sc">+</span> <span class="fl">1e7</span>))</span>
<span id="cb74-25"><a href="technical-validation-of-sccutseq.html#cb74-25" aria-hidden="true" tabindex="-1"></a><span class="fu">setkey</span>(closest, chr, start)</span>
<span id="cb74-26"><a href="technical-validation-of-sccutseq.html#cb74-26" aria-hidden="true" tabindex="-1"></a>plot_window <span class="ot">=</span> bins[closest, roll <span class="ot">=</span> <span class="st">&quot;nearest&quot;</span>]</span>
<span id="cb74-27"><a href="technical-validation-of-sccutseq.html#cb74-27" aria-hidden="true" tabindex="-1"></a>zoom_index <span class="ot">=</span> <span class="fu">which</span>(bins<span class="sc">$</span>chr <span class="sc">==</span> plot_window[<span class="dv">1</span>, chr] <span class="sc">&amp;</span> bins<span class="sc">$</span>start <span class="sc">&gt;=</span> plot_window[<span class="dv">1</span>, start] <span class="sc">&amp;</span> bins<span class="sc">$</span>end <span class="sc">&lt;=</span> plot_window[<span class="dv">2</span>, end])</span>
<span id="cb74-28"><a href="technical-validation-of-sccutseq.html#cb74-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-29"><a href="technical-validation-of-sccutseq.html#cb74-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset data on both samples and genomic region</span></span>
<span id="cb74-30"><a href="technical-validation-of-sccutseq.html#cb74-30" aria-hidden="true" tabindex="-1"></a>total <span class="ot">=</span> <span class="fu">cbind</span>(treated[zoom_index, ..treated_coi], untreated[zoom_index, ..untreated_coi])</span>
<span id="cb74-31"><a href="technical-validation-of-sccutseq.html#cb74-31" aria-hidden="true" tabindex="-1"></a>bins <span class="ot">=</span> bins[zoom_index]</span>
<span id="cb74-32"><a href="technical-validation-of-sccutseq.html#cb74-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-33"><a href="technical-validation-of-sccutseq.html#cb74-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Make annotation data table</span></span>
<span id="cb74-34"><a href="technical-validation-of-sccutseq.html#cb74-34" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">sample =</span> <span class="fu">c</span>(treated_coi, untreated_coi),</span>
<span id="cb74-35"><a href="technical-validation-of-sccutseq.html#cb74-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">variable =</span> <span class="st">&quot;condition&quot;</span>,</span>
<span id="cb74-36"><a href="technical-validation-of-sccutseq.html#cb74-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">value =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;treated&quot;</span>, <span class="fu">length</span>(treated_coi)), <span class="fu">rep</span>(<span class="st">&quot;untreated&quot;</span>, <span class="fu">length</span>(untreated_coi))))</span>
<span id="cb74-37"><a href="technical-validation-of-sccutseq.html#cb74-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-38"><a href="technical-validation-of-sccutseq.html#cb74-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb74-39"><a href="technical-validation-of-sccutseq.html#cb74-39" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(total, bins, <span class="at">annotation =</span> annot, <span class="at">linesize =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/zoom%20in%20heatmap%20TK6-1.png" width="672" /></p>
<p>Plot UMAPs of chromosome 11 copy number profiles. Take note that we did not set a seed for this analysis, so the UMAP can look slightly different than the figure we show. The differences are minimal and do not change any of the conclusions we draw from these plots.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="technical-validation-of-sccutseq.html#cb75-1" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">merge</span>(treated, untreated, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>))</span>
<span id="cb75-2"><a href="technical-validation-of-sccutseq.html#cb75-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-3"><a href="technical-validation-of-sccutseq.html#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Make UMAP dataframe only using chr 11 profiles</span></span>
<span id="cb75-4"><a href="technical-validation-of-sccutseq.html#cb75-4" aria-hidden="true" tabindex="-1"></a>total_umap <span class="ot">=</span> <span class="fu">umap</span>(<span class="fu">t</span>(dt[chr <span class="sc">==</span> <span class="st">&quot;11&quot;</span>, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(dt)]), <span class="at">n_neighbors =</span> <span class="dv">24</span>, <span class="at">spread =</span> <span class="dv">1</span>, <span class="at">min_dist =</span> <span class="dv">0</span>)</span>
<span id="cb75-5"><a href="technical-validation-of-sccutseq.html#cb75-5" aria-hidden="true" tabindex="-1"></a>umap_dt <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">x =</span> total_umap[, <span class="dv">1</span>],</span>
<span id="cb75-6"><a href="technical-validation-of-sccutseq.html#cb75-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y =</span> total_umap[, <span class="dv">2</span>],</span>
<span id="cb75-7"><a href="technical-validation-of-sccutseq.html#cb75-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sample =</span> <span class="fu">colnames</span>(dt[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(dt)]),</span>
<span id="cb75-8"><a href="technical-validation-of-sccutseq.html#cb75-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="fu">ifelse</span>(<span class="fu">colnames</span>(dt[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(dt)]) <span class="sc">%in%</span> <span class="fu">colnames</span>(treated), <span class="st">&quot;Treated&quot;</span>, <span class="st">&quot;Untreated&quot;</span>))</span>
<span id="cb75-9"><a href="technical-validation-of-sccutseq.html#cb75-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-10"><a href="technical-validation-of-sccutseq.html#cb75-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot UMAP</span></span>
<span id="cb75-11"><a href="technical-validation-of-sccutseq.html#cb75-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(umap_dt, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> group)) <span class="sc">+</span></span>
<span id="cb75-12"><a href="technical-validation-of-sccutseq.html#cb75-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb75-13"><a href="technical-validation-of-sccutseq.html#cb75-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb75-14"><a href="technical-validation-of-sccutseq.html#cb75-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x  =</span> <span class="st">&quot;UMAP 1&quot;</span>, <span class="at">y =</span> <span class="st">&quot;UMAP 2&quot;</span>, <span class="at">color =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/TK6%20UMAPs-1.png" width="672" /></p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="technical-validation-of-sccutseq.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cluster UMAP using DBSCAN</span></span>
<span id="cb76-2"><a href="technical-validation-of-sccutseq.html#cb76-2" aria-hidden="true" tabindex="-1"></a>clones <span class="ot">=</span> dbscan<span class="sc">::</span><span class="fu">hdbscan</span>(umap_dt[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)],</span>
<span id="cb76-3"><a href="technical-validation-of-sccutseq.html#cb76-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">minPts =</span> <span class="dv">9</span>)</span>
<span id="cb76-4"><a href="technical-validation-of-sccutseq.html#cb76-4" aria-hidden="true" tabindex="-1"></a>umap_dt[, clone <span class="sc">:</span><span class="er">=</span> LETTERS[clones<span class="sc">$</span>cluster <span class="sc">+</span> <span class="dv">1</span>]]</span>
<span id="cb76-5"><a href="technical-validation-of-sccutseq.html#cb76-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-6"><a href="technical-validation-of-sccutseq.html#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot clusters</span></span>
<span id="cb76-7"><a href="technical-validation-of-sccutseq.html#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(umap_dt, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> clone)) <span class="sc">+</span></span>
<span id="cb76-8"><a href="technical-validation-of-sccutseq.html#cb76-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb76-9"><a href="technical-validation-of-sccutseq.html#cb76-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb76-10"><a href="technical-validation-of-sccutseq.html#cb76-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x  =</span> <span class="st">&quot;UMAP 1&quot;</span>, <span class="at">y =</span> <span class="st">&quot;UMAP 2&quot;</span>, <span class="at">color =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/TK6%20UMAPs-2.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="trrfer-subclone-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="comparison-of-sccutseq-and-act.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
