<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>18 TRR/FER subclone analysis | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</title>
  <meta name="description" content="18 TRR/FER subclone analysis | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="18 TRR/FER subclone analysis | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="18 TRR/FER subclone analysis | High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="validation-of-pseudodiploid-deletions.html"/>
<link rel="next" href="cosmic-deletions-and-targeted-panel-mutations.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="preprocessing.html"><a href="preprocessing.html"><i class="fa fa-check"></i><b>2</b> Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="preprocessing.html"><a href="preprocessing.html#demultiplexing-of-fastq-files"><i class="fa fa-check"></i><b>2.1</b> Demultiplexing of fastq files</a></li>
<li class="chapter" data-level="2.2" data-path="preprocessing.html"><a href="preprocessing.html#adapter-trimming"><i class="fa fa-check"></i><b>2.2</b> Adapter trimming</a></li>
<li class="chapter" data-level="2.3" data-path="preprocessing.html"><a href="preprocessing.html#aligning-of-cell-specific-fastq-files"><i class="fa fa-check"></i><b>2.3</b> Aligning of cell specific fastq files</a></li>
<li class="chapter" data-level="2.4" data-path="preprocessing.html"><a href="preprocessing.html#moving-barcodes-and-umis-to-bam-tags"><i class="fa fa-check"></i><b>2.4</b> Moving barcodes and UMIs to bam tags</a></li>
<li class="chapter" data-level="2.5" data-path="preprocessing.html"><a href="preprocessing.html#deduplication"><i class="fa fa-check"></i><b>2.5</b> Deduplication</a></li>
<li class="chapter" data-level="2.6" data-path="preprocessing.html"><a href="preprocessing.html#qc-summary"><i class="fa fa-check"></i><b>2.6</b> QC summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="copy-number-calling.html"><a href="copy-number-calling.html"><i class="fa fa-check"></i><b>3</b> Copy number calling</a>
<ul>
<li class="chapter" data-level="3.1" data-path="copy-number-calling.html"><a href="copy-number-calling.html#transform-bam-files"><i class="fa fa-check"></i><b>3.1</b> Transform bam files</a></li>
<li class="chapter" data-level="3.2" data-path="copy-number-calling.html"><a href="copy-number-calling.html#count-reads"><i class="fa fa-check"></i><b>3.2</b> Count reads</a></li>
<li class="chapter" data-level="3.3" data-path="copy-number-calling.html"><a href="copy-number-calling.html#calling-copy-numbers"><i class="fa fa-check"></i><b>3.3</b> Calling copy numbers</a></li>
<li class="chapter" data-level="3.4" data-path="copy-number-calling.html"><a href="copy-number-calling.html#generate-plots"><i class="fa fa-check"></i><b>3.4</b> Generate plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html"><i class="fa fa-check"></i><b>4</b> Technical validation of scCUTseq</a>
<ul>
<li class="chapter" data-level="4.1" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#duplication-rates-of-cutseq-vs-malbac"><i class="fa fa-check"></i><b>4.1</b> Duplication rates of CUTseq vs MALBAC</a></li>
<li class="chapter" data-level="4.2" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-malbac-scaling"><i class="fa fa-check"></i><b>4.2</b> Pearson correlation MALBAC scaling</a></li>
<li class="chapter" data-level="4.3" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#representative-sccutseq-and-smalbac-profiles-of-skbr3-cells"><i class="fa fa-check"></i><b>4.3</b> Representative scCUTseq and sMALBAC profiles of SKBR3 cells</a></li>
<li class="chapter" data-level="4.4" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#lorenz-curves-of-sccutseq-and-scaled-malbac"><i class="fa fa-check"></i><b>4.4</b> Lorenz curves of (sc)CUTseq and (scaled) MALBAC</a></li>
<li class="chapter" data-level="4.5" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#pearson-correlation-of-fixed-versus-live-skbr3-cells"><i class="fa fa-check"></i><b>4.5</b> Pearson correlation of fixed versus live SKBR3 cells</a></li>
<li class="chapter" data-level="4.6" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#genomewide-copy-number-profiles-of-cell-lines"><i class="fa fa-check"></i><b>4.6</b> Genomewide copy number profiles of cell lines</a></li>
<li class="chapter" data-level="4.7" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#cross-contamination-experiment-with-dm-and-hs"><i class="fa fa-check"></i><b>4.7</b> Cross-contamination experiment with Dm and Hs</a></li>
<li class="chapter" data-level="4.8" data-path="technical-validation-of-sccutseq.html"><a href="technical-validation-of-sccutseq.html#detection-of-crispr-induced-deletions"><i class="fa fa-check"></i><b>4.8</b> Detection of CRISPR induced deletions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html"><i class="fa fa-check"></i><b>5</b> Comparison of scCUTseq and ACT</a>
<ul>
<li class="chapter" data-level="5.1" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-breadth-of-coverage"><i class="fa fa-check"></i><b>5.1</b> Calculate breadth of coverage</a></li>
<li class="chapter" data-level="5.2" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#calculate-overdispersion"><i class="fa fa-check"></i><b>5.2</b> Calculate overdispersion</a></li>
<li class="chapter" data-level="5.3" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-genomewide-heatmaps-of-act-and-sccutseq-copynumber-profiles"><i class="fa fa-check"></i><b>5.3</b> Plot genomewide heatmaps of ACT and scCUTseq copynumber profiles</a></li>
<li class="chapter" data-level="5.4" data-path="comparison-of-sccutseq-and-act.html"><a href="comparison-of-sccutseq-and-act.html#plot-profile-plots-of-example-profiles"><i class="fa fa-check"></i><b>5.4</b> Plot profile plots of example profiles</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><i class="fa fa-check"></i><b>6</b> Spatially resolved single-cell CNA profiling across the prostate</a>
<ul>
<li class="chapter" data-level="6.1" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-pathologist-annotation"><i class="fa fa-check"></i><b>6.1</b> Spatial pathologist annotation</a></li>
<li class="chapter" data-level="6.2" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#spatial-distribution-of-rnaseq-clusters-scnas-and-mutations-in-prostate"><i class="fa fa-check"></i><b>6.2</b> Spatial distribution of RNAseq clusters, SCNAs and mutations in prostate</a></li>
<li class="chapter" data-level="6.3" data-path="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html"><a href="spatially-resolved-single-cell-cna-profiling-across-the-prostate.html#cell-type-distribution"><i class="fa fa-check"></i><b>6.3</b> Cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="random-forest-classification.html"><a href="random-forest-classification.html"><i class="fa fa-check"></i><b>7</b> Random forest classification</a>
<ul>
<li class="chapter" data-level="7.1" data-path="random-forest-classification.html"><a href="random-forest-classification.html#loading-in-data"><i class="fa fa-check"></i><b>7.1</b> Loading in data</a></li>
<li class="chapter" data-level="7.2" data-path="random-forest-classification.html"><a href="random-forest-classification.html#obtain-training-set-and-validation-set"><i class="fa fa-check"></i><b>7.2</b> Obtain training set and validation set</a></li>
<li class="chapter" data-level="7.3" data-path="random-forest-classification.html"><a href="random-forest-classification.html#training-the-model"><i class="fa fa-check"></i><b>7.3</b> Training the model</a></li>
<li class="chapter" data-level="7.4" data-path="random-forest-classification.html"><a href="random-forest-classification.html#validating-the-model"><i class="fa fa-check"></i><b>7.4</b> Validating the model</a></li>
<li class="chapter" data-level="7.5" data-path="random-forest-classification.html"><a href="random-forest-classification.html#plotting-number-of-cells-that-pass-qc"><i class="fa fa-check"></i><b>7.5</b> Plotting number of cells that pass QC</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><i class="fa fa-check"></i><b>8</b> Copy number profiles of brain neurons and skeletal muscle cells</a>
<ul>
<li class="chapter" data-level="8.1" data-path="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html"><a href="copy-number-profiles-of-brain-neurons-and-skeletal-muscle-cells.html#plot-genomewide-heatmaps"><i class="fa fa-check"></i><b>8.1</b> Plot genomewide heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html"><i class="fa fa-check"></i><b>9</b> Copy number profiles of other prostate cancer patients</a>
<ul>
<li class="chapter" data-level="9.1" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-genomewide-heatmap"><i class="fa fa-check"></i><b>9.1</b> Plot genomewide heatmap</a></li>
<li class="chapter" data-level="9.2" data-path="copy-number-profiles-of-other-prostate-cancer-patients.html"><a href="copy-number-profiles-of-other-prostate-cancer-patients.html#plot-cell-type-distribution"><i class="fa fa-check"></i><b>9.2</b> Plot cell type distribution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html"><i class="fa fa-check"></i><b>10</b> NEBNext versus scCUTseq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="nebnext-versus-sccutseq.html"><a href="nebnext-versus-sccutseq.html#validating-detected-deletions-using-bulk-nebnext-sequencing"><i class="fa fa-check"></i><b>10.1</b> Validating detected deletions using bulk NEBNext sequencing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html"><i class="fa fa-check"></i><b>11</b> Whole-Genome Sequencing of blood</a>
<ul>
<li class="chapter" data-level="11.1" data-path="whole-genome-sequencing-of-blood.html"><a href="whole-genome-sequencing-of-blood.html#plotting-wgs-data-for-p3-and-p6-blood"><i class="fa fa-check"></i><b>11.1</b> Plotting WGS data for P3 and P6 blood</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html"><i class="fa fa-check"></i><b>12</b> Clonal analysis pseudo-diploid cells</a>
<ul>
<li class="chapter" data-level="12.1" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#running-medicc2"><i class="fa fa-check"></i><b>12.1</b> Running MEDICC2</a></li>
<li class="chapter" data-level="12.2" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#plot-medicc2-results"><i class="fa fa-check"></i><b>12.2</b> Plot MEDICC2 results</a></li>
<li class="chapter" data-level="12.3" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#identify-tumour-subclones"><i class="fa fa-check"></i><b>12.3</b> Identify tumour subclones</a></li>
<li class="chapter" data-level="12.4" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-scnas-per-subclones"><i class="fa fa-check"></i><b>12.4</b> Calculate SCNAs per subclones</a></li>
<li class="chapter" data-level="12.5" data-path="clonal-analysis-pseudo-diploid-cells.html"><a href="clonal-analysis-pseudo-diploid-cells.html#calculate-entropy-per-subclone"><i class="fa fa-check"></i><b>12.5</b> Calculate entropy per subclone</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html"><i class="fa fa-check"></i><b>13</b> Phylogenetic analysis of breast cancer</a>
<ul>
<li class="chapter" data-level="13.1" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#performing-phylogenetic-analysis"><i class="fa fa-check"></i><b>13.1</b> Performing phylogenetic analysis</a></li>
<li class="chapter" data-level="13.2" data-path="phylogenetic-analysis-of-breast-cancer.html"><a href="phylogenetic-analysis-of-breast-cancer.html#plotting-phylogenetic-tree-and-heatmap"><i class="fa fa-check"></i><b>13.2</b> Plotting phylogenetic tree and heatmap</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html"><i class="fa fa-check"></i><b>14</b> Subclonal distribution in prostate</a>
<ul>
<li class="chapter" data-level="14.1" data-path="subclonal-distribution-in-prostate.html"><a href="subclonal-distribution-in-prostate.html#plotting-spatial-subclonal-distribution"><i class="fa fa-check"></i><b>14.1</b> Plotting spatial subclonal distribution</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html"><i class="fa fa-check"></i><b>15</b> Individual subclone distribution in prostate</a>
<ul>
<li class="chapter" data-level="15.1" data-path="individual-subclone-distribution-in-prostate.html"><a href="individual-subclone-distribution-in-prostate.html#plotting-individual-subclone-distribution"><i class="fa fa-check"></i><b>15.1</b> Plotting individual subclone distribution</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="monster-cells.html"><a href="monster-cells.html"><i class="fa fa-check"></i><b>16</b> Monster cells</a>
<ul>
<li class="chapter" data-level="16.1" data-path="monster-cells.html"><a href="monster-cells.html#clustering-copynumber-profiles"><i class="fa fa-check"></i><b>16.1</b> Clustering copynumber profiles</a></li>
<li class="chapter" data-level="16.2" data-path="monster-cells.html"><a href="monster-cells.html#plotting-genomewide-heatmaps"><i class="fa fa-check"></i><b>16.2</b> Plotting genomewide heatmaps</a></li>
<li class="chapter" data-level="16.3" data-path="monster-cells.html"><a href="monster-cells.html#plot-distribution"><i class="fa fa-check"></i><b>16.3</b> Plot distribution</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="validation-of-pseudodiploid-deletions.html"><a href="validation-of-pseudodiploid-deletions.html"><i class="fa fa-check"></i><b>17</b> Validation of pseudodiploid deletions</a></li>
<li class="chapter" data-level="18" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html"><i class="fa fa-check"></i><b>18</b> TRR/FER subclone analysis</a>
<ul>
<li class="chapter" data-level="18.1" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones"><i class="fa fa-check"></i><b>18.1</b> Plotting deleted genes from TRR/FER subclones</a></li>
<li class="chapter" data-level="18.2" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs"><i class="fa fa-check"></i><b>18.2</b> Co-deletions of TSGs</a></li>
<li class="chapter" data-level="18.3" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#tcga-deletion-analysis"><i class="fa fa-check"></i><b>18.3</b> TCGA deletion analysis</a></li>
<li class="chapter" data-level="18.4" data-path="trrfer-subclone-analysis.html"><a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations"><i class="fa fa-check"></i><b>18.4</b> Targeted sequencing mutations</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html"><i class="fa fa-check"></i><b>19</b> COSMIC deletions and targeted panel mutations</a>
<ul>
<li class="chapter" data-level="19.1" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html#trrfer-specific-subclones"><i class="fa fa-check"></i><b>19.1</b> TRR/FER specific subclones</a></li>
<li class="chapter" data-level="19.2" data-path="cosmic-deletions-and-targeted-panel-mutations.html"><a href="cosmic-deletions-and-targeted-panel-mutations.html#variant-allele-frequency-snvs"><i class="fa fa-check"></i><b>19.2</b> Variant allele frequency SNVs</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>20</b> Session info</a>
<ul>
<li class="chapter" data-level="20.1" data-path="session-info.html"><a href="session-info.html#dockerfile"><i class="fa fa-check"></i><b>20.1</b> Dockerfile</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High clonal diversity and spatial genetic admixture in early prostate cancer and surrounding normal tissue</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="trrfer-subclone-analysis" class="section level1 hasAnchor" number="18">
<h1><span class="header-section-number">18</span> TRR/FER subclone analysis<a href="trrfer-subclone-analysis.html#trrfer-subclone-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This section produces all the figures used for Figure 4.</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="trrfer-subclone-analysis.html#cb126-1" tabindex="-1"></a><span class="co"># Source setup file</span></span>
<span id="cb126-2"><a href="trrfer-subclone-analysis.html#cb126-2" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/setup.R&quot;</span>)</span>
<span id="cb126-3"><a href="trrfer-subclone-analysis.html#cb126-3" tabindex="-1"></a></span>
<span id="cb126-4"><a href="trrfer-subclone-analysis.html#cb126-4" tabindex="-1"></a><span class="co"># Load functions</span></span>
<span id="cb126-5"><a href="trrfer-subclone-analysis.html#cb126-5" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./functions/plotHeatmap.R&quot;</span>)</span></code></pre></div>
<div id="plotting-deleted-genes-from-trrfer-subclones" class="section level2 hasAnchor" number="18.1">
<h2><span class="header-section-number">18.1</span> Plotting deleted genes from TRR/FER subclones<a href="trrfer-subclone-analysis.html#plotting-deleted-genes-from-trrfer-subclones" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="trrfer-subclone-analysis.html#cb127-1" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P3_median_cn.tsv&quot;</span>)</span>
<span id="cb127-2"><a href="trrfer-subclone-analysis.html#cb127-2" tabindex="-1"></a>clones <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P3_clones.tsv&quot;</span>)</span>
<span id="cb127-3"><a href="trrfer-subclone-analysis.html#cb127-3" tabindex="-1"></a>cosmic <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/genelists/cosmic_gene-census.tsv&quot;</span>)</span>
<span id="cb127-4"><a href="trrfer-subclone-analysis.html#cb127-4" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>, <span class="at">header =</span> F)</span>
<span id="cb127-5"><a href="trrfer-subclone-analysis.html#cb127-5" tabindex="-1"></a></span>
<span id="cb127-6"><a href="trrfer-subclone-analysis.html#cb127-6" tabindex="-1"></a><span class="co"># Merge clones with annot</span></span>
<span id="cb127-7"><a href="trrfer-subclone-analysis.html#cb127-7" tabindex="-1"></a>clones[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample_id)]</span>
<span id="cb127-8"><a href="trrfer-subclone-analysis.html#cb127-8" tabindex="-1"></a>clones_annot <span class="ot">=</span> <span class="fu">merge</span>(clones, annot, <span class="at">by.x =</span> <span class="st">&quot;library&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;V1&quot;</span>)</span>
<span id="cb127-9"><a href="trrfer-subclone-analysis.html#cb127-9" tabindex="-1"></a></span>
<span id="cb127-10"><a href="trrfer-subclone-analysis.html#cb127-10" tabindex="-1"></a><span class="co"># Select clones that are only present in cancer/focal</span></span>
<span id="cb127-11"><a href="trrfer-subclone-analysis.html#cb127-11" tabindex="-1"></a>total <span class="ot">=</span> clones_annot[, .(<span class="fu">paste</span>(<span class="fu">unique</span>(V3), <span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)), by <span class="ot">=</span> cluster]</span>
<span id="cb127-12"><a href="trrfer-subclone-analysis.html#cb127-12" tabindex="-1"></a>clones_selected <span class="ot">=</span> total[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;Normal&quot;</span>, V1)]</span>
<span id="cb127-13"><a href="trrfer-subclone-analysis.html#cb127-13" tabindex="-1"></a></span>
<span id="cb127-14"><a href="trrfer-subclone-analysis.html#cb127-14" tabindex="-1"></a><span class="co"># Get selected</span></span>
<span id="cb127-15"><a href="trrfer-subclone-analysis.html#cb127-15" tabindex="-1"></a>profiles <span class="ot">=</span> profiles[cluster <span class="sc">%in%</span> clones_selected<span class="sc">$</span>cluster]</span>
<span id="cb127-16"><a href="trrfer-subclone-analysis.html#cb127-16" tabindex="-1"></a>profiles <span class="ot">=</span> profiles[(total_cn <span class="sc">!=</span> <span class="dv">2</span> <span class="sc">&amp;</span> chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>) <span class="sc">|</span> (total_cn <span class="sc">!=</span> <span class="dv">1</span> <span class="sc">&amp;</span> chr <span class="sc">==</span> <span class="st">&quot;X&quot;</span>)]</span>
<span id="cb127-17"><a href="trrfer-subclone-analysis.html#cb127-17" tabindex="-1"></a></span>
<span id="cb127-18"><a href="trrfer-subclone-analysis.html#cb127-18" tabindex="-1"></a><span class="co"># Get overlaps</span></span>
<span id="cb127-19"><a href="trrfer-subclone-analysis.html#cb127-19" tabindex="-1"></a><span class="fu">setkey</span>(profiles, chr, start, end)</span>
<span id="cb127-20"><a href="trrfer-subclone-analysis.html#cb127-20" tabindex="-1"></a><span class="fu">setkey</span>(cosmic, chr, start, end)</span>
<span id="cb127-21"><a href="trrfer-subclone-analysis.html#cb127-21" tabindex="-1"></a>overlaps <span class="ot">=</span> <span class="fu">foverlaps</span>(cosmic, profiles)[<span class="sc">!</span><span class="fu">is.na</span>(total_cn)]</span>
<span id="cb127-22"><a href="trrfer-subclone-analysis.html#cb127-22" tabindex="-1"></a>overlaps[, alteration <span class="sc">:=</span> <span class="fu">ifelse</span>(total_cn <span class="sc">&lt;</span> <span class="dv">2</span>, <span class="st">&quot;Deleted&quot;</span>, <span class="st">&quot;Amplified&quot;</span>)]</span>
<span id="cb127-23"><a href="trrfer-subclone-analysis.html#cb127-23" tabindex="-1"></a></span>
<span id="cb127-24"><a href="trrfer-subclone-analysis.html#cb127-24" tabindex="-1"></a><span class="co"># Get unique</span></span>
<span id="cb127-25"><a href="trrfer-subclone-analysis.html#cb127-25" tabindex="-1"></a>unique_overlaps <span class="ot">=</span> <span class="fu">unique</span>(overlaps[, .(name, gene, alteration)])</span>
<span id="cb127-26"><a href="trrfer-subclone-analysis.html#cb127-26" tabindex="-1"></a></span>
<span id="cb127-27"><a href="trrfer-subclone-analysis.html#cb127-27" tabindex="-1"></a><span class="co"># Make oncoprint</span></span>
<span id="cb127-28"><a href="trrfer-subclone-analysis.html#cb127-28" tabindex="-1"></a>unique_overlaps_wide <span class="ot">=</span> <span class="fu">dcast</span>(unique_overlaps, name <span class="sc">~</span> gene, <span class="at">value.var =</span> <span class="st">&quot;alteration&quot;</span>)</span>
<span id="cb127-29"><a href="trrfer-subclone-analysis.html#cb127-29" tabindex="-1"></a>mat <span class="ot">=</span> <span class="fu">as.matrix</span>(unique_overlaps_wide[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(unique_overlaps_wide)])</span>
<span id="cb127-30"><a href="trrfer-subclone-analysis.html#cb127-30" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">=</span> unique_overlaps_wide<span class="sc">$</span>name</span>
<span id="cb127-31"><a href="trrfer-subclone-analysis.html#cb127-31" tabindex="-1"></a></span>
<span id="cb127-32"><a href="trrfer-subclone-analysis.html#cb127-32" tabindex="-1"></a><span class="co"># Plot oncoprint</span></span>
<span id="cb127-33"><a href="trrfer-subclone-analysis.html#cb127-33" tabindex="-1"></a><span class="co"># colors</span></span>
<span id="cb127-34"><a href="trrfer-subclone-analysis.html#cb127-34" tabindex="-1"></a>cols <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Set1&quot;</span>)[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)]</span>
<span id="cb127-35"><a href="trrfer-subclone-analysis.html#cb127-35" tabindex="-1"></a><span class="fu">names</span>(cols) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Amplified&quot;</span>, <span class="st">&quot;Deleted&quot;</span>)</span>
<span id="cb127-36"><a href="trrfer-subclone-analysis.html#cb127-36" tabindex="-1"></a></span>
<span id="cb127-37"><a href="trrfer-subclone-analysis.html#cb127-37" tabindex="-1"></a><span class="fu">oncoPrint</span>(<span class="fu">t</span>(mat),</span>
<span id="cb127-38"><a href="trrfer-subclone-analysis.html#cb127-38" tabindex="-1"></a>          <span class="at">alter_fun =</span> <span class="fu">list</span>(</span>
<span id="cb127-39"><a href="trrfer-subclone-analysis.html#cb127-39" tabindex="-1"></a>            <span class="at">background =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.9</span>, <span class="at">height =</span> <span class="fl">0.9</span>, <span class="at">fill =</span> <span class="st">&quot;#FFFFFF&quot;</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> .<span class="dv">1</span>),</span>
<span id="cb127-40"><a href="trrfer-subclone-analysis.html#cb127-40" tabindex="-1"></a>            <span class="at">Amplified =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.85</span>, <span class="at">height =</span> <span class="fl">0.85</span>, <span class="at">fill =</span> cols[<span class="st">&quot;Amplified&quot;</span>]),</span>
<span id="cb127-41"><a href="trrfer-subclone-analysis.html#cb127-41" tabindex="-1"></a>            <span class="at">Deleted =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.85</span>, <span class="at">height =</span> <span class="fl">0.85</span>, <span class="at">fill =</span> cols[<span class="st">&quot;Deleted&quot;</span>])),</span>
<span id="cb127-42"><a href="trrfer-subclone-analysis.html#cb127-42" tabindex="-1"></a>          <span class="at">col =</span> cols, <span class="at">border =</span> <span class="st">&quot;black&quot;</span>, <span class="at">show_column_names =</span> T, <span class="at">show_row_names =</span> T, <span class="at">remove_empty_rows =</span> T,</span>
<span id="cb127-43"><a href="trrfer-subclone-analysis.html#cb127-43" tabindex="-1"></a>          <span class="at">show_pct =</span> F, <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">17</span>), <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">22</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20oncoprint%20TRR_FER-1.png" width="672" /></p>
<p>Distributions of the cells belonging to these subclones can be found in the Supplementary Figure 12-15 section.</p>
</div>
<div id="co-deletions-of-tsgs" class="section level2 hasAnchor" number="18.2">
<h2><span class="header-section-number">18.2</span> Co-deletions of TSGs<a href="trrfer-subclone-analysis.html#co-deletions-of-tsgs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To see where the cells harbouring deletions of multiple genes simultaneously are located, we go back to single-cell level (from subclonal level) and plot these cells in the prostate heatmaps.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="trrfer-subclone-analysis.html#cb128-1" tabindex="-1"></a><span class="co"># Load in data</span></span>
<span id="cb128-2"><a href="trrfer-subclone-analysis.html#cb128-2" tabindex="-1"></a>count_all <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_cnv.rds&quot;</span>)<span class="sc">$</span>stats</span>
<span id="cb128-3"><a href="trrfer-subclone-analysis.html#cb128-3" tabindex="-1"></a>profiles <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/P3_pseudodiploid_500kb.rds&quot;</span>)</span>
<span id="cb128-4"><a href="trrfer-subclone-analysis.html#cb128-4" tabindex="-1"></a>cosmic <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/genelists/cosmic_gene-census.tsv&quot;</span>)</span>
<span id="cb128-5"><a href="trrfer-subclone-analysis.html#cb128-5" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>, <span class="at">header =</span> F)</span>
<span id="cb128-6"><a href="trrfer-subclone-analysis.html#cb128-6" tabindex="-1"></a>clones <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/subclones/P3_clones.tsv&quot;</span>)</span>
<span id="cb128-7"><a href="trrfer-subclone-analysis.html#cb128-7" tabindex="-1"></a></span>
<span id="cb128-8"><a href="trrfer-subclone-analysis.html#cb128-8" tabindex="-1"></a><span class="co"># Get cosmic genes that are amp/del</span></span>
<span id="cb128-9"><a href="trrfer-subclone-analysis.html#cb128-9" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">pivot_longer</span>(profiles, <span class="at">cols =</span> <span class="fu">colnames</span>(profiles[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(profiles)]))</span>
<span id="cb128-10"><a href="trrfer-subclone-analysis.html#cb128-10" tabindex="-1"></a><span class="fu">setDT</span>(dt)</span>
<span id="cb128-11"><a href="trrfer-subclone-analysis.html#cb128-11" tabindex="-1"></a>dt <span class="ot">=</span> dt[value <span class="sc">!=</span> <span class="dv">2</span>, ]</span>
<span id="cb128-12"><a href="trrfer-subclone-analysis.html#cb128-12" tabindex="-1"></a>dt[, value <span class="sc">:=</span> <span class="fu">ifelse</span>(value <span class="sc">&lt;</span> <span class="dv">2</span>, <span class="st">&quot;Deleted&quot;</span>, <span class="st">&quot;Amplified&quot;</span>)]</span>
<span id="cb128-13"><a href="trrfer-subclone-analysis.html#cb128-13" tabindex="-1"></a></span>
<span id="cb128-14"><a href="trrfer-subclone-analysis.html#cb128-14" tabindex="-1"></a><span class="co"># Setkeys</span></span>
<span id="cb128-15"><a href="trrfer-subclone-analysis.html#cb128-15" tabindex="-1"></a><span class="fu">setkey</span>(dt, chr, start, end)</span>
<span id="cb128-16"><a href="trrfer-subclone-analysis.html#cb128-16" tabindex="-1"></a><span class="fu">setkey</span>(cosmic, chr, start, end)</span>
<span id="cb128-17"><a href="trrfer-subclone-analysis.html#cb128-17" tabindex="-1"></a></span>
<span id="cb128-18"><a href="trrfer-subclone-analysis.html#cb128-18" tabindex="-1"></a><span class="co"># Get overlap with COSMIC</span></span>
<span id="cb128-19"><a href="trrfer-subclone-analysis.html#cb128-19" tabindex="-1"></a>overlap <span class="ot">=</span> <span class="fu">foverlaps</span>(cosmic, dt)</span>
<span id="cb128-20"><a href="trrfer-subclone-analysis.html#cb128-20" tabindex="-1"></a>overlap <span class="ot">=</span> overlap[<span class="fu">complete.cases</span>(overlap)]</span>
<span id="cb128-21"><a href="trrfer-subclone-analysis.html#cb128-21" tabindex="-1"></a>overlap <span class="ot">=</span> <span class="fu">unique</span>(overlap, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;name&quot;</span>, <span class="st">&quot;value&quot;</span>, <span class="st">&quot;gene&quot;</span>))</span>
<span id="cb128-22"><a href="trrfer-subclone-analysis.html#cb128-22" tabindex="-1"></a></span>
<span id="cb128-23"><a href="trrfer-subclone-analysis.html#cb128-23" tabindex="-1"></a><span class="co"># Merge overlap with section info and subclone info</span></span>
<span id="cb128-24"><a href="trrfer-subclone-analysis.html#cb128-24" tabindex="-1"></a>overlap[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, name)]</span>
<span id="cb128-25"><a href="trrfer-subclone-analysis.html#cb128-25" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">merge</span>(overlap, annot, <span class="at">by.x =</span> <span class="st">&quot;library&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;V1&quot;</span>)</span>
<span id="cb128-26"><a href="trrfer-subclone-analysis.html#cb128-26" tabindex="-1"></a><span class="co">#dt = merge(dt, clones, by.x = &quot;name&quot;, by.y = &quot;sample_id&quot;)</span></span>
<span id="cb128-27"><a href="trrfer-subclone-analysis.html#cb128-27" tabindex="-1"></a></span>
<span id="cb128-28"><a href="trrfer-subclone-analysis.html#cb128-28" tabindex="-1"></a><span class="co"># Subclones and genes to check</span></span>
<span id="cb128-29"><a href="trrfer-subclone-analysis.html#cb128-29" tabindex="-1"></a>genes_check <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;FOXO1&quot;</span>, <span class="st">&quot;FOXO3&quot;</span>, <span class="st">&quot;RB1&quot;</span>, <span class="st">&quot;BRCA2&quot;</span>, <span class="st">&quot;CCNC&quot;</span>, <span class="st">&quot;CDX2&quot;</span>, <span class="st">&quot;LATS2&quot;</span>, <span class="st">&quot;PRDM1&quot;</span>)</span>
<span id="cb128-30"><a href="trrfer-subclone-analysis.html#cb128-30" tabindex="-1"></a></span>
<span id="cb128-31"><a href="trrfer-subclone-analysis.html#cb128-31" tabindex="-1"></a><span class="co"># Subset</span></span>
<span id="cb128-32"><a href="trrfer-subclone-analysis.html#cb128-32" tabindex="-1"></a>dt <span class="ot">=</span> dt[gene <span class="sc">%in%</span> genes_check, ]</span>
<span id="cb128-33"><a href="trrfer-subclone-analysis.html#cb128-33" tabindex="-1"></a></span>
<span id="cb128-34"><a href="trrfer-subclone-analysis.html#cb128-34" tabindex="-1"></a><span class="co"># Check co-deletions</span></span>
<span id="cb128-35"><a href="trrfer-subclone-analysis.html#cb128-35" tabindex="-1"></a>codels <span class="ot">=</span> dt[, <span class="fu">paste</span>(gene, <span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>), by <span class="ot">=</span> .(name, V2, value)]</span>
<span id="cb128-36"><a href="trrfer-subclone-analysis.html#cb128-36" tabindex="-1"></a>codels_counts <span class="ot">=</span> codels[, .N, by <span class="ot">=</span> .(V2, V1)]</span>
<span id="cb128-37"><a href="trrfer-subclone-analysis.html#cb128-37" tabindex="-1"></a></span>
<span id="cb128-38"><a href="trrfer-subclone-analysis.html#cb128-38" tabindex="-1"></a><span class="co"># Get total number of pseudodiploid per section</span></span>
<span id="cb128-39"><a href="trrfer-subclone-analysis.html#cb128-39" tabindex="-1"></a>total_pseudo <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">library =</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(profiles[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(profiles)])))</span>
<span id="cb128-40"><a href="trrfer-subclone-analysis.html#cb128-40" tabindex="-1"></a>total_pseudo <span class="ot">=</span> <span class="fu">merge</span>(total_pseudo, annot, <span class="at">by.x =</span> <span class="st">&quot;library&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;V1&quot;</span>)</span>
<span id="cb128-41"><a href="trrfer-subclone-analysis.html#cb128-41" tabindex="-1"></a>counts_pseudo <span class="ot">=</span> total_pseudo[, .(<span class="at">total_pseudo =</span> .N), by <span class="ot">=</span> V2]</span>
<span id="cb128-42"><a href="trrfer-subclone-analysis.html#cb128-42" tabindex="-1"></a></span>
<span id="cb128-43"><a href="trrfer-subclone-analysis.html#cb128-43" tabindex="-1"></a><span class="co"># Get total cells</span></span>
<span id="cb128-44"><a href="trrfer-subclone-analysis.html#cb128-44" tabindex="-1"></a>count_all[, library <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sample)]</span>
<span id="cb128-45"><a href="trrfer-subclone-analysis.html#cb128-45" tabindex="-1"></a>count_all <span class="ot">=</span> <span class="fu">merge</span>(count_all, annot, <span class="at">by.x =</span> <span class="st">&quot;library&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;V1&quot;</span>)</span>
<span id="cb128-46"><a href="trrfer-subclone-analysis.html#cb128-46" tabindex="-1"></a>counts_total <span class="ot">=</span> count_all[classifier_prediction <span class="sc">==</span> <span class="st">&quot;good&quot;</span>, .(<span class="at">total =</span> .N), by <span class="ot">=</span> .(V2)]</span>
<span id="cb128-47"><a href="trrfer-subclone-analysis.html#cb128-47" tabindex="-1"></a></span>
<span id="cb128-48"><a href="trrfer-subclone-analysis.html#cb128-48" tabindex="-1"></a><span class="co"># Get fraction pseudo</span></span>
<span id="cb128-49"><a href="trrfer-subclone-analysis.html#cb128-49" tabindex="-1"></a>codels_counts <span class="ot">=</span> <span class="fu">merge</span>(codels_counts, counts_pseudo, <span class="at">by =</span> <span class="st">&quot;V2&quot;</span>)</span>
<span id="cb128-50"><a href="trrfer-subclone-analysis.html#cb128-50" tabindex="-1"></a>codels_counts[, fraction_pseudo <span class="sc">:=</span> N <span class="sc">/</span> total_pseudo]</span>
<span id="cb128-51"><a href="trrfer-subclone-analysis.html#cb128-51" tabindex="-1"></a></span>
<span id="cb128-52"><a href="trrfer-subclone-analysis.html#cb128-52" tabindex="-1"></a><span class="co"># Add fraction total</span></span>
<span id="cb128-53"><a href="trrfer-subclone-analysis.html#cb128-53" tabindex="-1"></a>codels_counts <span class="ot">=</span> <span class="fu">merge</span>(codels_counts, counts_total, <span class="at">by =</span> <span class="st">&quot;V2&quot;</span>)</span>
<span id="cb128-54"><a href="trrfer-subclone-analysis.html#cb128-54" tabindex="-1"></a>codels_counts[, fraction_total <span class="sc">:=</span> N <span class="sc">/</span> total]</span>
<span id="cb128-55"><a href="trrfer-subclone-analysis.html#cb128-55" tabindex="-1"></a></span>
<span id="cb128-56"><a href="trrfer-subclone-analysis.html#cb128-56" tabindex="-1"></a><span class="co"># Plot the combination with all deletions</span></span>
<span id="cb128-57"><a href="trrfer-subclone-analysis.html#cb128-57" tabindex="-1"></a>combination <span class="ot">=</span> <span class="st">&quot;LATS2;CDX2;BRCA2;FOXO1;RB1;CCNC;PRDM1;FOXO3&quot;</span></span>
<span id="cb128-58"><a href="trrfer-subclone-analysis.html#cb128-58" tabindex="-1"></a></span>
<span id="cb128-59"><a href="trrfer-subclone-analysis.html#cb128-59" tabindex="-1"></a><span class="co"># Select the combination of interest</span></span>
<span id="cb128-60"><a href="trrfer-subclone-analysis.html#cb128-60" tabindex="-1"></a>dt <span class="ot">=</span> codels_counts[V1 <span class="sc">==</span> combination]</span>
<span id="cb128-61"><a href="trrfer-subclone-analysis.html#cb128-61" tabindex="-1"></a></span>
<span id="cb128-62"><a href="trrfer-subclone-analysis.html#cb128-62" tabindex="-1"></a><span class="co"># Fill all with NAs</span></span>
<span id="cb128-63"><a href="trrfer-subclone-analysis.html#cb128-63" tabindex="-1"></a>fill_dt <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">V2 =</span> annot[<span class="sc">!</span>V2 <span class="sc">%in%</span> dt<span class="sc">$</span>V2, V2],</span>
<span id="cb128-64"><a href="trrfer-subclone-analysis.html#cb128-64" tabindex="-1"></a>                     <span class="at">V1 =</span> <span class="cn">NA</span>,</span>
<span id="cb128-65"><a href="trrfer-subclone-analysis.html#cb128-65" tabindex="-1"></a>                     <span class="at">N =</span> <span class="cn">NA</span>,</span>
<span id="cb128-66"><a href="trrfer-subclone-analysis.html#cb128-66" tabindex="-1"></a>                     <span class="at">total_pseudo =</span> <span class="cn">NA</span>,</span>
<span id="cb128-67"><a href="trrfer-subclone-analysis.html#cb128-67" tabindex="-1"></a>                     <span class="at">fraction_pseudo =</span> <span class="cn">NA</span>,</span>
<span id="cb128-68"><a href="trrfer-subclone-analysis.html#cb128-68" tabindex="-1"></a>                     <span class="at">total =</span> <span class="cn">NA</span>,</span>
<span id="cb128-69"><a href="trrfer-subclone-analysis.html#cb128-69" tabindex="-1"></a>                     <span class="at">fraction_total =</span> <span class="cn">NA</span>)</span>
<span id="cb128-70"><a href="trrfer-subclone-analysis.html#cb128-70" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">rbindlist</span>(<span class="fu">list</span>(dt, fill_dt), <span class="at">use.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb128-71"><a href="trrfer-subclone-analysis.html#cb128-71" tabindex="-1"></a></span>
<span id="cb128-72"><a href="trrfer-subclone-analysis.html#cb128-72" tabindex="-1"></a><span class="co"># Add coordinates</span></span>
<span id="cb128-73"><a href="trrfer-subclone-analysis.html#cb128-73" tabindex="-1"></a>dt[, x <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, V2))]</span>
<span id="cb128-74"><a href="trrfer-subclone-analysis.html#cb128-74" tabindex="-1"></a>dt[, y <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, V2))]</span>
<span id="cb128-75"><a href="trrfer-subclone-analysis.html#cb128-75" tabindex="-1"></a></span>
<span id="cb128-76"><a href="trrfer-subclone-analysis.html#cb128-76" tabindex="-1"></a><span class="co"># Plot the number of cells with co-deletion</span></span>
<span id="cb128-77"><a href="trrfer-subclone-analysis.html#cb128-77" tabindex="-1"></a><span class="fu">ggplot</span>(dt, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> N, <span class="at">label =</span> N)) <span class="sc">+</span></span>
<span id="cb128-78"><a href="trrfer-subclone-analysis.html#cb128-78" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb128-79"><a href="trrfer-subclone-analysis.html#cb128-79" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">size =</span> <span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb128-80"><a href="trrfer-subclone-analysis.html#cb128-80" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(combination, <span class="st">&quot; deletions&quot;</span>)) <span class="sc">+</span></span>
<span id="cb128-81"><a href="trrfer-subclone-analysis.html#cb128-81" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;Cells with deletion&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="at">direction =</span> <span class="dv">1</span>, <span class="at">na.value =</span> <span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb128-82"><a href="trrfer-subclone-analysis.html#cb128-82" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(dt<span class="sc">$</span>y), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb128-83"><a href="trrfer-subclone-analysis.html#cb128-83" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(dt<span class="sc">$</span>x), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb128-84"><a href="trrfer-subclone-analysis.html#cb128-84" tabindex="-1"></a>  <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>y)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>y))) <span class="sc">+</span> </span>
<span id="cb128-85"><a href="trrfer-subclone-analysis.html#cb128-85" tabindex="-1"></a>  <span class="fu">scale_x_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>x)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>x))) <span class="sc">+</span></span>
<span id="cb128-86"><a href="trrfer-subclone-analysis.html#cb128-86" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/co-deletion-1.png" width="672" /></p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="trrfer-subclone-analysis.html#cb129-1" tabindex="-1"></a><span class="co"># Plot fraction of pseudodiploid cells</span></span>
<span id="cb129-2"><a href="trrfer-subclone-analysis.html#cb129-2" tabindex="-1"></a><span class="fu">ggplot</span>(dt, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> fraction_pseudo, <span class="at">label =</span> N)) <span class="sc">+</span></span>
<span id="cb129-3"><a href="trrfer-subclone-analysis.html#cb129-3" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb129-4"><a href="trrfer-subclone-analysis.html#cb129-4" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">size =</span> <span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb129-5"><a href="trrfer-subclone-analysis.html#cb129-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(combination, <span class="st">&quot; deletions&quot;</span>)) <span class="sc">+</span></span>
<span id="cb129-6"><a href="trrfer-subclone-analysis.html#cb129-6" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;fraction of all</span><span class="sc">\n</span><span class="st">pseudo-diploid cells&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="at">direction =</span> <span class="dv">1</span>, <span class="at">na.value =</span> <span class="st">&quot;grey&quot;</span>, <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>(), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb129-7"><a href="trrfer-subclone-analysis.html#cb129-7" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(dt<span class="sc">$</span>y), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb129-8"><a href="trrfer-subclone-analysis.html#cb129-8" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(dt<span class="sc">$</span>x), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb129-9"><a href="trrfer-subclone-analysis.html#cb129-9" tabindex="-1"></a>  <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>y)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>y))) <span class="sc">+</span> </span>
<span id="cb129-10"><a href="trrfer-subclone-analysis.html#cb129-10" tabindex="-1"></a>  <span class="fu">scale_x_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>x)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>x))) <span class="sc">+</span></span>
<span id="cb129-11"><a href="trrfer-subclone-analysis.html#cb129-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/co-deletion-2.png" width="672" /></p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="trrfer-subclone-analysis.html#cb130-1" tabindex="-1"></a><span class="co"># Plot fraction of total cells</span></span>
<span id="cb130-2"><a href="trrfer-subclone-analysis.html#cb130-2" tabindex="-1"></a><span class="fu">ggplot</span>(dt, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> fraction_total, <span class="at">label =</span> N)) <span class="sc">+</span></span>
<span id="cb130-3"><a href="trrfer-subclone-analysis.html#cb130-3" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb130-4"><a href="trrfer-subclone-analysis.html#cb130-4" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">size =</span> <span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb130-5"><a href="trrfer-subclone-analysis.html#cb130-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(combination, <span class="st">&quot; deletions&quot;</span>)) <span class="sc">+</span></span>
<span id="cb130-6"><a href="trrfer-subclone-analysis.html#cb130-6" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;fraction of</span><span class="sc">\n</span><span class="st">all cells&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="at">direction =</span> <span class="dv">1</span>, <span class="at">na.value =</span> <span class="st">&quot;grey&quot;</span>, <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>(), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb130-7"><a href="trrfer-subclone-analysis.html#cb130-7" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(dt<span class="sc">$</span>y), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb130-8"><a href="trrfer-subclone-analysis.html#cb130-8" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(dt<span class="sc">$</span>x), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb130-9"><a href="trrfer-subclone-analysis.html#cb130-9" tabindex="-1"></a>  <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>y)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>y))) <span class="sc">+</span> </span>
<span id="cb130-10"><a href="trrfer-subclone-analysis.html#cb130-10" tabindex="-1"></a>  <span class="fu">scale_x_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>x)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>x))) <span class="sc">+</span></span>
<span id="cb130-11"><a href="trrfer-subclone-analysis.html#cb130-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/co-deletion-3.png" width="672" /></p>
</div>
<div id="tcga-deletion-analysis" class="section level2 hasAnchor" number="18.3">
<h2><span class="header-section-number">18.3</span> TCGA deletion analysis<a href="trrfer-subclone-analysis.html#tcga-deletion-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we wanted to see if these deletions are also present (and enriched) in the Prostate Adenocarcinoma (PRAD) TCGA dataset. First we looked into enrichment of deletions of certain genomic regions in TCGA. We used <a href="https://github.com/broadinstitute/gistic2">GISTIC2</a> for this. You can download the TCGA copy number data at <a href="https://www.cbioportal.org/study/cnSegments?id=prad_tcga">cbioportal</a>. Then we ran GISTIC2 with the following command:</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb131-1"><a href="trrfer-subclone-analysis.html#cb131-1" tabindex="-1"></a><span class="ex">gistic2</span> gp_gistic2_from_seg <span class="at">-b</span> {base_dir} <span class="at">-gcm</span> extreme <span class="at">-seg</span> {TCGA_copynumber_segment_file} <span class="at">-maxseg</span> 2000 <span class="at">-broad</span> 1 <span class="at">-ta</span> 0.1 <span class="at">-td</span> 0.1 <span class="at">-conf</span> 0.99 <span class="at">-brlen</span> 0.7 <span class="at">-armpeel</span> 1</span></code></pre></div>
<p>We then used the <code>scores.gistic</code> output file from GISTIC2 to visualize the signficantly deleted (and amplified) regions in PRAD.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="trrfer-subclone-analysis.html#cb132-1" tabindex="-1"></a><span class="co"># Load GISTIC data</span></span>
<span id="cb132-2"><a href="trrfer-subclone-analysis.html#cb132-2" tabindex="-1"></a>dt <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/TCGA/scores.gistic&quot;</span>)</span>
<span id="cb132-3"><a href="trrfer-subclone-analysis.html#cb132-3" tabindex="-1"></a></span>
<span id="cb132-4"><a href="trrfer-subclone-analysis.html#cb132-4" tabindex="-1"></a><span class="co"># Make wide</span></span>
<span id="cb132-5"><a href="trrfer-subclone-analysis.html#cb132-5" tabindex="-1"></a>dt_amp <span class="ot">=</span> dt[Type <span class="sc">==</span> <span class="st">&quot;Amp&quot;</span>, .(<span class="at">chr =</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, Chromosome), <span class="at">start =</span> Start, <span class="at">end =</span> End, <span class="at">qvalue =</span> <span class="sc">-</span><span class="dv">1</span><span class="sc">*</span><span class="st">`</span><span class="at">-log10(q-value)</span><span class="st">`</span>)]</span>
<span id="cb132-6"><a href="trrfer-subclone-analysis.html#cb132-6" tabindex="-1"></a>dt_del <span class="ot">=</span> dt[Type <span class="sc">==</span> <span class="st">&quot;Del&quot;</span>, .(<span class="at">chr =</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, Chromosome), <span class="at">start =</span> Start, <span class="at">end =</span> End, <span class="at">qvalue =</span> <span class="st">`</span><span class="at">-log10(q-value)</span><span class="st">`</span>)]</span>
<span id="cb132-7"><a href="trrfer-subclone-analysis.html#cb132-7" tabindex="-1"></a></span>
<span id="cb132-8"><a href="trrfer-subclone-analysis.html#cb132-8" tabindex="-1"></a><span class="fu">circos.clear</span>()</span>
<span id="cb132-9"><a href="trrfer-subclone-analysis.html#cb132-9" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="st">&quot;start.degree&quot;</span> <span class="ot">=</span> <span class="dv">90</span>, <span class="st">&quot;gap.degree&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">1</span>, <span class="dv">21</span>), <span class="dv">10</span>), <span class="at">points.overflow.warning=</span><span class="cn">FALSE</span>, <span class="at">track.height =</span> <span class="fl">0.25</span>, <span class="at">track.margin =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="dv">0</span>))</span>
<span id="cb132-10"><a href="trrfer-subclone-analysis.html#cb132-10" tabindex="-1"></a><span class="fu">circos.initializeWithIdeogram</span>(<span class="at">plotType =</span> <span class="fu">c</span>(<span class="st">&quot;ideogram&quot;</span>, <span class="st">&quot;axis&quot;</span>, <span class="st">&quot;labels&quot;</span>), <span class="at">chromosome.index =</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>))</span>
<span id="cb132-11"><a href="trrfer-subclone-analysis.html#cb132-11" tabindex="-1"></a><span class="co"># Plot deletion q value track</span></span>
<span id="cb132-12"><a href="trrfer-subclone-analysis.html#cb132-12" tabindex="-1"></a><span class="fu">circos.genomicTrack</span>(dt_del, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>), <span class="at">panel.fun =</span> <span class="cf">function</span>(region, value, ...) {</span>
<span id="cb132-13"><a href="trrfer-subclone-analysis.html#cb132-13" tabindex="-1"></a>  <span class="fu">circos.genomicLines</span>(region, value, <span class="at">numeric.column =</span> <span class="st">&quot;del&quot;</span>, <span class="at">col =</span> <span class="st">&quot;#377EB8&quot;</span>, <span class="at">area =</span> <span class="cn">TRUE</span>)</span>
<span id="cb132-14"><a href="trrfer-subclone-analysis.html#cb132-14" tabindex="-1"></a>})</span>
<span id="cb132-15"><a href="trrfer-subclone-analysis.html#cb132-15" tabindex="-1"></a><span class="fu">circos.yaxis</span>(<span class="at">side =</span> <span class="st">&quot;left&quot;</span>, <span class="at">at =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">50</span>, <span class="dv">100</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">50</span>, <span class="dv">100</span>), <span class="at">sector.index =</span> <span class="st">&quot;chr1&quot;</span>, <span class="at">labels.niceFacing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb132-16"><a href="trrfer-subclone-analysis.html#cb132-16" tabindex="-1"></a></span>
<span id="cb132-17"><a href="trrfer-subclone-analysis.html#cb132-17" tabindex="-1"></a><span class="co"># Plot amplification q value track</span></span>
<span id="cb132-18"><a href="trrfer-subclone-analysis.html#cb132-18" tabindex="-1"></a><span class="fu">circos.genomicTrack</span>(dt_amp, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">0</span>), <span class="at">panel.fun =</span> <span class="cf">function</span>(region, value, ...) {</span>
<span id="cb132-19"><a href="trrfer-subclone-analysis.html#cb132-19" tabindex="-1"></a>  <span class="fu">circos.genomicLines</span>(region, value, <span class="at">numeric.column =</span> <span class="st">&quot;amp&quot;</span>, <span class="at">col =</span> <span class="st">&quot;#E41A1C&quot;</span>, <span class="at">area =</span> <span class="cn">TRUE</span>, <span class="at">baseline =</span> <span class="st">&quot;top&quot;</span>, <span class="at">type =</span> <span class="st">&quot;l&quot;</span>)</span>
<span id="cb132-20"><a href="trrfer-subclone-analysis.html#cb132-20" tabindex="-1"></a>})</span>
<span id="cb132-21"><a href="trrfer-subclone-analysis.html#cb132-21" tabindex="-1"></a><span class="fu">circos.yaxis</span>(<span class="at">side =</span> <span class="st">&quot;left&quot;</span>, <span class="at">at =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="sc">-</span><span class="dv">5</span>, <span class="dv">0</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">5</span>, <span class="dv">0</span>), <span class="at">sector.index =</span> <span class="st">&quot;chr1&quot;</span>, <span class="at">labels.niceFacing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot%20GISTIC2%20circos-1.png" width="672" /></p>
<p>Next, we checked the mutation-status (deletions/amplfications/SNVs) of the genes that we found commonly deleted in our samples in TCGA PRAD. For this we load in the CNA data (<code>.seg</code> file) and we download the mutation data. Note, the downloading and loading of the mutation data can take a couple of minutes.</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="trrfer-subclone-analysis.html#cb133-1" tabindex="-1"></a><span class="co"># Load TCGA data</span></span>
<span id="cb133-2"><a href="trrfer-subclone-analysis.html#cb133-2" tabindex="-1"></a>segments <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data//TCGA/TCGA_firehose-segments.seg&quot;</span>)</span>
<span id="cb133-3"><a href="trrfer-subclone-analysis.html#cb133-3" tabindex="-1"></a>clinical <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/TCGA/TCGA_firehose_clinical_data_patient.txt&quot;</span>)</span>
<span id="cb133-4"><a href="trrfer-subclone-analysis.html#cb133-4" tabindex="-1"></a>cosmic <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/genelists/cosmic_gene-census.tsv&quot;</span>)</span>
<span id="cb133-5"><a href="trrfer-subclone-analysis.html#cb133-5" tabindex="-1"></a></span>
<span id="cb133-6"><a href="trrfer-subclone-analysis.html#cb133-6" tabindex="-1"></a><span class="co"># Download mutation data</span></span>
<span id="cb133-7"><a href="trrfer-subclone-analysis.html#cb133-7" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">GDCquery</span>(<span class="at">project =</span> <span class="st">&quot;TCGA-PRAD&quot;</span>, <span class="at">data.category =</span> <span class="st">&quot;Simple Nucleotide Variation&quot;</span>, <span class="at">access =</span> <span class="st">&quot;open&quot;</span>,</span>
<span id="cb133-8"><a href="trrfer-subclone-analysis.html#cb133-8" tabindex="-1"></a>               <span class="at">data.type =</span> <span class="st">&quot;Masked Somatic Mutation&quot;</span>, <span class="at">workflow.type =</span> <span class="st">&quot;Aliquot Ensemble Somatic Variant Merging and Masking&quot;</span>)</span>
<span id="cb133-9"><a href="trrfer-subclone-analysis.html#cb133-9" tabindex="-1"></a><span class="fu">GDCdownload</span>(res, <span class="at">directory =</span> <span class="st">&quot;./data/TCGA/gdc_query/&quot;</span>)</span>
<span id="cb133-10"><a href="trrfer-subclone-analysis.html#cb133-10" tabindex="-1"></a>tcga_muts <span class="ot">=</span> <span class="fu">GDCprepare</span>(res, <span class="at">directory =</span> <span class="st">&quot;./data/TCGA/gdc_query/&quot;</span>)</span>
<span id="cb133-11"><a href="trrfer-subclone-analysis.html#cb133-11" tabindex="-1"></a></span>
<span id="cb133-12"><a href="trrfer-subclone-analysis.html#cb133-12" tabindex="-1"></a><span class="co"># Set as data.table and rename columns for ease of use</span></span>
<span id="cb133-13"><a href="trrfer-subclone-analysis.html#cb133-13" tabindex="-1"></a><span class="fu">setDT</span>(tcga_muts)</span>
<span id="cb133-14"><a href="trrfer-subclone-analysis.html#cb133-14" tabindex="-1"></a><span class="fu">setnames</span>(tcga_muts, <span class="st">&quot;Gene&quot;</span>, <span class="st">&quot;gene&quot;</span>)</span>
<span id="cb133-15"><a href="trrfer-subclone-analysis.html#cb133-15" tabindex="-1"></a></span>
<span id="cb133-16"><a href="trrfer-subclone-analysis.html#cb133-16" tabindex="-1"></a><span class="co"># Get actual patient ID</span></span>
<span id="cb133-17"><a href="trrfer-subclone-analysis.html#cb133-17" tabindex="-1"></a>tcga_muts[, ID <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;-0.*|-0.*&quot;</span>, <span class="st">&quot;&quot;</span>, Tumor_Sample_Barcode)]</span>
<span id="cb133-18"><a href="trrfer-subclone-analysis.html#cb133-18" tabindex="-1"></a>segments[, ID <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;-0.*&quot;</span>, <span class="st">&quot;&quot;</span>, ID)]</span>
<span id="cb133-19"><a href="trrfer-subclone-analysis.html#cb133-19" tabindex="-1"></a></span>
<span id="cb133-20"><a href="trrfer-subclone-analysis.html#cb133-20" tabindex="-1"></a><span class="co"># Select IDs that are in cn data and vice versa</span></span>
<span id="cb133-21"><a href="trrfer-subclone-analysis.html#cb133-21" tabindex="-1"></a>tcga_muts <span class="ot">=</span> tcga_muts[ID <span class="sc">%in%</span> segments<span class="sc">$</span>ID <span class="sc">&amp;</span> Variant_Classification <span class="sc">==</span> <span class="st">&quot;Missense_Mutation&quot;</span>, .(ID, Hugo_Symbol)]</span>
<span id="cb133-22"><a href="trrfer-subclone-analysis.html#cb133-22" tabindex="-1"></a>tcga_muts[, alteration <span class="sc">:=</span> <span class="st">&quot;SNV&quot;</span>]</span>
<span id="cb133-23"><a href="trrfer-subclone-analysis.html#cb133-23" tabindex="-1"></a><span class="fu">setnames</span>(tcga_muts, <span class="st">&quot;Hugo_Symbol&quot;</span>, <span class="st">&quot;gene&quot;</span>)</span>
<span id="cb133-24"><a href="trrfer-subclone-analysis.html#cb133-24" tabindex="-1"></a></span>
<span id="cb133-25"><a href="trrfer-subclone-analysis.html#cb133-25" tabindex="-1"></a><span class="co"># Select only IDs that are also in the mutation data.table</span></span>
<span id="cb133-26"><a href="trrfer-subclone-analysis.html#cb133-26" tabindex="-1"></a>segments <span class="ot">=</span> segments[ID <span class="sc">%in%</span> tcga_muts<span class="sc">$</span>ID]</span>
<span id="cb133-27"><a href="trrfer-subclone-analysis.html#cb133-27" tabindex="-1"></a></span>
<span id="cb133-28"><a href="trrfer-subclone-analysis.html#cb133-28" tabindex="-1"></a><span class="co"># Setnames</span></span>
<span id="cb133-29"><a href="trrfer-subclone-analysis.html#cb133-29" tabindex="-1"></a><span class="fu">setnames</span>(segments, <span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>, <span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;num_mark&quot;</span>, <span class="st">&quot;segmean&quot;</span>))</span>
<span id="cb133-30"><a href="trrfer-subclone-analysis.html#cb133-30" tabindex="-1"></a></span>
<span id="cb133-31"><a href="trrfer-subclone-analysis.html#cb133-31" tabindex="-1"></a><span class="co"># Get deletions only and remove X from cosmic</span></span>
<span id="cb133-32"><a href="trrfer-subclone-analysis.html#cb133-32" tabindex="-1"></a>segments[segmean <span class="sc">&lt;=</span> <span class="sc">-</span>.<span class="dv">5</span>, alteration <span class="sc">:=</span> <span class="st">&quot;Deleted&quot;</span>]</span>
<span id="cb133-33"><a href="trrfer-subclone-analysis.html#cb133-33" tabindex="-1"></a>segments[segmean <span class="sc">&gt;=</span> .<span class="dv">5</span>, alteration <span class="sc">:=</span> <span class="st">&quot;Amplified&quot;</span>]</span>
<span id="cb133-34"><a href="trrfer-subclone-analysis.html#cb133-34" tabindex="-1"></a>segments <span class="ot">=</span> segments[<span class="sc">!</span><span class="fu">is.na</span>(alteration), ]</span>
<span id="cb133-35"><a href="trrfer-subclone-analysis.html#cb133-35" tabindex="-1"></a></span>
<span id="cb133-36"><a href="trrfer-subclone-analysis.html#cb133-36" tabindex="-1"></a>cosmic <span class="ot">=</span> cosmic[chr <span class="sc">!=</span> <span class="st">&quot;X&quot;</span>,]</span>
<span id="cb133-37"><a href="trrfer-subclone-analysis.html#cb133-37" tabindex="-1"></a>cosmic[, chr <span class="sc">:=</span> <span class="fu">as.integer</span>(chr)]</span>
<span id="cb133-38"><a href="trrfer-subclone-analysis.html#cb133-38" tabindex="-1"></a></span>
<span id="cb133-39"><a href="trrfer-subclone-analysis.html#cb133-39" tabindex="-1"></a><span class="co"># Setkeys</span></span>
<span id="cb133-40"><a href="trrfer-subclone-analysis.html#cb133-40" tabindex="-1"></a><span class="fu">setkey</span>(segments, chr, start, end)</span>
<span id="cb133-41"><a href="trrfer-subclone-analysis.html#cb133-41" tabindex="-1"></a><span class="fu">setkey</span>(cosmic, chr, start, end)</span>
<span id="cb133-42"><a href="trrfer-subclone-analysis.html#cb133-42" tabindex="-1"></a></span>
<span id="cb133-43"><a href="trrfer-subclone-analysis.html#cb133-43" tabindex="-1"></a><span class="co"># Get overlap with COSMIC</span></span>
<span id="cb133-44"><a href="trrfer-subclone-analysis.html#cb133-44" tabindex="-1"></a>overlap <span class="ot">=</span> <span class="fu">foverlaps</span>(cosmic, segments)</span>
<span id="cb133-45"><a href="trrfer-subclone-analysis.html#cb133-45" tabindex="-1"></a>overlap <span class="ot">=</span> overlap[<span class="fu">complete.cases</span>(overlap)]</span>
<span id="cb133-46"><a href="trrfer-subclone-analysis.html#cb133-46" tabindex="-1"></a></span>
<span id="cb133-47"><a href="trrfer-subclone-analysis.html#cb133-47" tabindex="-1"></a><span class="co"># Get unique IDS</span></span>
<span id="cb133-48"><a href="trrfer-subclone-analysis.html#cb133-48" tabindex="-1"></a>overlap <span class="ot">=</span> <span class="fu">unique</span>(overlap, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>, <span class="st">&quot;gene&quot;</span>))</span>
<span id="cb133-49"><a href="trrfer-subclone-analysis.html#cb133-49" tabindex="-1"></a></span>
<span id="cb133-50"><a href="trrfer-subclone-analysis.html#cb133-50" tabindex="-1"></a><span class="co"># Merge with mutations</span></span>
<span id="cb133-51"><a href="trrfer-subclone-analysis.html#cb133-51" tabindex="-1"></a>overlap <span class="ot">=</span> overlap[, .(ID, alteration, gene)]</span>
<span id="cb133-52"><a href="trrfer-subclone-analysis.html#cb133-52" tabindex="-1"></a>overlap <span class="ot">=</span> <span class="fu">rbind</span>(overlap, tcga_muts[, .(ID, gene, alteration)])</span>
<span id="cb133-53"><a href="trrfer-subclone-analysis.html#cb133-53" tabindex="-1"></a></span>
<span id="cb133-54"><a href="trrfer-subclone-analysis.html#cb133-54" tabindex="-1"></a><span class="co"># Select genes</span></span>
<span id="cb133-55"><a href="trrfer-subclone-analysis.html#cb133-55" tabindex="-1"></a>genes_select <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;FOXO1&quot;</span>, <span class="st">&quot;FOXO3&quot;</span>, <span class="st">&quot;FOXP1&quot;</span>, <span class="st">&quot;RB1&quot;</span>, <span class="st">&quot;CCNC&quot;</span>, <span class="st">&quot;CDX2&quot;</span>, <span class="st">&quot;LATS2&quot;</span>, <span class="st">&quot;PRDM1&quot;</span>, <span class="st">&quot;BRCA2&quot;</span>)</span>
<span id="cb133-56"><a href="trrfer-subclone-analysis.html#cb133-56" tabindex="-1"></a>overlap <span class="ot">=</span> overlap[gene <span class="sc">%in%</span> genes_select, ]</span>
<span id="cb133-57"><a href="trrfer-subclone-analysis.html#cb133-57" tabindex="-1"></a></span>
<span id="cb133-58"><a href="trrfer-subclone-analysis.html#cb133-58" tabindex="-1"></a><span class="co"># Dcast </span></span>
<span id="cb133-59"><a href="trrfer-subclone-analysis.html#cb133-59" tabindex="-1"></a>result <span class="ot">=</span> <span class="fu">dcast</span>(overlap, gene <span class="sc">~</span> ID, <span class="at">value.var =</span> <span class="st">&quot;alteration&quot;</span>, <span class="at">fun.aggregate =</span> <span class="cf">function</span>(x) <span class="fu">paste</span>(x, <span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>))</span>
<span id="cb133-60"><a href="trrfer-subclone-analysis.html#cb133-60" tabindex="-1"></a></span>
<span id="cb133-61"><a href="trrfer-subclone-analysis.html#cb133-61" tabindex="-1"></a><span class="co"># Prepare oncoprint</span></span>
<span id="cb133-62"><a href="trrfer-subclone-analysis.html#cb133-62" tabindex="-1"></a>cols <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Set1&quot;</span>)[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>)]</span>
<span id="cb133-63"><a href="trrfer-subclone-analysis.html#cb133-63" tabindex="-1"></a><span class="fu">names</span>(cols) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Amplified&quot;</span>, <span class="st">&quot;Deleted&quot;</span>, <span class="st">&quot;SNV&quot;</span>)</span>
<span id="cb133-64"><a href="trrfer-subclone-analysis.html#cb133-64" tabindex="-1"></a></span>
<span id="cb133-65"><a href="trrfer-subclone-analysis.html#cb133-65" tabindex="-1"></a>mat <span class="ot">=</span> <span class="fu">as.matrix</span>(result[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(result)])</span>
<span id="cb133-66"><a href="trrfer-subclone-analysis.html#cb133-66" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">=</span> result[[<span class="dv">1</span>]]</span>
<span id="cb133-67"><a href="trrfer-subclone-analysis.html#cb133-67" tabindex="-1"></a></span>
<span id="cb133-68"><a href="trrfer-subclone-analysis.html#cb133-68" tabindex="-1"></a><span class="co"># Add clinical Gleason score</span></span>
<span id="cb133-69"><a href="trrfer-subclone-analysis.html#cb133-69" tabindex="-1"></a>clinical <span class="ot">=</span> clinical[, .(PATIENT_ID, GLEASON_SCORE, GLEASON_PATTERN_PRIMARY, GLEASON_PATTERN_SECONDARY)]</span>
<span id="cb133-70"><a href="trrfer-subclone-analysis.html#cb133-70" tabindex="-1"></a>clinical[, gleason <span class="sc">:=</span> <span class="fu">paste0</span>(GLEASON_PATTERN_PRIMARY, <span class="st">&quot;+&quot;</span>, GLEASON_PATTERN_SECONDARY)]</span>
<span id="cb133-71"><a href="trrfer-subclone-analysis.html#cb133-71" tabindex="-1"></a></span>
<span id="cb133-72"><a href="trrfer-subclone-analysis.html#cb133-72" tabindex="-1"></a><span class="co"># Assign clinical grade groups based on gleason</span></span>
<span id="cb133-73"><a href="trrfer-subclone-analysis.html#cb133-73" tabindex="-1"></a>clinical[GLEASON_SCORE <span class="sc">==</span> <span class="dv">6</span>, grade_group <span class="sc">:=</span> <span class="st">&quot;Grade Group 1&quot;</span>]</span>
<span id="cb133-74"><a href="trrfer-subclone-analysis.html#cb133-74" tabindex="-1"></a>clinical[gleason <span class="sc">==</span> <span class="st">&quot;3+4&quot;</span>, grade_group <span class="sc">:=</span> <span class="st">&quot;Grade Group 2&quot;</span>]</span>
<span id="cb133-75"><a href="trrfer-subclone-analysis.html#cb133-75" tabindex="-1"></a>clinical[gleason <span class="sc">==</span> <span class="st">&quot;4+3&quot;</span>, grade_group <span class="sc">:=</span> <span class="st">&quot;Grade Group 3&quot;</span>]</span>
<span id="cb133-76"><a href="trrfer-subclone-analysis.html#cb133-76" tabindex="-1"></a>clinical[GLEASON_SCORE <span class="sc">==</span> <span class="dv">8</span>, grade_group <span class="sc">:=</span> <span class="st">&quot;Grade Group 4&quot;</span>]</span>
<span id="cb133-77"><a href="trrfer-subclone-analysis.html#cb133-77" tabindex="-1"></a>clinical[GLEASON_SCORE <span class="sc">&gt;</span> <span class="dv">8</span>, grade_group <span class="sc">:=</span> <span class="st">&quot;Grade Group 5&quot;</span>]</span>
<span id="cb133-78"><a href="trrfer-subclone-analysis.html#cb133-78" tabindex="-1"></a></span>
<span id="cb133-79"><a href="trrfer-subclone-analysis.html#cb133-79" tabindex="-1"></a><span class="co"># Add annotation and reorder</span></span>
<span id="cb133-80"><a href="trrfer-subclone-analysis.html#cb133-80" tabindex="-1"></a>clinical <span class="ot">=</span> clinical[PATIENT_ID <span class="sc">%in%</span> <span class="fu">colnames</span>(mat), ]</span>
<span id="cb133-81"><a href="trrfer-subclone-analysis.html#cb133-81" tabindex="-1"></a>clinical <span class="ot">=</span> clinical[<span class="fu">match</span>(<span class="fu">colnames</span>(mat), PATIENT_ID)]</span>
<span id="cb133-82"><a href="trrfer-subclone-analysis.html#cb133-82" tabindex="-1"></a></span>
<span id="cb133-83"><a href="trrfer-subclone-analysis.html#cb133-83" tabindex="-1"></a><span class="co">#annot_cols = list(&quot;Gleason score&quot; = c(&quot;Gleason low&quot; = brewer.pal(3, &quot;Set1&quot;)[2], &quot;Gleason high&quot; = brewer.pal(3, &quot;Set1&quot;)[1]))</span></span>
<span id="cb133-84"><a href="trrfer-subclone-analysis.html#cb133-84" tabindex="-1"></a>annot_cols <span class="ot">=</span> <span class="fu">list</span>(<span class="st">&quot;Grade Group&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Grade Group 1&quot;</span> <span class="ot">=</span> <span class="fu">pal_npg</span>()(<span class="dv">5</span>)[<span class="dv">1</span>], <span class="st">&quot;Grade Group 2&quot;</span> <span class="ot">=</span> <span class="fu">pal_npg</span>()(<span class="dv">5</span>)[<span class="dv">2</span>], <span class="st">&quot;Grade Group 3&quot;</span> <span class="ot">=</span> <span class="fu">pal_npg</span>()(<span class="dv">5</span>)[<span class="dv">3</span>],</span>
<span id="cb133-85"><a href="trrfer-subclone-analysis.html#cb133-85" tabindex="-1"></a>                                    <span class="st">&quot;Grade Group 4&quot;</span> <span class="ot">=</span> <span class="fu">pal_npg</span>()(<span class="dv">5</span>)[<span class="dv">4</span>], <span class="st">&quot;Grade Group 5&quot;</span> <span class="ot">=</span> <span class="fu">pal_npg</span>()(<span class="dv">5</span>)[<span class="dv">5</span>]))</span>
<span id="cb133-86"><a href="trrfer-subclone-analysis.html#cb133-86" tabindex="-1"></a>onco_annot <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="st">`</span><span class="at">Grade Group</span><span class="st">`</span> <span class="ot">=</span> clinical<span class="sc">$</span>grade_group, <span class="at">col =</span> annot_cols)</span>
<span id="cb133-87"><a href="trrfer-subclone-analysis.html#cb133-87" tabindex="-1"></a></span>
<span id="cb133-88"><a href="trrfer-subclone-analysis.html#cb133-88" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb133-89"><a href="trrfer-subclone-analysis.html#cb133-89" tabindex="-1"></a><span class="fu">oncoPrint</span>(mat,</span>
<span id="cb133-90"><a href="trrfer-subclone-analysis.html#cb133-90" tabindex="-1"></a>          <span class="at">alter_fun =</span> <span class="fu">list</span>(</span>
<span id="cb133-91"><a href="trrfer-subclone-analysis.html#cb133-91" tabindex="-1"></a>            <span class="at">background =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.9</span>, <span class="at">height =</span> <span class="fl">0.9</span>, <span class="at">fill =</span> <span class="st">&quot;#FFFFFF&quot;</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> .<span class="dv">1</span>),</span>
<span id="cb133-92"><a href="trrfer-subclone-analysis.html#cb133-92" tabindex="-1"></a>            <span class="at">Amplified =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.85</span>, <span class="at">height =</span> <span class="fl">0.9</span>, <span class="at">fill =</span> cols[<span class="st">&quot;Amplified&quot;</span>]),</span>
<span id="cb133-93"><a href="trrfer-subclone-analysis.html#cb133-93" tabindex="-1"></a>            <span class="at">Deleted =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.85</span>, <span class="at">height =</span> <span class="fl">0.9</span>, <span class="at">fill =</span> cols[<span class="st">&quot;Deleted&quot;</span>]),</span>
<span id="cb133-94"><a href="trrfer-subclone-analysis.html#cb133-94" tabindex="-1"></a>            <span class="at">SNV =</span> <span class="fu">alter_graphic</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">width =</span> <span class="fl">0.85</span>, <span class="at">height =</span> <span class="fl">0.45</span>, <span class="at">fill =</span> cols[<span class="st">&quot;SNV&quot;</span>])), </span>
<span id="cb133-95"><a href="trrfer-subclone-analysis.html#cb133-95" tabindex="-1"></a>          <span class="at">col =</span> cols, <span class="at">border =</span> <span class="st">&quot;black&quot;</span>, <span class="at">show_column_names =</span> F, <span class="at">show_row_names =</span> T, <span class="at">remove_empty_rows =</span> T,</span>
<span id="cb133-96"><a href="trrfer-subclone-analysis.html#cb133-96" tabindex="-1"></a>          <span class="at">show_pct =</span> F, <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>), <span class="at">top_annotation =</span> <span class="cn">NULL</span>, <span class="at">bottom_annotation =</span> onco_annot)</span></code></pre></div>
<p><img src="_main_files/figure-html/TCGA%20oncoprint-1.png" width="672" /></p>
</div>
<div id="targeted-sequencing-mutations" class="section level2 hasAnchor" number="18.4">
<h2><span class="header-section-number">18.4</span> Targeted sequencing mutations<a href="trrfer-subclone-analysis.html#targeted-sequencing-mutations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Finally, we also plot the mutation data of certain genes from our targeted deep sequencing.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="trrfer-subclone-analysis.html#cb134-1" tabindex="-1"></a>muts <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./data/mutations/P3_filtered_SNVs.tsv&quot;</span>)</span>
<span id="cb134-2"><a href="trrfer-subclone-analysis.html#cb134-2" tabindex="-1"></a>annot <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">&quot;./annotation/P3.tsv&quot;</span>, <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;library&quot;</span>, <span class="st">&quot;section&quot;</span>, <span class="st">&quot;pathology&quot;</span>))</span>
<span id="cb134-3"><a href="trrfer-subclone-analysis.html#cb134-3" tabindex="-1"></a></span>
<span id="cb134-4"><a href="trrfer-subclone-analysis.html#cb134-4" tabindex="-1"></a><span class="co"># Remove underscore in sample</span></span>
<span id="cb134-5"><a href="trrfer-subclone-analysis.html#cb134-5" tabindex="-1"></a>muts[, SAMPLE <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, SAMPLE)]</span>
<span id="cb134-6"><a href="trrfer-subclone-analysis.html#cb134-6" tabindex="-1"></a></span>
<span id="cb134-7"><a href="trrfer-subclone-analysis.html#cb134-7" tabindex="-1"></a><span class="co"># Remove LOC/2nd genes</span></span>
<span id="cb134-8"><a href="trrfer-subclone-analysis.html#cb134-8" tabindex="-1"></a>muts[, GeneName <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;LOC.*:&quot;</span>, <span class="st">&quot;&quot;</span>, GeneName)]</span>
<span id="cb134-9"><a href="trrfer-subclone-analysis.html#cb134-9" tabindex="-1"></a>muts[, GeneName <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>, <span class="st">&quot;&quot;</span>, GeneName)]</span>
<span id="cb134-10"><a href="trrfer-subclone-analysis.html#cb134-10" tabindex="-1"></a></span>
<span id="cb134-11"><a href="trrfer-subclone-analysis.html#cb134-11" tabindex="-1"></a><span class="co"># Get number of mutations</span></span>
<span id="cb134-12"><a href="trrfer-subclone-analysis.html#cb134-12" tabindex="-1"></a>num_muts <span class="ot">=</span> muts[, .N, by <span class="ot">=</span> SAMPLE]</span>
<span id="cb134-13"><a href="trrfer-subclone-analysis.html#cb134-13" tabindex="-1"></a>num_muts[, SAMPLE <span class="sc">:=</span> <span class="fu">gsub</span>(<span class="st">&quot;P.&quot;</span>, <span class="st">&quot;&quot;</span>, SAMPLE)]</span>
<span id="cb134-14"><a href="trrfer-subclone-analysis.html#cb134-14" tabindex="-1"></a><span class="fu">setnames</span>(num_muts, <span class="fu">c</span>(<span class="st">&quot;section&quot;</span>, <span class="st">&quot;n_muts&quot;</span>))</span>
<span id="cb134-15"><a href="trrfer-subclone-analysis.html#cb134-15" tabindex="-1"></a></span>
<span id="cb134-16"><a href="trrfer-subclone-analysis.html#cb134-16" tabindex="-1"></a><span class="co"># Merge with annotation</span></span>
<span id="cb134-17"><a href="trrfer-subclone-analysis.html#cb134-17" tabindex="-1"></a>num_muts <span class="ot">=</span> <span class="fu">merge</span>(num_muts, annot, <span class="at">by =</span> <span class="st">&quot;section&quot;</span>)</span>
<span id="cb134-18"><a href="trrfer-subclone-analysis.html#cb134-18" tabindex="-1"></a></span>
<span id="cb134-19"><a href="trrfer-subclone-analysis.html#cb134-19" tabindex="-1"></a><span class="co"># Add coordinates</span></span>
<span id="cb134-20"><a href="trrfer-subclone-analysis.html#cb134-20" tabindex="-1"></a>num_muts[, x <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb134-21"><a href="trrfer-subclone-analysis.html#cb134-21" tabindex="-1"></a>num_muts[, y <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb134-22"><a href="trrfer-subclone-analysis.html#cb134-22" tabindex="-1"></a></span>
<span id="cb134-23"><a href="trrfer-subclone-analysis.html#cb134-23" tabindex="-1"></a><span class="co"># Fill dt</span></span>
<span id="cb134-24"><a href="trrfer-subclone-analysis.html#cb134-24" tabindex="-1"></a>fill_dt <span class="ot">=</span> <span class="fu">data.table</span>(<span class="at">section =</span> annot[<span class="sc">!</span>section <span class="sc">%in%</span> num_muts<span class="sc">$</span>section, section],</span>
<span id="cb134-25"><a href="trrfer-subclone-analysis.html#cb134-25" tabindex="-1"></a>                     <span class="at">n_muts =</span> <span class="cn">NA</span>,</span>
<span id="cb134-26"><a href="trrfer-subclone-analysis.html#cb134-26" tabindex="-1"></a>                     <span class="at">library =</span> <span class="cn">NA</span>,</span>
<span id="cb134-27"><a href="trrfer-subclone-analysis.html#cb134-27" tabindex="-1"></a>                     <span class="at">pathology =</span> <span class="cn">NA</span>)</span>
<span id="cb134-28"><a href="trrfer-subclone-analysis.html#cb134-28" tabindex="-1"></a>fill_dt[, x <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb134-29"><a href="trrfer-subclone-analysis.html#cb134-29" tabindex="-1"></a>fill_dt[, y <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, section))]</span>
<span id="cb134-30"><a href="trrfer-subclone-analysis.html#cb134-30" tabindex="-1"></a>num_muts <span class="ot">=</span> <span class="fu">rbind</span>(num_muts, fill_dt)</span>
<span id="cb134-31"><a href="trrfer-subclone-analysis.html#cb134-31" tabindex="-1"></a></span>
<span id="cb134-32"><a href="trrfer-subclone-analysis.html#cb134-32" tabindex="-1"></a><span class="co"># Plot gene specific heatmaps</span></span>
<span id="cb134-33"><a href="trrfer-subclone-analysis.html#cb134-33" tabindex="-1"></a>genelist <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;LRP1B&quot;</span>, <span class="st">&quot;SPTA1&quot;</span>, <span class="st">&quot;SPOP&quot;</span>, <span class="st">&quot;FOXA1&quot;</span>, <span class="st">&quot;FOXP1&quot;</span>)</span>
<span id="cb134-34"><a href="trrfer-subclone-analysis.html#cb134-34" tabindex="-1"></a></span>
<span id="cb134-35"><a href="trrfer-subclone-analysis.html#cb134-35" tabindex="-1"></a><span class="fu">lapply</span>(genelist, <span class="cf">function</span>(gene) {</span>
<span id="cb134-36"><a href="trrfer-subclone-analysis.html#cb134-36" tabindex="-1"></a>  dt <span class="ot">=</span> muts[GeneName <span class="sc">==</span> gene, .(<span class="at">count =</span> .N), by <span class="ot">=</span> .(SAMPLE)]</span>
<span id="cb134-37"><a href="trrfer-subclone-analysis.html#cb134-37" tabindex="-1"></a>  </span>
<span id="cb134-38"><a href="trrfer-subclone-analysis.html#cb134-38" tabindex="-1"></a>  <span class="co"># Add sections</span></span>
<span id="cb134-39"><a href="trrfer-subclone-analysis.html#cb134-39" tabindex="-1"></a>  dt <span class="ot">=</span> <span class="fu">rbind</span>(dt, num_muts[, .(<span class="at">SAMPLE =</span> section, <span class="at">count =</span> <span class="cn">NA</span>)])</span>
<span id="cb134-40"><a href="trrfer-subclone-analysis.html#cb134-40" tabindex="-1"></a>  dt <span class="ot">=</span> dt[, .(<span class="at">count =</span> <span class="fu">sum</span>(count, <span class="at">na.rm =</span> T)), by <span class="ot">=</span> SAMPLE]</span>
<span id="cb134-41"><a href="trrfer-subclone-analysis.html#cb134-41" tabindex="-1"></a>  dt[count <span class="sc">==</span> <span class="dv">0</span>, count <span class="sc">:=</span> <span class="cn">NA</span>]</span>
<span id="cb134-42"><a href="trrfer-subclone-analysis.html#cb134-42" tabindex="-1"></a>  </span>
<span id="cb134-43"><a href="trrfer-subclone-analysis.html#cb134-43" tabindex="-1"></a>  <span class="co"># Add coordinates</span></span>
<span id="cb134-44"><a href="trrfer-subclone-analysis.html#cb134-44" tabindex="-1"></a>  dt[, x <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L|C.&quot;</span>, <span class="st">&quot;&quot;</span>, SAMPLE))]</span>
<span id="cb134-45"><a href="trrfer-subclone-analysis.html#cb134-45" tabindex="-1"></a>  dt[, y <span class="sc">:=</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;L.|C&quot;</span>, <span class="st">&quot;&quot;</span>, SAMPLE))]</span>
<span id="cb134-46"><a href="trrfer-subclone-analysis.html#cb134-46" tabindex="-1"></a>  </span>
<span id="cb134-47"><a href="trrfer-subclone-analysis.html#cb134-47" tabindex="-1"></a>  <span class="co"># Plot</span></span>
<span id="cb134-48"><a href="trrfer-subclone-analysis.html#cb134-48" tabindex="-1"></a>  <span class="fu">ggplot</span>(dt, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> count)) <span class="sc">+</span></span>
<span id="cb134-49"><a href="trrfer-subclone-analysis.html#cb134-49" tabindex="-1"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb134-50"><a href="trrfer-subclone-analysis.html#cb134-50" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> gene) <span class="sc">+</span></span>
<span id="cb134-51"><a href="trrfer-subclone-analysis.html#cb134-51" tabindex="-1"></a>    <span class="fu">scale_fill_distiller</span>(<span class="at">name =</span> <span class="st">&quot;Number of somatic</span><span class="sc">\n</span><span class="st">non-synonymous</span><span class="sc">\n</span><span class="st">mutations detected&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="at">direction =</span> <span class="dv">1</span>, <span class="at">na.value =</span> <span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb134-52"><a href="trrfer-subclone-analysis.html#cb134-52" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(dt<span class="sc">$</span>y), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb134-53"><a href="trrfer-subclone-analysis.html#cb134-53" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">seq</span>(<span class="at">from =</span> .<span class="dv">5</span>, <span class="at">to =</span> <span class="fu">max</span>(dt<span class="sc">$</span>x), <span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb134-54"><a href="trrfer-subclone-analysis.html#cb134-54" tabindex="-1"></a>    <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>y)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>y))) <span class="sc">+</span> </span>
<span id="cb134-55"><a href="trrfer-subclone-analysis.html#cb134-55" tabindex="-1"></a>    <span class="fu">scale_x_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>x)), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">max</span>(dt<span class="sc">$</span>x))) <span class="sc">+</span></span>
<span id="cb134-56"><a href="trrfer-subclone-analysis.html#cb134-56" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb134-57"><a href="trrfer-subclone-analysis.html#cb134-57" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb134-58"><a href="trrfer-subclone-analysis.html#cb134-58" tabindex="-1"></a>})</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="_main_files/figure-html/targeted%20sequencing-1.png" width="672" /></p>
<pre><code>## 
## [[2]]</code></pre>
<p><img src="_main_files/figure-html/targeted%20sequencing-2.png" width="672" /></p>
<pre><code>## 
## [[3]]</code></pre>
<p><img src="_main_files/figure-html/targeted%20sequencing-3.png" width="672" /></p>
<pre><code>## 
## [[4]]</code></pre>
<p><img src="_main_files/figure-html/targeted%20sequencing-4.png" width="672" /></p>
<pre><code>## 
## [[5]]</code></pre>
<p><img src="_main_files/figure-html/targeted%20sequencing-5.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="validation-of-pseudodiploid-deletions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cosmic-deletions-and-targeted-panel-mutations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
